{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "83c24ee5-187b-44a4-8ed3-7986cbf1273d",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import sys\n",
    "# Use local biomni\n",
    "sys.path.insert(0, \"/home/runai-home/Biomni/\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "edc8be00-e387-411f-8f29-2a2577de4f12",
   "metadata": {},
   "outputs": [],
   "source": [
    "from pathlib import Path\n",
    "from collections import defaultdict\n",
    "from difflib import SequenceMatcher"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "25b9c924-7603-4553-97e7-661da03bb0a0",
   "metadata": {},
   "outputs": [],
   "source": [
    "from biomni.cogzero.canonicalizer import canonicalize_output, compare_outputs\n",
    "from biomni.cogzero.utils import load_data, save_data\n",
    "from biomni.cogzero.nb_utils import *"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "2f2a8f75-498e-434e-80d4-b778de4db204",
   "metadata": {},
   "outputs": [],
   "source": [
    "ver = \"v0\"\n",
    "tasks_dir = Path(f\"/mlbio_scratch/kodryan/Biomni/data/generated/tasks/{ver}\")\n",
    "raw_tasks_dir = tasks_dir / \"raw\"\n",
    "all_tasks_dict = load_data_dict(raw_tasks_dir)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "4a9e0be9-6844-438d-a4c1-c1e6dd13820f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "trial_0:\t21 tasks\n",
      "trial_1:\t20 tasks\n",
      "trial_2:\t20 tasks\n",
      "trial_3:\t21 tasks\n",
      "trial_4:\t21 tasks\n",
      "trial_5:\t21 tasks\n",
      "trial_6:\t21 tasks\n",
      "trial_7:\t21 tasks\n",
      "trial_8:\t21 tasks\n",
      "trial_9:\t21 tasks\n",
      "trial_10:\t21 tasks\n",
      "trial_11:\t21 tasks\n",
      "trial_12:\t21 tasks\n",
      "trial_13:\t21 tasks\n",
      "trial_14:\t21 tasks\n",
      "trial_15:\t21 tasks\n",
      "-------------------------\n",
      "Total:  \t334 tasks\n"
     ]
    }
   ],
   "source": [
    "clean_tasks = []\n",
    "\n",
    "for trial, tasks in all_tasks_dict.items():\n",
    "    print(f\"{trial}:\\t{len(tasks)} tasks\")\n",
    "    for task in tasks:\n",
    "        clean_tasks.append(task)\n",
    "\n",
    "print(25 * '-')\n",
    "print(f\"Total:  \\t{len(clean_tasks)} tasks\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "dc1983e0-de89-42f5-9cf0-e91e5294f874",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_tasks_by_gta(task_list: list[dict[str, str]], sim_thresh=0.9) -> dict[str, list[dict[str, str]]]:\n",
    "    \"\"\"\n",
    "    Bin tasks by ground-truth answers.\n",
    "    Tasks with similar answers end up in the same bin.\n",
    "    \"\"\"\n",
    "    tasks_by_gta = defaultdict(list)\n",
    "\n",
    "    for task in task_list:\n",
    "        task_gta = task[\"ground_truth_answer\"]\n",
    "        task_key = canonicalize_output(task_gta.lower())\n",
    "        sm = SequenceMatcher(a=task_key)\n",
    "        \n",
    "        bin_key = task_key\n",
    "        for key in tasks_by_gta:\n",
    "            sm.set_seq2(key)\n",
    "            if sm.ratio() > sim_thresh:\n",
    "                bin_key = key\n",
    "                break\n",
    "                \n",
    "        tasks_by_gta[bin_key].append(task)\n",
    "\n",
    "    return tasks_by_gta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "f6ff4b46-a31c-4698-a99b-8085629b7011",
   "metadata": {},
   "outputs": [],
   "source": [
    "tasks_by_gta = get_tasks_by_gta(clean_tasks)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "f7c4610c-f49c-4b3a-a295-14804470ef39",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "nonmalignant                                                                                        1 tasks\n",
      "nb_7767_3495_reg1                                                                                   1 tasks\n",
      "paep                                                                                                2 tasks\n",
      "lymphocyte                                                                                          5 tasks\n",
      "natural_killer_cell                                                                                 1 tasks\n",
      "monocyte                                                                                            2 tasks\n",
      "3357                                                                                                3 tasks\n",
      "ensg00000286009                                                                                     1 tasks\n",
      "gabaergic neuron                                                                                    1 tasks\n",
      "hoxd13                                                                                              1 tasks\n",
      "cd4.memory                                                                                          1 tasks\n",
      "15-20                                                                                               1 tasks\n",
      "ac022414.1                                                                                          1 tasks\n",
      "nmj accessory                                                                                       5 tasks\n",
      "mf-i                                                                                                3 tasks\n",
      "goblet cell                                                                                         6 tasks\n",
      "b_cell                                                                                              3 tasks\n",
      "r24_p2538                                                                                           2 tasks\n",
      "csf3                                                                                                1 tasks\n",
      "uba52                                                                                               1 tasks\n",
      "tom_vct                                                                                             2 tasks\n",
      "hepatocyte                                                                                          2 tasks\n",
      "5                                                                                                   2 tasks\n",
      "4                                                                                                   3 tasks\n",
      "cardiac_endothelial_cell                                                                            1 tasks\n",
      "rap1gap2                                                                                            1 tasks\n",
      "control                                                                                             9 tasks\n",
      "3427                                                                                                2 tasks\n",
      "l5 extratelencephalic projecting glutamatergic cortical neuron                                      1 tasks\n",
      "chondroprog::wholelimb                                                                              1 tasks\n",
      "naive.bcells.isghi                                                                                  1 tasks\n",
      "xist                                                                                                5 tasks\n",
      "myh1+myh7+                                                                                          1 tasks\n",
      "b-lymphocytes                                                                                       3 tasks\n",
      "alzheimer disease                                                                                   2 tasks\n",
      "cl:2000060                                                                                          1 tasks\n",
      "interferon-response                                                                                 2 tasks\n",
      "csh1                                                                                                2 tasks\n",
      "cyp11b2                                                                                             1 tasks\n",
      "anbl12p1-like chemotherapy                                                                          5 tasks\n",
      "mt-co3                                                                                              3 tasks\n",
      "plasma cell                                                                                         1 tasks\n",
      "naaladl2                                                                                            1 tasks\n",
      "69.5100                                                                                             1 tasks\n",
      "arhgap24                                                                                            1 tasks\n",
      "macro                                                                                               1 tasks\n",
      "mono.il1b+                                                                                          1 tasks\n",
      "464c                                                                                                2 tasks\n",
      "nuclei                                                                                              2 tasks\n",
      "mf-isn(fg)                                                                                          5 tasks\n",
      "cells                                                                                               1 tasks\n",
      "9                                                                                                   6 tasks\n",
      "b cell                                                                                              4 tasks\n",
      "1.2370                                                                                              1 tasks\n",
      "scdonor_tg1_mother                                                                                  1 tasks\n",
      "tchh                                                                                                1 tasks\n",
      "chp667                                                                                              1 tasks\n",
      "13                                                                                                  2 tasks\n",
      "pkhd1l1+ ec                                                                                         1 tasks\n",
      "cd4t_cm                                                                                             1 tasks\n",
      "endothelial_cell                                                                                    4 tasks\n",
      "adipocyte of omentum tissue                                                                         4 tasks\n",
      "grin2a                                                                                              2 tasks\n",
      "wholelimb                                                                                           1 tasks\n",
      "caecum                                                                                              2 tasks\n",
      "35-40                                                                                               1 tasks\n",
      "glis3-as1                                                                                           1 tasks\n",
      "astrocyte                                                                                           1 tasks\n",
      "hofbauer cell                                                                                       1 tasks\n",
      "6720483e21rik                                                                                       3 tasks\n",
      "evt_late_3                                                                                          5 tasks\n",
      "cortical cell of adrenal gland                                                                      1 tasks\n",
      "chp667, inc doxorubicin                                                                             1 tasks\n",
      "cardiac_neuron                                                                                      2 tasks\n",
      "cxcl9                                                                                               1 tasks\n",
      "b cell::3350                                                                                        1 tasks\n",
      "1                                                                                                   1 tasks\n",
      "in-mge-pv                                                                                           1 tasks\n",
      "primitive red blood cell                                                                            1 tasks\n",
      "mesenteric lymph node                                                                               3 tasks\n",
      "tnfrsf12a+muscs                                                                                     3 tasks\n",
      "rpl10                                                                                               1 tasks\n",
      "500c                                                                                                2 tasks\n",
      "granulosa_cell                                                                                      1 tasks\n",
      "r24_p2324                                                                                           1 tasks\n",
      "lm                                                                                                  1 tasks\n",
      "8                                                                                                   5 tasks\n",
      "14                                                                                                  3 tasks\n",
      "3                                                                                                   3 tasks\n",
      "20                                                                                                  2 tasks\n",
      "central memory cd4-positive, alpha-beta t cell                                                      2 tasks\n",
      "fp236383.1                                                                                          2 tasks\n",
      "3384                                                                                                1 tasks\n",
      "en-l2_3-it                                                                                          1 tasks\n",
      "mes3                                                                                                1 tasks\n",
      "young                                                                                               2 tasks\n",
      "mf-iisc(fg)                                                                                         2 tasks\n",
      "tg                                                                                                  5 tasks\n",
      "syncytiotrophoblast cell                                                                            2 tasks\n",
      "adrn-calcium                                                                                        1 tasks\n",
      "anbl 09p1                                                                                           2 tasks\n",
      "kcne1                                                                                               1 tasks\n",
      "va8                                                                                                 1 tasks\n",
      "progenitor                                                                                          1 tasks\n",
      "hoxa10                                                                                              1 tasks\n",
      "ilc3s.rorgt+                                                                                        1 tasks\n",
      "tnfrsf12a_plus_muscs                                                                                1 tasks\n",
      "mesothelium                                                                                         1 tasks\n",
      "mf_typei(cytoplasmic)                                                                               2 tasks\n",
      "nmj                                                                                                 1 tasks\n",
      "r24_p2392::oligodendrocyte                                                                          1 tasks\n",
      "ralyl                                                                                               1 tasks\n",
      "cxcl10                                                                                              1 tasks\n",
      "va1                                                                                                 1 tasks\n",
      "first_trimester                                                                                     1 tasks\n",
      "5386stdy7537944                                                                                     1 tasks\n",
      "rps4y1                                                                                              1 tasks\n",
      "adipocyte                                                                                           1 tasks\n",
      "myh2+myh7+                                                                                          2 tasks\n",
      "70-75                                                                                               1 tasks\n",
      "oocyte                                                                                              1 tasks\n",
      "pcdh9                                                                                               1 tasks\n",
      "extravillous trophoblast                                                                            1 tasks\n",
      "kidney cell                                                                                         1 tasks\n",
      "2                                                                                                   2 tasks\n",
      "anbl0532-like therapy with cisplatin & etoposide                                                    2 tasks\n",
      "cpa3                                                                                                1 tasks\n",
      "cytotoxic t cell                                                                                    1 tasks\n",
      "3313                                                                                                2 tasks\n",
      "oligodendrocyte                                                                                     1 tasks\n",
      "neuron                                                                                              2 tasks\n",
      "cd3g                                                                                                1 tasks\n",
      "mf_typeii(fg)                                                                                       1 tasks\n",
      "hyb                                                                                                 1 tasks\n",
      "stem_cell                                                                                           1 tasks\n",
      "stroma a                                                                                            1 tasks\n",
      "inhibitory interneuron                                                                              1 tasks\n",
      "pooled_deprived_p7                                                                                  1 tasks\n",
      "synj2bp-cox16                                                                                       1 tasks\n",
      "mural_cell                                                                                          1 tasks\n",
      "mis-c                                                                                               3 tasks\n",
      "mo_cd16                                                                                             1 tasks\n",
      "3382                                                                                                1 tasks\n",
      "adh+fibro                                                                                           2 tasks\n",
      "b cell zone reticular cell                                                                          1 tasks\n",
      "myeov                                                                                               1 tasks\n",
      "cd16-nk                                                                                             1 tasks\n",
      "mature_enterocyte                                                                                   1 tasks\n",
      "gpr141                                                                                              1 tasks\n",
      "pf                                                                                                  3 tasks\n",
      "g1/m phase early                                                                                    1 tasks\n",
      "neuroblast (sensu vertebrata)                                                                       1 tasks\n",
      "zic2                                                                                                1 tasks\n",
      "mcm8-as1                                                                                            1 tasks\n",
      "l4 intratelencephalic projecting glutamatergic neuron                                               1 tasks\n",
      "rdh10+distalmes                                                                                     1 tasks\n",
      "cd36                                                                                                1 tasks\n",
      "25-30                                                                                               1 tasks\n",
      "20-25                                                                                               1 tasks\n",
      "luzp4                                                                                               1 tasks\n",
      "ighg1                                                                                               1 tasks\n",
      "hofbauer cell::tg                                                                                   1 tasks\n",
      "cxcl8                                                                                               1 tasks\n",
      "primary tumor resection + laparotomy post 3 cycles of induction chemo per anbl12p1                  1 tasks\n",
      "mybpc3                                                                                              1 tasks\n",
      "effector memory cd8-positive, alpha-beta t cell                                                     1 tasks\n",
      "3431                                                                                                1 tasks\n",
      "nrgn::glutamatergic neuron                                                                          1 tasks\n",
      "transmes                                                                                            1 tasks\n",
      "fos                                                                                                 1 tasks\n",
      "myh1+myh2+myh7+                                                                                     2 tasks\n",
      "25                                                                                                  1 tasks\n",
      "11                                                                                                  2 tasks\n",
      "slfn14                                                                                              1 tasks\n",
      "zfp36                                                                                               1 tasks\n",
      "fibroblast                                                                                          2 tasks\n",
      "mt3                                                                                                 1 tasks\n",
      "mp_lyve1lo/mhcii_hi                                                                                 1 tasks\n",
      "t cell                                                                                              1 tasks\n",
      "ark35::0.0314                                                                                       1 tasks\n",
      "small intestine peyer's patch                                                                       3 tasks\n",
      "b-plasma                                                                                            1 tasks\n",
      "pericyte::pick disease                                                                              1 tasks\n",
      "15                                                                                                  1 tasks\n",
      "anbl12p1-like                                                                                       1 tasks\n",
      "myocarditis                                                                                         1 tasks\n",
      "mis-c::t_prolif                                                                                     1 tasks\n",
      "anxa1                                                                                               1 tasks\n",
      "mes1::wholelimb                                                                                     1 tasks\n",
      "cd74::dc.il1b_high.cd103+                                                                           1 tasks\n",
      "zfand5                                                                                              1 tasks\n",
      "specialised_mf_populations                                                                          1 tasks\n",
      "mf_type-myh8+                                                                                       2 tasks\n",
      "phagocyte                                                                                           1 tasks\n",
      "r24_p2392                                                                                           1 tasks\n",
      "srcin1                                                                                              1 tasks\n",
      "6                                                                                                   1 tasks\n",
      "malat1                                                                                              1 tasks\n",
      "unknown                                                                                             1 tasks\n",
      "5478stdy7935101                                                                                     1 tasks\n",
      "tenocyte                                                                                            1 tasks\n",
      "26                                                                                                  1 tasks\n",
      "r24_p2730                                                                                           1 tasks\n",
      "sct                                                                                                 1 tasks\n",
      "nrg1                                                                                                1 tasks\n",
      "ch_18                                                                                               1 tasks\n",
      "3350::9                                                                                             1 tasks\n",
      "147                                                                                                 1 tasks\n",
      "thigh                                                                                               1 tasks\n",
      "rbc                                                                                                 1 tasks\n",
      "pericyte                                                                                            1 tasks\n",
      "microglial cell                                                                                     1 tasks\n"
     ]
    }
   ],
   "source": [
    "nonsingle_keys = []\n",
    "\n",
    "for gta_key, tasks in tasks_by_gta.items():\n",
    "    n_tasks = len(tasks)\n",
    "    print(f\"{gta_key:100}{n_tasks} tasks\")\n",
    "    if n_tasks > 1:\n",
    "        nonsingle_keys.append(gta_key)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "1f4f170b-c7ae-4dca-b65b-f7271525d9a6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "All keys: 212 \t nonsingle keys: 63\n",
      "All tasks: 334 \t nonsingle tasks: 185\n"
     ]
    }
   ],
   "source": [
    "print(f\"All keys: {len(tasks_by_gta)} \\t nonsingle keys: {len(nonsingle_keys)}\")\n",
    "print(f\"All tasks: {sum(len(tasks) for tasks in tasks_by_gta.values())} \\t \\\n",
    "nonsingle tasks: {sum(len(tasks) for key, tasks in tasks_by_gta.items() if key in nonsingle_keys)}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "b46891de-2efa-4aa3-8389-a234a7003e1a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Group paep\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper2/data/dataset3.h5ad\n",
      "Description: In the snRNA-seq dataset of macrophages from high‑risk neuroblastoma before (DX) and after (PTX) chemotherapy, identify the gene that shows the strongest statistically significant up‑regulation (largest positive log₂ fold‑change, FDR < 0.05) in post‑treatment macrophages compared to diagnostic macrophages. Answer format: a single HGNC gene symbol (shape: gene_symbol).\n",
      "Answer: PAEP\n",
      "\n",
      "Task 1\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper2/data/dataset3.h5ad\n",
      "Description: Within the macrophage nuclei, identify the gene that shows the strongest statistically significant up‑regulation (largest positive log2‑fold‑change, adjusted p‑value < 0.05) in the PTX stage compared to the DX stage. Answer format: a single HGNC gene symbol (shape: gene_symbol).\n",
      "Answer: PAEP\n",
      "\n",
      "Group lymphocyte\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper4/data/dataset1.h5ad\n",
      "Description: From the provided single‑cell RNA‑seq dataset, identify which cell type shows the greatest enrichment in the MIS-C condition relative to the Control condition when comparing cell‑type proportions (MIS-C proportion divided by Control proportion). Answer format: a single cell type label exactly as it appears in the 'cell_type' column (shape: label).\n",
      "Answer: lymphocyte\n",
      "\n",
      "Task 1\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper4/data/dataset1.h5ad\n",
      "Description: Using the provided single‑cell dataset, compute the proportion of each cell type within the Control group and within all disease groups combined (any Condition not equal to 'Control'). Calculate the log2 fold‑change of disease proportion over control proportion for each cell type (add a pseudocount of 1e-6 to avoid division by zero). Identify the cell type with the highest log2 fold‑change. Answer format: a single cell‑type label exactly as it appears in the 'cell_type' column (shape: label).\n",
      "Answer: lymphocyte\n",
      "\n",
      "Task 2\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper4/data/dataset1.h5ad\n",
      "Description: Compute the proportion of each cell type (from the 'cell_type' column) within each condition in the dataset. Then calculate the fold‑change of the proportion in the 'MIS-C' condition relative to the 'Control' condition (MIS-C proportion divided by Control proportion). Identify the cell type with the highest fold‑change. Answer format: a single cell type label exactly as it appears in the 'cell_type' column (case‑sensitive, shape: label).\n",
      "Answer: lymphocyte\n",
      "\n",
      "Task 3\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper4/data/dataset1.h5ad\n",
      "Description: Using the provided AnnData object, compute for each cell type the proportion of cells belonging to the disease 'myocarditis' and the proportion belonging to 'healthy'. Calculate the log2 fold‑change of these proportions (log2(prop_myocarditis / prop_healthy)) for each cell type. Identify the cell type with the highest positive log2 fold‑change (i.e., the strongest enrichment in myocarditis relative to healthy). Answer format: a single cell type label exactly as it appears in the 'cell_type' column (shape: label).\n",
      "Answer: lymphocyte\n",
      "\n",
      "Task 4\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper4/data/dataset1.h5ad\n",
      "Description: Using the provided single‑cell RNA‑seq dataset of human cardiac tissue, compute the proportion of each annotated cell type within the two disease conditions ('healthy' and 'myocarditis'). Identify the cell type whose proportion shows the greatest positive log2 fold‑change (myocarditis vs healthy). Answer format: a single cell‑type label exactly as it appears in the 'cell_type' column (shape: label).\n",
      "Answer: lymphocyte\n",
      "\n",
      "Group monocyte\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper4/data/my_dataset.h5ad\n",
      "Description: Identify the cell type that is most enriched in the MIS-C condition relative to the Control condition, using an odds ratio calculated from a 2 × 2 contingency table (add a 0.5 pseudocount to each cell to avoid division by zero). Answer format: a single lowercase label with underscores (shape: label).\n",
      "Answer: monocyte\n",
      "\n",
      "Task 1\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper4/data/my_dataset.h5ad\n",
      "Description: Using the provided AnnData object, calculate the proportion of cells labeled as 'MIS-C' within each coarse cell type found in the 'cell_type' column. Identify the cell type that has the highest proportion of MIS-C cells. Answer format: a single cell type label exactly as it appears in the 'cell_type' column (shape: label).\n",
      "Answer: monocyte\n",
      "\n",
      "Group 3357\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper5/data/dataset2.h5ad\n",
      "Description: From the provided snRNA‑seq dataset of human visceral adipose tissue, compute for each donor (column 'donor_id') the proportion of nuclei annotated as \"macrophage\" in the 'cell_type' column. Identify the donor with the highest macrophage proportion. Answer format: a single donor identifier (shape: label).\n",
      "Answer: 3357\n",
      "\n",
      "Task 1\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper5/data/dataset2.h5ad\n",
      "Description: Identify the donor (donor_id) that contributes the highest proportion of macrophage cells in the provided visceral adipose snRNA‑seq dataset. Answer format: a single donor identifier exactly as it appears in the donor_id column (shape: label).\n",
      "Answer: 3357\n",
      "\n",
      "Task 2\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper5/data/dataset2.h5ad\n",
      "Description: Identify the donor (donor_id) that exhibits the highest proportion of cells annotated as \"macrophage\" among all donors in the provided single-nucleus RNA‑seq dataset. Answer format: a single donor ID string (shape: label).\n",
      "Answer: 3357\n",
      "\n",
      "Group nmj accessory\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/nature/paper2/processed_myonuclei_data.h5ad\n",
      "Description: Using the provided myonuclei single-nucleus RNA‑seq dataset, compute for each cell type in the 'annotation_level1' annotation the proportion of nuclei belonging to the 'old' age bin and the proportion belonging to the 'young' age bin. Add a pseudocount of 1 to the counts of each age bin for every cell type to avoid division‑by‑zero. Calculate the log2 enrichment as log2(old_proportion / young_proportion). Identify the cell type with the highest log2 enrichment (i.e., strongest relative enrichment in old nuclei). Answer format: a single label matching exactly one entry of the 'annotation_level1' column (case‑sensitive, including spaces).\n",
      "Answer: NMJ accessory\n",
      "\n",
      "Task 1\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/nature/paper2/processed_myonuclei_data.h5ad\n",
      "Description: Across the donor age groups present in the dataset, compute the proportion of cells belonging to each annotation_level1 cell type within each age group, then calculate the Spearman rank correlation between the numeric midpoint of the age group and the cell‑type proportion. Identify the cell type that shows the strongest positive monotonic trend (i.e., the highest Spearman rho). Answer format: a single cell‑type label exactly as it appears in the annotation_level1 column (shape: label).\n",
      "Answer: NMJ accessory\n",
      "\n",
      "Task 2\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/nature/paper2/processed_myonuclei_data.h5ad\n",
      "Description: From the provided muscle nuclei single-nucleus RNA‑seq dataset, compute for each cell type (column 'annotation_level1') the proportion of cells belonging to the 'old' age bin (Age_bin == 'old'). Identify the cell type with the highest proportion of old cells. Answer format: a single label exactly matching the value in the 'annotation_level1' column (case‑sensitive, may contain spaces).\n",
      "Answer: NMJ accessory\n",
      "\n",
      "Task 3\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/nature/paper2/processed_myonuclei_data.h5ad\n",
      "Description: Using the provided myonuclei dataset, for each SampleID compute the proportion of nuclei belonging to each 'annotation_level1' cell type. Convert each sample's 'Age_group' (e.g., \"15-20\") to the numeric midpoint (e.g., 17.5). Calculate the Spearman rank correlation (ρ) between the cell‑type proportion and the numeric age midpoint across samples. Identify the cell type with the strongest positive age‑associated increase (largest positive ρ). Answer format: a single label exactly as it appears in the 'annotation_level1' column (shape: label).\n",
      "Answer: NMJ accessory\n",
      "\n",
      "Task 4\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/nature/paper2/processed_myonuclei_data.h5ad\n",
      "Description: Using the provided muscle nuclei dataset, compute for each cell type annotation (annotation_level1) the proportion of nuclei belonging to that annotation within each age group. Convert the age groups to numeric midpoints (15-20 → 17.5, 20-25 → 22.5, 25-30 → 27.5, 35-40 → 37.5, 55-60 → 57.5, 60-65 → 62.5, 70-75 → 72.5). For each annotation, calculate the Spearman correlation (ρ) between the numeric age and its proportion across the age groups. Identify the annotation with the highest positive Spearman ρ. Answer format: a single label exactly as it appears in the \"annotation_level1\" column (shape: label).\n",
      "Answer: NMJ accessory\n",
      "\n",
      "Group mf-i\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/nature/paper2/processed_myonucleicells_data.h5ad\n",
      "Description: From the AnnData file, calculate for each cell type in the column `annotation_level1` its proportion of cells within each batch (`batch` column, values 'cells' and 'nuclei'). Identify the cell type with the greatest absolute difference in proportion between the two batches. Provide the cell‑type label exactly as it appears in `annotation_level1`. Answer format: a single label (shape: label).\n",
      "Answer: MF-I\n",
      "\n",
      "Task 1\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/nature/paper2/processed_human_muscle_data.h5ad\n",
      "Description: Using the provided AnnData file, calculate the proportion of each annotation_level2 cell type within each assay batch (batch column, values cells and nuclei). Determine which cell type shows the greatest absolute difference in proportion between the two batches. Report the exact cell-type name as a single label (case‑sensitive, include any hyphens).\n",
      "Answer: MF-I\n",
      "\n",
      "Task 2\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/nature/paper2/processed_myonucleicells_data.h5ad\n",
      "Description: Using the provided AnnData object, compute the proportion of each broad cell type (annotation_level0) within each age bin (young vs old). Identify the cell type with the greatest absolute difference in proportion between the two age bins. Answer format: a single label exactly matching the cell type name as it appears in annotation_level0 (shape: label).\n",
      "Answer: MF-I\n",
      "\n",
      "Group goblet cell\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/nature/paper5/mouse_data.h5ad\n",
      "Description: Identify the epithelial cell type whose proportion of cells increases most strongly from the most proximal segment (segment 1) to the most distal segment (segment 30) of the mouse small intestine. For each segment, compute the proportion of each cell type (as defined in the 'cell.type' column) among all cells in that segment, fit a simple linear regression of proportion versus segment number (1‑30), and select the cell type with the largest positive slope. Answer format: a single cell type label exactly as it appears in the 'cell.type' column (case‑sensitive).\n",
      "Answer: Goblet cell\n",
      "\n",
      "Task 1\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/nature/paper5/mouse_data.h5ad\n",
      "Description: Across the 30 intestinal segments (1 = proximal, 30 = distal) in the provided mouse epithelial scRNA‑seq dataset, compute the proportion of each cell type within each segment and calculate the Spearman rank correlation between segment number and cell‑type proportion. Identify the cell type that shows the strongest positive monotonic trend (i.e., the highest Spearman rho). Answer format: a single cell‑type label exactly as it appears in the 'cell.type' column (shape: label).\n",
      "Answer: Goblet cell\n",
      "\n",
      "Task 2\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/nature/paper5/mouse_data.h5ad\n",
      "Description: Identify the epithelial cell type whose relative abundance (proportion of cells) shows the strongest monotonic increase from the proximal (segment 1) to the distal (segment 30) end of the mouse small intestine. Answer format: a single label exactly as it appears in the 'cell.type' column (shape: label).\n",
      "Answer: Goblet cell\n",
      "\n",
      "Task 3\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/nature/paper5/mouse_data.h5ad\n",
      "Description: For each of the 30 intestinal segments (1‑30) compute the proportion of cells belonging to each cell type (as listed in the 'cell.type' column). Then calculate the Spearman correlation between segment number and the proportion of each cell type. Identify the cell type with the highest positive Spearman correlation (strongest monotonic increase along the proximal‑distal axis). Provide the answer as a single label exactly matching the value in the 'cell.type' column (case‑sensitive, including spaces).\n",
      "Answer: Goblet cell\n",
      "\n",
      "Task 4\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/nature/paper5/mouse_data.h5ad\n",
      "Description: Identify the epithelial cell type whose proportion across intestinal segments (1 to 30) shows the strongest positive monotonic gradient (highest Spearman correlation) with segment number. Answer format: a single cell type label exactly as it appears in the 'cell.type' column (shape: label).\n",
      "Answer: Goblet cell\n",
      "\n",
      "Task 5\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/nature/paper5/mouse_data.h5ad\n",
      "Description: Identify the cell type whose proportion across the 30 intestinal segments shows the strongest positive monotonic trend (i.e., the highest Pearson correlation between segment number and the cell-type proportion). Answer format: a single cell type label exactly as it appears in the 'cell.type' column (shape: label).\n",
      "Answer: Goblet cell\n",
      "\n",
      "Group b_cell\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/nature/paper6/data.h5ad\n",
      "Description: Identify the cell type that exhibits the greatest relative increase in proportion in OLD mice compared to YOUNG mice within this dataset. Report the cell type name as a single lowercase token with any spaces replaced by underscores (shape: label).\n",
      "Answer: b_cell\n",
      "\n",
      "Task 1\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/nature/paper6/data.h5ad\n",
      "Description: Using the provided single‑cell RNA‑seq atlas of the aging mouse ovary, compute for each annotated cell type the gene with the largest absolute log2 fold‑change (log2FC) between OLD and YOUNG mice (log2FC = log2(mean_OLD / mean_YOUNG), using CPM‑normalized log1p expression). Identify the cell type whose maximal absolute log2FC is the greatest among all cell types. Answer format: a single lowercase label with underscores (shape: label).\n",
      "Answer: b_cell\n",
      "\n",
      "Task 2\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/nature/paper6/data.h5ad\n",
      "Description: Identify the cell type that shows the greatest enrichment in the OLD mice compared to YOUNG mice, measured as the fold‑change of its proportion (OLD/YOUNG). Report the cell type name as a single label with spaces replaced by underscores (shape: label).\n",
      "Answer: B_cell\n",
      "\n",
      "Group r24_p2538\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/cell/paper1/data.h5ad\n",
      "Description: From the provided human brain snRNA‑seq dataset, compute for each donor the percentage of cells annotated as \"microglial cell\" among that donor's cells. Identify the donor with the highest microglial proportion. Answer format: a single donor identifier (shape: label, exact case as in the data).\n",
      "Answer: R24_P2538\n",
      "\n",
      "Task 1\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/cell/paper1/data.h5ad\n",
      "Description: From the provided human brain snRNA‑seq dataset, identify the donor (donor_id) that has the highest fraction of cells labeled as \"microglial cell\". Answer format: a single donor identifier string exactly as it appears in the 'donor_id' column (shape: label).\n",
      "Answer: R24_P2538\n",
      "\n",
      "Group tom_vct\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/cell/paper5/data.h5ad\n",
      "Description: Using the provided single-cell RNA‑seq dataset of human trophoblast organoids, determine which annotated cell type shows the strongest transcriptional response to cytokine exposure. Define the response strength as the total number of genes that are significantly up‑regulated in the cytokine‑treated condition versus the no‑cytokine condition (adjusted p‑value < 0.05 and log2 fold‑change > 0). Report the cell‑type label exactly as it appears in the 'author_cell_type' column. Answer format: a single label (shape: label).\n",
      "Answer: TOM_VCT\n",
      "\n",
      "Task 1\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/cell/paper5/data.h5ad\n",
      "Description: Identify the author_cell_type (value in the 'author_cell_type' column of adata.obs) that exhibits the largest number of significantly up‑regulated genes (adjusted p‑value < 0.05 and log2 fold‑change > 0) when comparing cytokine‑treated cells versus no‑cytokine cells using Wilcoxon rank‑sum test. Answer format: a single cell‑type label (shape: label).\n",
      "Answer: TOM_VCT\n",
      "\n",
      "Group hepatocyte\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper2/data/dataset1.h5ad\n",
      "Description: Using the provided snRNA‑seq dataset, identify the cell type whose relative abundance increases the most after chemotherapy (Stage_Code = PTX) compared to diagnosis (Stage_Code = DX). For each cell type, compute its proportion of cells within each stage, calculate the log2 fold‑change (log2( proportion_PT X / proportion_DX )), and consider only cell types present in both stages. Report the exact cell_type label (as it appears in the 'cell_type' column) with the highest positive log2 fold‑change. Answer format: a single label string.\n",
      "Answer: hepatocyte\n",
      "\n",
      "Task 1\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper2/data/dataset1.h5ad\n",
      "Description: Using the provided snRNA‑seq dataset, compare gene expression between the two treatment stages (Stage_Code = DX for diagnosis and Stage_Code = PTX for post‑chemotherapy). For each cell type, compute the log2 fold‑change (log2FC) of every gene as log2((mean PTX + 1e‑6) / (mean DX + 1e‑6)) using raw counts. Identify, for each cell type, the gene with the highest positive log2FC, then determine which cell type has the largest such positive log2FC among all cell types. Answer format: a single cell‑type label (shape: label).\n",
      "Answer: hepatocyte\n",
      "\n",
      "Group 5\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper2/data/dataset2.h5ad\n",
      "Description: Using the provided single-nucleus RNA‑seq dataset (dataset2.h5ad), compute for each Seurat cluster (column 'seurat_clusters') the proportion of cells belonging to each treatment condition (column 'Treatment'). Then calculate, for each cluster‑treatment pair, an enrichment score defined as (cluster proportion) / (global proportion of that treatment across all cells). Identify the cluster whose maximum enrichment score (i.e., the most over‑represented treatment in that cluster) is the highest among all clusters. Report the identifier of that cluster exactly as it appears in the 'seurat_clusters' column. Answer format: a single label (shape: label).\n",
      "Answer: 5\n",
      "\n",
      "Task 1\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper2/data/dataset1.h5ad\n",
      "Description: Among the unsupervised Seurat clusters (column 'seurat_clusters') in the provided single-nucleus RNA‑seq dataset, identify the cluster that exhibits the highest proportion of cells annotated with the clinical response \"Complete\" (consider only cells with a non‑missing Response value). Answer format: a single integer label representing the cluster ID (shape: label).\n",
      "Answer: 5\n",
      "\n",
      "Group 4\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper2/data/dataset3.h5ad\n",
      "Description: Using the provided snRNA‑seq dataset of macrophages, perform a differential expression analysis between post‑chemotherapy (Stage_Code = PTX) and pre‑chemotherapy (Stage_Code = DX) cells **within each Seurat cluster** (seurat_clusters). For each cluster, count the number of genes that are significantly up‑regulated in PTX compared to DX, defined as adjusted p‑value < 0.05 and log2 fold‑change > 0.5 (t‑test). Identify the cluster with the **largest** such count. Answer format: a single integer label corresponding to the Seurat cluster identifier (shape: label).\n",
      "Answer: 4\n",
      "\n",
      "Task 1\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper2/data/dataset2.h5ad\n",
      "Description: Identify the seurat_clusters label (numeric cluster identifier) that exhibits the highest proportion of MYCN-amplified cells (MYCN_Status == 'AMP') among all clusters. Answer format: a single integer label (shape: label).\n",
      "Answer: 4\n",
      "\n",
      "Task 2\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper2/data/dataset2.h5ad\n",
      "Description: Identify the seurat_cluster (label) that exhibits the highest average expression of the MYCN oncogene across all cells in the dataset. Use the 'seurat_clusters' annotation and the expression matrix. Answer format: a single cluster label as a string (e.g., '4'), shape: label.\n",
      "Answer: 4\n",
      "\n",
      "Group control\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper4/data/my_dataset.h5ad\n",
      "Description: Among the experimental conditions in the dataset, which condition exhibits the highest proportion of cells annotated as \"cycling macrophage\"? Answer format: a single label (shape: label).\n",
      "Answer: Control\n",
      "\n",
      "Task 1\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper4/data/my_dataset.h5ad\n",
      "Description: Using the provided single-cell dataset, consider only cells annotated as \"cycling macrophage\" (cell_type == \"cycling macrophage\"). For each disease condition in the \"Condition\" column, compute the proportion of cycling macrophage cells among all cells of that condition. Report the condition that has the highest proportion. Answer format: a single condition label (shape: label), exactly as it appears in the \"Condition\" column.\n",
      "Answer: Control\n",
      "\n",
      "Task 2\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper4/data/lymph_dataset.h5ad\n",
      "Description: Among cells annotated as cytotoxic T cell, which Condition shows the highest average expression of the gene CXCL9? Answer format: a single condition name exactly as it appears in the Condition column (shape: label).\n",
      "Answer: Control\n",
      "\n",
      "Task 3\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper4/data/my_dataset.h5ad\n",
      "Description: Identify the condition (as listed in the 'Condition' column) that has the highest proportion of cells annotated as \"cycling macrophage\". Answer format: a single label exactly matching one of the Condition values (shape: label).\n",
      "Answer: Control\n",
      "\n",
      "Task 4\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper4/data/lymph_dataset.h5ad\n",
      "Description: Identify the disease condition that exhibits the highest proportion of the cell type whose proportion across conditions shows the greatest variance. Answer format: a single label exactly matching the condition name as it appears in the 'Condition' column (shape: label).\n",
      "Answer: Control\n",
      "\n",
      "Task 5\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper4/data/my_dataset.h5ad\n",
      "Description: Identify the condition (as listed in the `Condition` column) that has the highest proportion of cells annotated as \"cycling macrophage\". Provide the condition name exactly as it appears (case‑sensitive). Answer format: a single label (shape: label).\n",
      "Answer: Control\n",
      "\n",
      "Task 6\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper4/data/lymph_dataset.h5ad\n",
      "Description: Within the B cell population, identify which experimental condition shows the highest average expression of the cytokine gene CXCL10. Provide the condition name exactly as it appears in the 'Condition' column (case‑sensitive). Answer format: a single label (shape: label).\n",
      "Answer: Control\n",
      "\n",
      "Task 7\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper4/data/my_dataset.h5ad\n",
      "Description: Determine which disease condition exhibits the highest proportion of cells annotated as \"cycling macrophage\" among all cells in the dataset. Report the condition name exactly as it appears in the \"Condition\" column (case-sensitive). Answer format: a single condition label (shape: label).\n",
      "Answer: Control\n",
      "\n",
      "Task 8\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper4/data/my_dataset.h5ad\n",
      "Description: Using the provided AnnData object, calculate for each Condition the proportion of cells annotated as \"cycling macrophage\" (i.e., number of cycling macrophage cells divided by the total number of cells in that Condition). Identify the Condition with the highest proportion. Answer format: a single condition label exactly as it appears in the 'Condition' column (shape: label).\n",
      "Answer: Control\n",
      "\n",
      "Group 3427\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper5/data/VA_adipocytes.h5ad\n",
      "Description: Identify the donor (donor_id) that exhibits the highest average expression of the adipocyte marker gene ADIPOQ across all its cells in the provided dataset. Answer format: a single donor identifier (shape: label).\n",
      "Answer: 3427\n",
      "\n",
      "Task 1\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper5/data/VA_adipocytes.h5ad\n",
      "Description: Using the provided UMAP embedding (X_umap) for each cell, compute the variance of the UMAP1 and UMAP2 coordinates separately for each donor (column 'donor_id' in .obs). Then calculate the average of these two variances for each donor. Identify the donor with the highest average variance (i.e., the most dispersed cells in the UMAP space). Answer format: a single donor identifier as a label (shape: label).\n",
      "Answer: 3427\n",
      "\n",
      "Group xist\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/nature/paper2/processed_MuSC_data.h5ad\n",
      "Description: Within the ICAM1+MuSCs sub-population of the provided mouse muscle stem cell dataset, compare cells from the oldest age group (70-75) versus the youngest age group (15-20). Perform a Wilcoxon rank‑sum differential expression test on the expression matrix stored in the AnnData object, compute log2 fold‑changes, and adjust p‑values using the Benjamini‑Hochberg method. Among genes with adjusted p‑value < 0.05, identify the gene with the greatest absolute log2 fold‑change. Report the gene symbol as a single uppercase HGNC gene symbol (shape: gene_symbol).\n",
      "Answer: XIST\n",
      "\n",
      "Task 1\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/nature/paper2/processed_human_muscle_data.h5ad\n",
      "Description: Using the provided single‑cell RNA‑seq dataset, restrict the analysis to cells annotated as MuSC (muscle stem cells). Define two age groups: \"young\" = Age_group \"15-20\" and \"old\" = Age_group \"70-75\". Perform a differential expression test between old and young MuSC cells using Scanpy's rank_genes_groups with the Wilcoxon method, using the young group as reference and Benjamini‑Hochberg correction for multiple testing. Consider only genes with adjusted p‑value < 0.05. Identify the gene with the greatest absolute log2 fold‑change (old vs young) among these significant genes. Answer format: a single HGNC gene symbol in uppercase (shape: gene_symbol).\n",
      "Answer: XIST\n",
      "\n",
      "Task 2\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/cell/paper2/data.h5ad\n",
      "Description: In the provided placenta infection scRNA‑seq dataset, identify the cell type with the greatest number of cells. Within this cell type, compare cells infected with Listeria monocytogenes (infection label \"Lm\") to uninfected cells (infection label \"UI\"). Using the log‑normalized expression matrix, compute the mean expression of each gene in the Lm and UI groups, calculate the log2 fold change (mean_Lm – mean_UI), and report the HGNC gene symbol of the gene with the highest positive log2 fold change. Answer format: a single gene symbol (shape: gene_symbol).\n",
      "Answer: XIST\n",
      "\n",
      "Task 3\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/nature/paper2/processed_MuSC_data.h5ad\n",
      "Description: Within the dataset, select cells annotated as \"MuSC\" (annotation_level2 == \"MuSC\"). From these, keep only the two extreme age groups: 15-20 (young) and 70-75 (old). Normalize each cell to 10,000 counts, apply log1p transformation, and then perform differential expression using Scanpy's rank_genes_groups with groupby='Age_group', groups=['70-75'], reference='15-20', method='t-test_overestim_var'. Identify the gene with the largest positive log2 fold‑change (old vs young). Provide the gene symbol as a single uppercase string (shape: gene_symbol).\n",
      "Answer: XIST\n",
      "\n",
      "Task 4\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/nature/paper2/processed_human_muscle_data.h5ad\n",
      "Description: Using the provided AnnData file, subset the cells annotated as 'MF-I' (annotation_level0). Within this subset, compare the oldest age group (Age_group = '70-75') to the youngest age group (Age_group = '15-20') and perform differential expression analysis (Wilcoxon test, default Scanpy settings). Report the HGNC gene symbol of the gene with the largest positive log₂ fold‑change that has an adjusted p‑value < 0.05. Answer format: a single gene symbol (shape: gene_symbol, uppercase).\n",
      "Answer: XIST\n",
      "\n",
      "Group b-lymphocytes\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/nature/paper6/data.h5ad\n",
      "Description: Among the annotated cell types (column 'cluster.names'), which cell type exhibits the greatest fold change in its proportion of cells between old and young mouse ovaries (OLD divided by YOUNG), calculated from raw cell counts? Provide the cell type name exactly as it appears in the 'cluster.names' column. Answer format: a single label (shape: label).\n",
      "Answer: B-Lymphocytes\n",
      "\n",
      "Task 1\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/nature/paper6/data.h5ad\n",
      "Description: Using the provided single‑cell atlas of the aging mouse ovary, compute the proportion of each cell type (column 'cluster.names') within the two age groups labeled 'YOUNG' and 'OLD' in the 'age' column. Calculate the fold‑change of each cell type's proportion as (OLD proportion) / (YOUNG proportion). Identify the cell type with the largest fold‑change greater than 1 (i.e., the greatest relative increase from young to old). Answer format: a single cell‑type label (shape: label).\n",
      "Answer: B-Lymphocytes\n",
      "\n",
      "Task 2\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/nature/paper6/data.h5ad\n",
      "Description: Using the provided single‑cell atlas of the aging mouse ovary, compute for each cell cluster (column 'cluster.names') the proportion of cells from the OLD and YOUNG age groups (column 'age'), then calculate the enrichment ratio (OLD proportion divided by YOUNG proportion). Report the cluster name with the highest enrichment ratio. Answer format: a single cluster label (shape: label, case‑sensitive, exactly as it appears in 'cluster.names').\n",
      "Answer: B-Lymphocytes\n",
      "\n",
      "Group alzheimer disease\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/cell/paper1/data.h5ad\n",
      "Description: From the provided single-nucleus RNA‑seq dataset, compute the proportion of microglial cells within each disease category (Alzheimer disease, Pick disease, progressive supranuclear palsy, normal). Identify the disease that exhibits the highest microglial cell proportion. Answer format: a single disease name as a label (exact string as appears in the 'disease' column).\n",
      "Answer: Alzheimer disease\n",
      "\n",
      "Task 1\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/cell/paper1/data.h5ad\n",
      "Description: Using the provided single-nucleus RNA‑seq dataset, calculate for each disease category (as listed in the 'disease' column) the proportion of cells annotated as \"microglial cell\" (based on the 'cell_type' column). Identify the disease with the highest microglial cell proportion. Answer format: a single disease name exactly as it appears in the 'disease' column (shape: label).\n",
      "Answer: Alzheimer disease\n",
      "\n",
      "Group interferon-response\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/cell/paper3/data.h5ad\n",
      "Description: Using the provided scRNA‑seq dataset, compute the proportion of cells belonging to each transcriptionally defined cluster (column 'cluster_name') within each experimental condition ('Control' and 'Deprived'). Calculate the log2 fold‑change of the Deprived proportion over the Control proportion for every cluster (add a tiny epsilon = 1e‑6 to avoid division by zero). Report the exact cluster name that exhibits the highest positive log2 fold‑change (i.e., the greatest enrichment in the Deprived condition). Answer format: a single label (case‑preserved cluster name).\n",
      "Answer: Interferon-response\n",
      "\n",
      "Task 1\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/cell/paper3/data.h5ad\n",
      "Description: Determine which cell cluster (as labeled in the 'cluster_name' column) exhibits the greatest absolute change in its proportion of cells between the 'Control' and 'Deprived' experimental conditions. Report the cluster name exactly as it appears (case‑sensitive, including hyphens and spaces). Answer format: a single label (shape: label).\n",
      "Answer: Interferon-response\n",
      "\n",
      "Group csh1\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/cell/paper5/data.h5ad\n",
      "Description: Within the single-cell RNA‑seq dataset of human trophoblast organoids, consider only cells annotated as \"extravillous trophoblast\". Perform a differential expression analysis comparing the \"cytokines\" treatment group against the \"no_cytokines\" control group (using standard library‑size normalization, log‑transformation, and the Wilcoxon rank‑sum test on all genes). Identify the gene that shows the largest positive log2‑fold‑change (i.e., strongest up‑regulation) in the cytokine‑treated cells relative to control. Answer format: a single HGNC gene symbol (shape: gene_symbol).\n",
      "Answer: CSH1\n",
      "\n",
      "Task 1\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/cell/paper5/data.h5ad\n",
      "Description: Within the extravillous trophoblast cell type, identify the gene that shows the highest positive log2 fold-change when comparing cells treated with cytokines versus no cytokines. Answer format: a single HGNC gene symbol (shape: gene_symbol).\n",
      "Answer: CSH1\n",
      "\n",
      "Group anbl12p1-like chemotherapy\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper2/data/dataset2.h5ad\n",
      "Description: Within the snRNA‑seq dataset, cells are assigned to six Seurat clusters (0‑5). Considering only the cells in cluster 0, compute the average expression of the gene MYCN for each distinct value in the 'Treatment' metadata column. Identify the treatment that yields the highest mean MYCN expression. Answer format: a single label matching exactly the treatment string as it appears in adata.obs['Treatment'] (shape: label).\n",
      "Answer: ANBL12P1-like chemotherapy\n",
      "\n",
      "Task 1\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper2/data/dataset2.h5ad\n",
      "Description: Using the provided snRNA‑seq dataset of neoplastic cells, compute the average expression of the oncogene MYCN (gene symbol) in diagnostic (Stage_Code == \"DX\") cells and in each post‑treatment (Stage_Code == \"PTX\") regimen. For each post‑treatment regimen, calculate the log2‑fold‑change of the mean MYCN expression relative to the diagnostic mean (add a small epsilon = 1e‑6 to avoid division by zero). Identify the post‑treatment regimen with the largest positive log2‑fold‑change. Report the exact treatment name as it appears in the 'Treatment' column. Answer format: a single label (shape: label).\n",
      "Answer: ANBL12P1-like chemotherapy\n",
      "\n",
      "Task 2\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper2/data/dataset2.h5ad\n",
      "Description: Among the malignant neuroblast cells (cell_type == 'neuroblast (sensu Vertebrata)' and malignancy == 'Malignant'), compute the average expression of the oncogene MYCN for each Treatment group. Identify the Treatment label with the highest average MYCN expression. Answer format: a single treatment label string exactly as it appears in the data (shape: label).\n",
      "Answer: ANBL12P1-like chemotherapy\n",
      "\n",
      "Task 3\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper2/data/dataset2.h5ad\n",
      "Description: Identify the treatment condition (as listed in the 'Treatment' column of the provided AnnData object) that exhibits the highest average normalized expression of the gene MYCN across all cells. Provide the answer as a single exact label string (shape: label), matching the Treatment entry case‑sensitively and without extra spaces.\n",
      "Answer: ANBL12P1-like chemotherapy\n",
      "\n",
      "Task 4\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper2/data/dataset2.h5ad\n",
      "Description: Within the neuroblastoma cells (cell_type == 'neuroblast (sensu Vertebrata)') in the provided dataset, compute the average raw count expression of the oncogene MYCN for each distinct Treatment group. Return the Treatment name that yields the highest average MYCN expression. Answer format: a single label exactly matching the Treatment string (shape: label).\n",
      "Answer: ANBL12P1-like chemotherapy\n",
      "\n",
      "Group mt-co3\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper2/data/dataset3.h5ad\n",
      "Description: Within the most abundant seurat cluster of macrophage nuclei, perform differential expression between the post‑treatment (PTX) and diagnosis (DX) groups. Identify the gene that is most significantly up‑regulated in PTX versus DX (i.e., the up‑regulated gene with the smallest Benjamini‑Hochberg adjusted p‑value). Answer format: a single HGNC gene symbol (shape: gene_symbol).\n",
      "Answer: MT-CO3\n",
      "\n",
      "Task 1\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/nature/paper6/data.h5ad\n",
      "Description: In the provided single‑cell atlas of the aging mouse ovary, restrict the analysis to cells annotated as \"stromal cell\". Using the log‑normalized expression matrix, perform a differential expression comparison between cells from the OLD age group versus the YOUNG age group (Wilcoxon rank‑sum test, all genes). Identify the gene that exhibits the greatest absolute log2 fold‑change (|log2FC|) between the two age groups. Report the official mouse gene symbol of this gene. Answer format: a single gene symbol (shape: gene_symbol, uppercase).\n",
      "Answer: MT-CO3\n",
      "\n",
      "Task 2\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper2/data/dataset1.h5ad\n",
      "Description: Within the malignant cells of the dataset, identify the most abundant Seurat cluster (by cell count). In that dominant cluster, compare cells from the pre‑treatment stage (Stage_Code = \"DX\") to post‑treatment stage (Stage_Code = \"PTX\"). Using a Wilcoxon rank‑sum test (as implemented by Scanpy's rank_genes_groups) and considering genes with adjusted p‑value < 0.05, report the gene that shows the greatest positive log2 fold‑change (i.e., strongest up‑regulation) after chemotherapy. Answer format: a single HGNC gene symbol in uppercase (shape: gene_symbol).\n",
      "Answer: MT-CO3\n",
      "\n",
      "Group 464c\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/nature/paper2/processed_MuSC_data.h5ad\n",
      "Description: Among the donors in the dataset, which donor contributes the highest proportion of cells labeled as 'MYOG+MuSCs' (relative to its total cells)? Answer format: a single donor identifier string (shape: label).\n",
      "Answer: 464C\n",
      "\n",
      "Task 1\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/nature/paper2/processed_MuSC_data.h5ad\n",
      "Description: Identify the donor (DonorID) that exhibits the highest fraction of cells annotated as 'MYOG+MuSCs' among all cells from that donor. Report the DonorID exactly as it appears (e.g., '464C'). Answer format: a single label (shape: label).\n",
      "Answer: 464C\n",
      "\n",
      "Group nuclei\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/nature/paper2/processed_human_muscle_data.h5ad\n",
      "Description: Using the provided AnnData object, identify the cell type with the highest number of cells in the 'annotation_level1' column. Subset the data to this cell type, normalize, log‑transform, select 2,000 highly variable genes, scale, and compute 30 principal components. Build a k‑nearest‑neighbour graph with k = 30 on the PCA space and compute the inverse Simpson index (iLISI) of the 'batch' labels among each cell's neighbours. Report the batch (value from the 'batch' column) that has the lowest mean iLISI across all cells of this cell type. Answer format: a single label (shape: label).\n",
      "Answer: nuclei\n",
      "\n",
      "Task 1\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/nature/paper2/processed_myonucleicells_data.h5ad\n",
      "Description: Using the scANVI embedding (X_scANVI) provided in the AnnData object, compute the 30‑nearest‑neighbor graph (Euclidean distance) for all cells. For each cell, calculate the fraction of its 30 neighbors that belong to the same batch (the 'batch' column in .obs). Then average this fraction across all cells within each batch. Report the batch that has the highest average same‑batch neighbor fraction (i.e., the batch showing the poorest mixing). Answer format: a single lowercase label (shape: label).\n",
      "Answer: nuclei\n",
      "\n",
      "Group mf-isn(fg)\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/nature/paper2/processed_myonuclei_data.h5ad\n",
      "Description: Identify the annotation_level1 cell type that exhibits the highest average expression of the gene MYH7 across all nuclei in the dataset. Answer format: a single cell type label exactly as it appears in the 'annotation_level1' column (shape: label).\n",
      "Answer: MF-Isn(fg)\n",
      "\n",
      "Task 1\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/nature/paper2/processed_myonuclei_data.h5ad\n",
      "Description: Using the provided muscle nuclei dataset, compute for each fine-grained cell type (annotation_level1) the proportion of cells contributed by each donor (DonorID). Then calculate the coefficient of variation (CV = standard deviation / mean) of these donor-wise proportions across all donors. Report the cell type (annotation_level1 label) that exhibits the highest CV. Answer format: a single label (shape: label).\n",
      "Answer: MF-Isn(fg)\n",
      "\n",
      "Task 2\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/nature/paper2/processed_myonucleicells_data.h5ad\n",
      "Description: After removing cells with scrublet_score > 0.2, compute the average percent_mito for each cell type in the annotation_level0 column. Report the cell type (exact string as it appears in annotation_level0) that has the highest average percent_mito. Answer format: a single label (shape: label).\n",
      "Answer: MF-Isn(fg)\n",
      "\n",
      "Task 3\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/nature/paper2/processed_myonucleicells_data.h5ad\n",
      "Description: After removing cells with a Scrublet doublet score greater than 0.1, compute the mean percent_mito for each cell type defined in the column 'annotation_level1'. Which cell type has the highest average percent_mito? Answer format: a single label exactly as it appears in 'annotation_level1' (shape: label).\n",
      "Answer: MF-Isn(fg)\n",
      "\n",
      "Task 4\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/nature/paper2/processed_myonucleicells_data.h5ad\n",
      "Description: Using the provided AnnData file (/mlbio_scratch/panigrah/agents/data/nature/paper2/processed_myonucleicells_data.h5ad), compute the average raw counts expression of the gene MYH1 for each cell type defined in the 'annotation_level2' column of adata.obs. Identify the cell type with the highest mean MYH1 expression. Answer format: a single label that matches exactly one of the annotation_level2 categories (shape: label).\n",
      "Answer: MF-IIsn(fg)\n",
      "\n",
      "Group 9\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/nature/paper5/mouse_data.h5ad\n",
      "Description: Among the 30 intestinal segments (labeled 1 to 30) in the provided mouse epithelial single‑cell dataset, identify the segment that has the highest proportion of cells annotated as \"stem cell\" (i.e., #stem cells / total cells in the segment). Answer format: a single segment label (numeric string) with no spaces (shape: label).\n",
      "Answer: 9\n",
      "\n",
      "Task 1\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/nature/paper5/mouse_data.h5ad\n",
      "Description: Using the provided single-cell dataset of mouse intestinal epithelial cells, calculate for each intestinal segment (1 to 30) the proportion of cells annotated as \"Stem cell\" (i.e., number of Stem cell entries divided by total cells in that segment). Identify the segment that has the highest Stem cell proportion. Answer format: a single integer representing the segment number (shape: number).\n",
      "Answer: 9\n",
      "\n",
      "Task 2\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/nature/paper5/mouse_data.h5ad\n",
      "Description: Identify the intestinal segment (1-30) that has the highest proportion of Stem cells among all cells in the dataset. Answer format: a single integer representing the segment number (shape: number).\n",
      "Answer: 9\n",
      "\n",
      "Task 3\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/nature/paper5/mouse_data.h5ad\n",
      "Description: Using the provided single-cell RNA‑seq dataset of mouse intestinal epithelial cells, compute for each of the 30 intestinal segments the proportion of cells annotated as \"Stem cell\" (i.e., number of Stem cells divided by total cells in that segment). Identify the segment that has the highest Stem‑cell proportion. Answer format: a single segment identifier as a label (shape: label).\n",
      "Answer: 9\n",
      "\n",
      "Task 4\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper5/data/VA_adipocytes.h5ad\n",
      "Description: From the provided single-nucleus RNA‑seq dataset of human visceral adipose tissue, identify the Seurat cluster (column 'seurat_clusters' in .obs) that exhibits the highest median mitochondrial gene expression percentage (column 'percent.mt'). Report the cluster identifier exactly as it appears (shape: label).\n",
      "Answer: 9\n",
      "\n",
      "Task 5\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/nature/paper5/mouse_data.h5ad\n",
      "Description: Determine which intestinal segment (1‑30) has the highest proportion of Stem cells among all cells in the dataset. Report the segment number as an integer (shape: number).\n",
      "Answer: 9\n",
      "\n",
      "Group b cell\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/nature/paper6/data.h5ad\n",
      "Description: From the provided single‑cell atlas of the aging mouse ovary, determine which annotated cell type shows the greatest enrichment in the OLD age group relative to the YOUNG age group. Compute the proportion of cells of each type within each age, calculate the fold‑change (OLD proportion ÷ YOUNG proportion), and report the cell‑type label (exact string as it appears in the `cell_type` column) that has the highest fold‑change. Answer format: a single label (shape: label).\n",
      "Answer: B cell\n",
      "\n",
      "Task 1\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/nature/paper6/data.h5ad\n",
      "Description: Identify the cell type that exhibits the greatest absolute log2 fold-change in proportion between the OLD and YOUNG age groups. For each cell type, compute its proportion within each age (count of cells of that type divided by total cells of that age), calculate log2(proportion_OLD / proportion_YOUNG), take the absolute value, and report the cell type with the highest absolute value. Answer format: a single cell type label exactly as it appears in the `cell_type` column (shape: label).\n",
      "Answer: B cell\n",
      "\n",
      "Task 2\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper5/data/dataset2.h5ad\n",
      "Description: Using the provided snRNA‑seq AnnData object, compute a neighbourhood graph (e.g., Scanpy's default parameters: n_neighbors=15, n_pcs=30 after log‑normalization and HVG selection). For each cell, calculate an integration LISI (iLISI) score based on the 'donor_id' batch variable by measuring the Shannon entropy of donor labels within each cell's neighbourhood and converting it to exp(entropy). Then, average the iLISI scores for each cell type given in the 'cell_type' column. Identify the cell type with the lowest average iLISI (i.e., the most batch‑biased cell type). Answer format: a single cell‑type label (shape: label).\n",
      "Answer: B cell\n",
      "\n",
      "Task 3\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/nature/paper6/data.h5ad\n",
      "Description: Based on the provided single-cell atlas of the aging mouse ovary, compute the proportion of each cell type within the two age groups (YOUNG and OLD). Then calculate the log2 fold-change of these proportions (OLD vs YOUNG). Identify the cell type that shows the greatest positive log2 fold-change (i.e., the strongest increase in relative abundance in OLD compared to YOUNG). Answer format: a single cell type label exactly as it appears in the 'cell_type' column (shape: label).\n",
      "Answer: B cell\n",
      "\n",
      "Group 13\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper2/data/dataset3.h5ad\n",
      "Description: Using the provided snRNA‑seq dataset of macrophages, compute the proportion of cells belonging to each 'global_cluster' within the diagnostic stage (Stage_Code = 'DX') and the post‑treatment stage (Stage_Code = 'PTX'). Calculate the fold‑change as (PTX proportion + 1e‑6) / (DX proportion + 1e‑6). Identify the cluster with the highest fold‑change (i.e., the greatest increase in relative abundance after treatment). Answer format: a single label string exactly matching the 'global_cluster' identifier (shape: label).\n",
      "Answer: 13\n",
      "\n",
      "Task 1\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper5/data/VA_adipocytes.h5ad\n",
      "Description: Using the AnnData file at /mlbio_scratch/videnovi/papers_BIBA/paper5/data/VA_adipocytes.h5ad, examine the observation columns `seurat_clusters` (integer cluster IDs) and `donor_id`. For each cluster, count how many distinct donors are present. Identify the cluster(s) with the fewest distinct donors. If multiple clusters tie for this minimum, choose the one whose most frequent donor constitutes the highest proportion of cells in that cluster. Provide the chosen cluster label as a single integer (shape: label).\n",
      "Answer: 13\n",
      "\n",
      "Group endothelial_cell\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper5/data/dataset2.h5ad\n",
      "Description: Identify the annotated cell type that exhibits the highest average mitochondrial gene expression percentage (percent.mt) across all cells. Report the answer as a single lowercase label with words separated by underscores (shape: label).\n",
      "Answer: endothelial_cell\n",
      "\n",
      "Task 1\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper5/data/dataset2.h5ad\n",
      "Description: Among all cell types annotated in the dataset (column 'cell_type'), which cell type exhibits the highest average expression of the mitochondrial gene MT-ND1? Answer format: a single lowercase label exactly as it appears in the 'cell_type' column (shape: label).\n",
      "Answer: endothelial cell\n",
      "\n",
      "Task 2\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper5/data/dataset2.h5ad\n",
      "Description: Identify the cell type with the highest average mitochondrial RNA percentage (percent.mt) across all nuclei in the dataset. Answer format: a single lowercase label with underscores (shape: label).\n",
      "Answer: endothelial_cell\n",
      "\n",
      "Task 3\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper5/data/dataset2.h5ad\n",
      "Description: Using the provided AnnData object, compute the average mitochondrial transcript percentage (percent.mt) for each cell type listed in the 'cell_type' column of adata.obs. Report the cell type that has the highest mean percent.mt. Answer format: a single cell type label exactly as it appears in the 'cell_type' column (shape: label).\n",
      "Answer: endothelial cell\n",
      "\n",
      "Group adipocyte of omentum tissue\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper5/data/VA_adipocytes.h5ad\n",
      "Description: Using the AnnData file /mlbio_scratch/videnovi/papers_BIBA/paper5/data/VA_adipocytes.h5ad, perform the following analysis: (1) filter out cells with <500 total counts and genes expressed in <10 cells; (2) normalize each cell to 10,000 counts and log1p-transform; (3) select the 2,000 most highly variable genes (Seurat v3 method); (4) compute 30 principal components and build a 30‑nearest‑neighbor graph (using these PCs); (5) for each cell, count how many distinct donor IDs (column 'donor_id') are present among its 30 nearest neighbors; (6) for each annotated cell type (column 'cell_type'), calculate the average of these unique‑donor counts; (7) identify the cell type with the lowest average unique‑donor neighbor count (i.e., poorest donor mixing). Answer format: a single label string exactly matching the entry in the 'cell_type' column (case‑sensitive).\n",
      "Answer: adipocyte of omentum tissue\n",
      "\n",
      "Task 1\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper5/data/dataset2.h5ad\n",
      "Description: Using the provided snRNA‑seq dataset of human visceral adipose tissue, compute the average log‑normalized expression of the mature adipocyte marker genes ADIPOQ, PLIN1, FABP4 and LPL for each broad cell type annotation (the 'cell_type' column in .obs). Identify the cell type with the highest mean expression across these four markers. Return the cell type label exactly as it appears in the 'cell_type' column. Answer format: a single label (shape: label).\n",
      "Answer: adipocyte of omentum tissue\n",
      "\n",
      "Task 2\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper5/data/dataset2.h5ad\n",
      "Description: Identify the cell type that shows the highest average raw count expression of the adipogenic master regulator gene PPARG across all cells in the dataset. Report the answer as the exact cell type label found in the 'cell_type' column (case‑sensitive, including spaces). Answer format: a single label (shape: label).\n",
      "Answer: adipocyte of omentum tissue\n",
      "\n",
      "Task 3\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper5/data/dataset2.h5ad\n",
      "Description: Using the provided snRNA-seq dataset of human visceral adipose tissue, compute the average log‑CP10K expression of the gene PPARG within each annotated cell type (the 'cell_type' column). Normalize raw counts per cell to counts per 10 k, apply log1p, then calculate the mean of this transformed expression for each cell type. Return the cell type with the highest average expression. Answer format: a single cell‑type label exactly as it appears in adata.obs['cell_type'] (shape: label).\n",
      "Answer: adipocyte of omentum tissue\n",
      "\n",
      "Group grin2a\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper6/data/dataset1.h5ad\n",
      "Description: In the provided snMultiome dataset, restrict to cells where the 'Class' annotation equals 'Neuron'. For each gene, compute the Spearman correlation between its expression values (across these neuronal cells) and the numeric variable 'Estimated_postconceptional_age_in_days'. Identify the gene with the highest positive correlation (i.e., the strongest monotonic increase with age). Return the gene symbol as a single uppercase HGNC gene symbol (shape: gene_symbol).\n",
      "Answer: GRIN2A\n",
      "\n",
      "Task 1\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper6/data/dataset1.h5ad\n",
      "Description: Identify the neuronal gene whose expression has the strongest positive Spearman correlation with the estimated post‑conceptional age (in days) across all cells labeled as Class = 'Neuron', after applying Benjamini‑Hochberg FDR correction (α = 0.05). Provide the answer as a single HGNC gene symbol (shape: gene_symbol).\n",
      "Answer: GRIN2A\n",
      "\n",
      "Group caecum\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/nature/paper1/cxg_data.h5ad\n",
      "Description: Within the effector regulatory T‑cell subset (annotation = \"eTregs\"), compute the average expression of the gene Cd69 for each tissue (mesenteric lymph node, caecum, small intestine Peyer's patch, lamina propria of small intestine). Return the tissue name that shows the highest mean Cd69 expression. Answer format: a single tissue label (shape: label).\n",
      "Answer: caecum\n",
      "\n",
      "Task 1\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/nature/paper1/cxg_data.h5ad\n",
      "Description: Among the four tissues represented in the dataset (mesenteric lymph node, caecum, lamina propria of small intestine, small intestine Peyer's patch), which tissue has the highest proportion of cells annotated as Naive.CD4+_CP? Answer format: a single lowercase tissue name (shape: label).\n",
      "Answer: caecum\n",
      "\n",
      "Group 6720483e21rik\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/cell/paper3/data.h5ad\n",
      "Description: Using the provided scRNA‑seq AnnData object (data.h5ad), perform a log‑normalization (log1p) of the raw counts, then compute the average expression of each gene across cells in the two experimental conditions (Control vs Deprived). Calculate the log2 fold‑change (log2FC) for each gene as log2((mean_Deprived + 1e-6) / (mean_Control + 1e-6)). Identify the gene with the largest positive log2FC (i.e., most up‑regulated in the Deprived condition). Answer format: a single mouse gene symbol (shape: gene_symbol).\n",
      "Answer: 6720483E21Rik\n",
      "\n",
      "Task 1\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/cell/paper3/data.h5ad\n",
      "Description: Using the provided single‑cell RNA‑seq dataset, perform a differential expression analysis comparing the 'Deprived' condition against the 'Control' condition across all cells. Identify the gene that shows the strongest up‑regulation in the Deprived condition (i.e., the highest positive log2 fold‑change). Report the gene symbol exactly as it appears in the dataset (shape: gene_symbol).\n",
      "Answer: 6720483E21Rik\n",
      "\n",
      "Task 2\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/cell/paper3/data.h5ad\n",
      "Description: Using the provided scRNA‑seq dataset of mouse barrel cortex microglia, normalize the raw counts to 10,000 per cell and apply log1p transformation. Then perform differential expression between the 'Deprived' and 'Control' conditions (Wilcoxon test, grouping by the 'condition' column). Report the official gene symbol of the gene with the highest positive log2 fold‑change (i.e., the most up‑regulated gene in the Deprived condition). Answer format: a single gene symbol (shape: gene_symbol).\n",
      "Answer: 6720483E21Rik\n",
      "\n",
      "Group evt_late_3\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/cell/paper5/data.h5ad\n",
      "Description: Using the provided single‑cell RNA‑seq AnnData object, calculate the proportion of each cell type (as given in the 'author_cell_type' column) within each treatment group ('cytokines' and 'no_cytokines'). Determine which cell type exhibits the largest relative increase in proportion when comparing cytokine‑treated to untreated samples (i.e., the highest ratio of proportion_cytokines / proportion_no_cytokines). Provide the cell type name exactly as it appears in the data. Answer format: a single label (shape: label).\n",
      "Answer: EVT_late_3\n",
      "\n",
      "Task 1\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/cell/paper5/data.h5ad\n",
      "Description: Using the provided single-cell RNA‑seq dataset, compare the two experimental conditions in the 'treatment' column (\"cytokines\" vs \"no_cytokines\"). For each cell type listed in the 'author_cell_type' column, compute the proportion of cells belonging to that type within each condition, then calculate the log2 fold‑change of the proportion (cytokines / no_cytokines). Identify the cell type with the largest positive log2 fold‑change (i.e., the greatest relative increase after cytokine treatment). Answer format: a single label exactly as it appears in the 'author_cell_type' column (shape: label).\n",
      "Answer: EVT_late_3\n",
      "\n",
      "Task 2\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/cell/paper5/data.h5ad\n",
      "Description: Identify the cell type (value from the 'author_cell_type' column) whose relative proportion among all cells is most increased in the cytokine‑treated condition compared to the no‑cytokine control. Answer format: a single label exactly as it appears in the 'author_cell_type' column (shape: label).\n",
      "Answer: EVT_late_3\n",
      "\n",
      "Task 3\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/cell/paper5/data.h5ad\n",
      "Description: Using the provided single-cell RNA‑seq AnnData object, first normalize raw counts to 10 000 counts per cell (CP10K) and then apply log1p transformation. For each author‑annotated cell type (column 'author_cell_type'), compute the mean log‑expression of every gene separately in the two treatment groups ('cytokines' and 'no_cytokines'). Calculate the log2‑fold‑change for each gene as the difference of these means (treated minus control). Count, for each cell type, how many genes have log2‑fold‑change greater than 0.25. Report the cell type that has the largest such count. Answer format: a single label exactly as it appears in the 'author_cell_type' column (shape: label).\n",
      "Answer: EVT_late_3\n",
      "\n",
      "Task 4\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/cell/paper5/data.h5ad\n",
      "Description: Identify the author-provided cell type that exhibits the highest enrichment after cytokine treatment compared to the no‑cytokine control, measured as the log2 fold‑change of the proportion of cells (cytokines / no_cytokines). Answer format: a single cell‑type label exactly as it appears in the 'author_cell_type' column (shape: label).\n",
      "Answer: EVT_late_3\n",
      "\n",
      "Group cardiac_neuron\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper4/data/dataset1.h5ad\n",
      "Description: Identify the cell type that exhibits the greatest up‑regulation of the chemokine gene CXCL10 when comparing MIS‑C (Multisystem Inflammatory Syndrome in Children) to Control samples. Use the mean expression difference (MIS‑C mean minus Control mean) calculated from the provided normalized expression matrix. Answer format: a single cell‑type label in lowercase with underscores (shape: label).\n",
      "Answer: cardiac_neuron\n",
      "\n",
      "Task 1\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper4/data/dataset1.h5ad\n",
      "Description: Using the provided scRNA‑seq dataset of human cardiac tissue, compare the MIS‑C condition to the Control condition within each annotated cell type. For each cell type, count the number of genes whose log2 fold‑change (MIS‑C vs Control) exceeds 1 (i.e., at least a two‑fold increase). Identify the cell type with the greatest such count. Answer format: a single cell‑type label exactly as it appears in the 'cell_type' column (shape: label).\n",
      "Answer: cardiac neuron\n",
      "\n",
      "Group mesenteric lymph node\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/nature/paper1/cxg_data.h5ad\n",
      "Description: Determine which tissue (as listed in the 'tissue' column of the provided AnnData object) contains the highest proportion of regulatory T cells (cells with final_annotation == 'cTregs'). Report the tissue name exactly as it appears (shape: label).\n",
      "Answer: mesenteric lymph node\n",
      "\n",
      "Task 1\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/nature/paper1/cxg_data.h5ad\n",
      "Description: Calculate the proportion of cells annotated as 'cTregs' within each tissue (mesenteric lymph node, caecum, lamina propria of small intestine, small intestine Peyer's patch). Report the tissue that has the highest cTregs proportion. Answer format: the exact tissue name as it appears in the 'tissue' column (shape: label).\n",
      "Answer: mesenteric lymph node\n",
      "\n",
      "Task 2\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/nature/paper1/cxg_data.h5ad\n",
      "Description: Identify the tissue type that exhibits the highest proportion of cells annotated as \"cTregs\" (final_annotation). Report the tissue name in lowercase with words separated by underscores (shape: label).\n",
      "Answer: mesenteric_lymph_node\n",
      "\n",
      "Group tnfrsf12a+muscs\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/nature/paper2/processed_MuSC_data.h5ad\n",
      "Description: Among the MuSC sub‑states defined in the 'annotation_level2' column, identify the sub‑state that has the highest proportion of cells belonging to the 'old' age bin. Return the exact sub‑state label as it appears in the 'annotation_level2' column. Answer format: a single label (shape: label).\n",
      "Answer: TNFRSF12A+MuSCs\n",
      "\n",
      "Task 1\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/nature/paper2/processed_MuSC_data.h5ad\n",
      "Description: In the MuSC dataset, cells are annotated with subpopulation labels in the column 'annotation_level2' and grouped by the age bin column 'Age_bin' (values 'young' and 'old'). For each subpopulation, compute its proportion within each age bin (subpopulation cells / total cells of that age bin) and then calculate the enrichment ratio defined as (proportion in 'old') / (proportion in 'young'). Identify the subpopulation with the highest enrichment ratio, i.e., the one most enriched in old cells relative to young cells. Answer format: a single exact subpopulation label as it appears in 'annotation_level2' (shape: label).\n",
      "Answer: TNFRSF12A+MuSCs\n",
      "\n",
      "Task 2\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/nature/paper2/processed_MuSC_data.h5ad\n",
      "Description: Within the MuSC dataset, cells are annotated with sub‑type labels in the column 'annotation_level2' and belong to two age categories: 'young' and 'old' (as indicated by the 'Age_bin' column). Compute the proportion of each sub‑type within each age category, then calculate the log2 fold‑change of the old proportion over the young proportion (add a negligible pseudocount to avoid division by zero). Identify the sub‑type that has the highest positive log2 fold‑change (i.e., is most enriched in old compared to young). Answer format: a single label exactly matching the value in 'annotation_level2' (case‑sensitive).\n",
      "Answer: TNFRSF12A+MuSCs\n",
      "\n",
      "Group 500c\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/nature/paper2/processed_myonuclei_data.h5ad\n",
      "Description: Apply the following quality control filters to the muscle nuclei dataset: keep cells with scrublet_score < 0.2, n_counts ≥ 1000, n_genes ≥ 500, and percent_mito < 0.05. Exclude cells where myofiber_type is missing ('nan'). Then, for each donor, compute the fraction of retained cells that are annotated as MYH7+. Identify the donor with the highest MYH7+ fraction. Answer format: a single donor identifier (shape: label).\n",
      "Answer: 500C\n",
      "\n",
      "Task 1\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/nature/paper2/processed_human_muscle_data.h5ad\n",
      "Description: Identify the DonorID that has the highest fraction of cells annotated as \"MF-I\" (annotation_level2) among all donors in the dataset. Answer format: a single label exactly as it appears in the DonorID column (shape: label).\n",
      "Answer: 500C\n",
      "\n",
      "Group 8\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/cell/paper3/data.h5ad\n",
      "Description: Identify the final cluster (value from the 'final_clusters' column) that exhibits the highest relative enrichment in the Deprived condition compared to the Control condition, defined as the largest ratio of its proportion of cells in Deprived over its proportion in Control. Answer format: a single cluster label (shape: label).\n",
      "Answer: 8\n",
      "\n",
      "Task 1\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/cell/paper3/data.h5ad\n",
      "Description: From the provided single‑cell RNA‑seq dataset of mouse barrel cortex microglia, compute the proportion of cells belonging to the 'Deprived' condition within each cluster defined by the 'final_clusters' annotation. Identify the cluster label that has the highest proportion of Deprived cells. Answer format: a single cluster label (string) with no surrounding whitespace (shape: label).\n",
      "Answer: 8\n",
      "\n",
      "Task 2\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/cell/paper3/data.h5ad\n",
      "Description: Using the provided scRNA‑seq dataset, compute the enrichment of each transcriptional cluster (column 'final_clusters') in the 'Deprived' condition relative to the 'Control' condition. For each cluster, calculate the log2 enrichment ratio: log2[(fraction of Deprived cells in the cluster) / (fraction of Control cells in the cluster)]. Identify the cluster with the highest positive log2 enrichment (i.e., most over‑represented in Deprived). Answer format: a single cluster label as it appears in 'final_clusters' (shape: label).\n",
      "Answer: 8\n",
      "\n",
      "Task 3\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/cell/paper3/data.h5ad\n",
      "Description: Identify the cell cluster (final_clusters) that shows the greatest enrichment in the Deprived condition relative to Control, measured as the ratio of its proportion of cells in Deprived to its proportion in Control. Answer format: a single cluster label as it appears in the 'final_clusters' column (shape: label).\n",
      "Answer: 8\n",
      "\n",
      "Task 4\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/cell/paper3/data.h5ad\n",
      "Description: Identify the 'final_clusters' label (as a string) that exhibits the highest proportion of cells from the 'Deprived' condition among all clusters. Answer format: a single cluster label with no spaces (shape: label).\n",
      "Answer: 8\n",
      "\n",
      "Group 14\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper2/data/dataset1.h5ad\n",
      "Description: Using the provided snRNA‑seq dataset, compute for each integrated cluster (column 'integrated_snn_res.0.2') the proportion of cells labeled as 'Malignant' within the diagnosis stage (Stage_Code = DX) and the post‑chemotherapy stage (Stage_Code = PTX). Identify the cluster that shows the largest positive difference (post‑chemotherapy proportion minus diagnosis proportion). Answer format: a single cluster identifier as a string (shape: label).\n",
      "Answer: 14\n",
      "\n",
      "Task 1\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper2/data/dataset1.h5ad\n",
      "Description: Using the provided snRNA‑seq dataset, compute for each Seurat cluster (seurat_clusters) the fraction of cells annotated as 'Malignant' within the diagnosis (DX) and post‑chemotherapy (PTX) stages. Identify the cluster that shows the greatest positive increase in malignant fraction from DX to PTX (i.e., PTX fraction minus DX fraction is maximal). Answer format: a single cluster identifier as a label (e.g., \"14\").\n",
      "Answer: 14\n",
      "\n",
      "Task 2\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper2/data/dataset1.h5ad\n",
      "Description: Identify the Seurat cluster (value from the 'seurat_clusters' column) that exhibits the greatest increase in the proportion of malignant cells when comparing post‑chemotherapy (Stage_Code = PTX) to pre‑chemotherapy (Stage_Code = DX) samples. Answer format: a single label representing the cluster identifier (shape: label).\n",
      "Answer: 14\n",
      "\n",
      "Group 3\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper2/data/dataset3.h5ad\n",
      "Description: From the provided snRNA‑seq dataset of macrophages in high‑risk neuroblastoma, cells have been assigned to Seurat clusters (seurat_clusters) and annotated with treatment response categories (Response: Complete, Partial, Progressive, Stable Disease). Identify the Seurat cluster (label as it appears in the seurat_clusters column) that shows the highest enrichment for the 'Complete' response relative to its overall frequency in the dataset. Enrichment is defined as (proportion of 'Complete' cells within the cluster) divided by (overall proportion of 'Complete' cells across all cells). Answer format: a single integer label (shape: label).\n",
      "Answer: 3\n",
      "\n",
      "Task 1\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper2/data/dataset3.h5ad\n",
      "Description: From the provided snRNA‑seq dataset of macrophages, compute the proportion of cells with a 'Complete' response within each Seurat cluster (column 'seurat_clusters' in adata.obs). Identify the cluster that has the highest proportion of 'Complete' response. Answer format: a single cluster label as a string (shape: label).\n",
      "Answer: 3\n",
      "\n",
      "Task 2\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper5/data/dataset2.h5ad\n",
      "Description: Identify the Seurat cluster that has the highest number of macrophage cells among clusters that consist exclusively of macrophages (i.e., 100% macrophage purity). Answer format: a single cluster identifier as a string (shape: label).\n",
      "Answer: 3\n",
      "\n",
      "Group 20\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper4/data/dataset1.h5ad\n",
      "Description: Identify the Leiden clustering label (as a string) that exhibits the highest proportion of cells from the 'MIS-C' condition in the dataset. Answer format: a single label (shape: label).\n",
      "Answer: 20\n",
      "\n",
      "Task 1\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper4/data/dataset1.h5ad\n",
      "Description: Identify the Leiden cluster that shows the strongest enrichment of the MIS-C condition relative to the Control condition, measured as the highest positive log2 fold‑change of the proportion of MIS-C cells over Control cells. Answer format: a single cluster label as a string (shape: label).\n",
      "Answer: 20\n",
      "\n",
      "Group central memory cd4-positive, alpha-beta t cell\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper4/data/lymph_dataset.h5ad\n",
      "Description: Identify the cell type whose proportion is most increased in myocarditis samples compared to control (normal) samples, using the provided AnnData object. Compute the proportion of each cell type within each disease group, calculate the log2 fold‑change (myocarditis / normal), and select the cell type with the highest positive log2 FC. Answer format: a single label exactly matching the 'cell_type' entry (shape: label).\n",
      "Answer: central memory CD4-positive, alpha-beta T cell\n",
      "\n",
      "Task 1\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper4/data/lymph_dataset.h5ad\n",
      "Description: Identify the cell type that exhibits the highest enrichment of myocarditis compared to normal tissue, defined as the highest ratio of the proportion of myocarditis cells to normal cells within that cell type. Answer format: a single cell type label (shape: label).\n",
      "Answer: central memory CD4-positive, alpha-beta T cell\n",
      "\n",
      "Group fp236383.1\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper4/data/my_dataset.h5ad\n",
      "Description: Using the precomputed differential expression results stored in adata.uns['rank_genes_groups'], identify the gene among the top 100 DE genes that exhibits the largest positive log2‑fold‑change when comparing myocarditis vs normal samples. Answer format: a single HGNC gene symbol in uppercase (shape: gene_symbol).\n",
      "Answer: FP236383.1\n",
      "\n",
      "Task 1\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper4/data/my_dataset.h5ad\n",
      "Description: Identify the gene with the highest absolute log2 fold change in the differential expression analysis comparing Leiden_12 cluster 0 (reference) versus all other clusters. Answer format: a single HGNC gene symbol (shape: gene_symbol).\n",
      "Answer: FP236383.1\n",
      "\n",
      "Group young\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/nature/paper2/processed_MuSC_data.h5ad\n",
      "Description: Determine which Age_bin ('young' or 'old') shows the higher proportion of cells annotated as ICAM1+MuSCs among all MuSCs. Answer format: a single lowercase label (shape: label).\n",
      "Answer: young\n",
      "\n",
      "Task 1\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/nature/paper2/processed_myonucleicells_data.h5ad\n",
      "Description: Identify the cell type that exhibits the highest average expression of the gene MYH1 across the entire dataset. Within that cell type, compare the average MYH1 expression between the two age categories ('young' and 'old') as defined in the 'Age_bin' metadata. Report the age category that shows the higher mean MYH1 expression. Answer format: a single lowercase label (shape: label).\n",
      "Answer: young\n",
      "\n",
      "Group mf-iisc(fg)\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/nature/paper2/processed_human_muscle_data.h5ad\n",
      "Description: Using the provided AnnData object, compute the proportion of each cell type (annotation_level0) within the two age groups defined by the 'Age_bin' column (young and old). Identify the cell type that exhibits the greatest positive difference in proportion when comparing old versus young (i.e., proportion in old minus proportion in young is maximal). Answer format: a single label exactly as it appears in the 'annotation_level0' column (shape: label).\n",
      "Answer: MF-IIsc(fg)\n",
      "\n",
      "Task 1\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/nature/paper2/processed_myonucleicells_data.h5ad\n",
      "Description: Compute the overall proportion of nuclei (batch == 'nuclei') across all cells in the provided AnnData file. For each cell type in the 'annotation_level1' column, compute the proportion of nuclei among cells of that type. Calculate the absolute deviation of each cell type's nuclei proportion from the overall proportion. Identify the cell type(s) with the maximum deviation; if multiple cell types share this maximum, choose the one with the highest total cell count. Return the selected cell type label exactly as it appears in 'annotation_level1'. Answer format: a single label (shape: label).\n",
      "Answer: MF-IIsc(fg)\n",
      "\n",
      "Group tg\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/cell/paper2/data.h5ad\n",
      "Description: From the provided placenta infection single-cell dataset, determine which infection condition ('infection' column) has the highest relative abundance of Hofbauer cells ('cell_type' == 'Hofbauer cell'). Report the infection label exactly as it appears in the 'infection' column (case-sensitive). Answer format: a single label (shape: label).\n",
      "Answer: Tg\n",
      "\n",
      "Task 1\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/cell/paper2/data.h5ad\n",
      "Description: Among the four infection conditions (UI, Tg, Pf, Lm) in the provided placenta infection scRNA‑seq dataset, identify the infection condition that has the highest proportion of cells belonging to the cell type that is, on average across all infections, the most abundant (based on the 'cell_type_ontology_term_id' column). Provide the infection label exactly as it appears in the 'infection' column (case‑sensitive). Answer format: a single label (shape: label).\n",
      "Answer: Tg\n",
      "\n",
      "Task 2\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/cell/paper2/data.h5ad\n",
      "Description: For each infection condition (UI, Lm, Pf, Tg) compute the proportion of cells that are annotated with the most abundant cell type identifier CL:3000001. Report the infection label (exact case) that has the highest proportion. Answer format: a single infection label (shape: label).\n",
      "Answer: Tg\n",
      "\n",
      "Task 3\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/cell/paper2/data.h5ad\n",
      "Description: Identify the infection condition (one of UI, Lm, Tg, Pf) that has the highest proportion of cells belonging to the most abundant cell type across the entire dataset (the cell type with the greatest total cell count). Answer format: a single lowercase label (shape: label).\n",
      "Answer: tg\n",
      "\n",
      "Task 4\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/cell/paper2/data.h5ad\n",
      "Description: Using the provided placenta scRNA‑seq dataset, consider only the cells collected at 48 hours post‑infection (hpi = '48h'). For each infection condition (values in the 'infection' column: UI, Tg, Pf, Lm), compute the proportion of cells annotated as 'extravillous trophoblast' (cell_type = 'extravillous trophoblast'). Then calculate the fold‑change of each infection's proportion relative to the uninfected control (UI). Report the infection condition that exhibits the highest fold‑change (i.e., the strongest enrichment of extravillous trophoblast at 48 h compared to UI). Answer format: a single infection label (shape: label).\n",
      "Answer: Tg\n",
      "\n",
      "Group syncytiotrophoblast cell\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/cell/paper5/data.h5ad\n",
      "Description: For each cell type annotated in the 'cell_type' column, compute the average expression of every gene across cells under the two treatment conditions ('cytokines' and 'no_cytokines'). For each cell type, calculate the log2 fold‑change (log2FC) of each gene between the cytokine‑treated and untreated groups, adding a negligible pseudocount to avoid division by zero. Count how many genes have an absolute log2FC ≥ 1 (i.e., at least a 2‑fold change). Identify the cell type with the largest such gene count. Answer format: a single cell type label exactly as it appears in the 'cell_type' column (shape: label).\n",
      "Answer: syncytiotrophoblast cell\n",
      "\n",
      "Task 1\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/cell/paper5/data.h5ad\n",
      "Description: Identify the trophoblast cell type that exhibits the strongest treatment‑induced expression change, defined as the largest absolute log2 fold change (log2FC) observed for any gene when comparing cytokine‑treated cells versus no‑cytokine cells (averaged within each cell type). Answer format: a single lowercase label with underscores (shape: label).\n",
      "Answer: syncytiotrophoblast_cell\n",
      "\n",
      "Group anbl 09p1\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper2/data/dataset3.h5ad\n",
      "Description: Among the macrophage cells in the dataset, first exclude any samples (orig.ident) that contain fewer than 50 cells. Then compute the median `phagocytosis_score1` for each remaining Treatment group. Which Treatment has the highest median `phagocytosis_score1`? Answer format: the exact Treatment string as it appears in the `Treatment` column (shape: label).\n",
      "Answer: ANBL 09P1\n",
      "\n",
      "Task 1\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper2/data/dataset3.h5ad\n",
      "Description: Identify the Treatment group (value in the 'Treatment' column) whose macrophage nuclei have the highest average phagocytosis_score1. Answer format: a single treatment label exactly as it appears in the Treatment column (shape: label).\n",
      "Answer: ANBL 09P1\n",
      "\n",
      "Group mf_typei(cytoplasmic)\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/nature/paper2/processed_myonuclei_data.h5ad\n",
      "Description: Identify the annotation_level0 cell type with the highest average mitochondrial transcript proportion (percent_mito) across all nuclei. Answer format: a single label exactly as it appears in the annotation_level0 column (shape: label).\n",
      "Answer: MF_typeI(cytoplasmic)\n",
      "\n",
      "Task 1\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/nature/paper2/processed_myonuclei_data.h5ad\n",
      "Description: Using the scVI latent embeddings (X_scVI) provided in the AnnData object, compute for each top‑level cell type (annotation_level0) the average Euclidean distance of its nuclei to the centroid of that cell type in the scVI space. Identify the cell type with the smallest average distance (i.e., the most compact cluster). Answer format: a single cell‑type label exactly as it appears in annotation_level0 (shape: label).\n",
      "Answer: MF_typeII(cytoplasmic)\n",
      "\n",
      "Group myh2+myh7+\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/nature/paper2/processed_myonuclei_data.h5ad\n",
      "Description: Using the provided myonuclei single‑cell dataset, determine which non‑missing myofiber_type category has the highest average expression of the gene MYH7. Report the myofiber_type label exactly as it appears in the data (case‑sensitive, including any '+' characters). Answer format: a single label (shape: label).\n",
      "Answer: MYH2+MYH7+\n",
      "\n",
      "Task 1\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/nature/paper2/processed_myonuclei_data.h5ad\n",
      "Description: Among the myofiber_type categories present in the dataset (excluding the 'nan' category), which myofiber type shows the highest average raw expression level of the muscle gene MYH7? Answer format: a single myofiber_type label exactly as it appears in the 'myofiber_type' column (shape: label, case‑sensitive).\n",
      "Answer: MYH2+MYH7+\n",
      "\n",
      "Group 2\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper2/data/dataset2.h5ad\n",
      "Description: Using the snRNA‑seq AnnData file at /mlbio_scratch/videnovi/papers_BIBA/paper2/data/dataset2.h5ad, classify each cell into a 'baseline' or 'post' treatment group based on its 'Treatment' annotation (case‑insensitive): label as 'baseline' if the string contains any of the substrings \"primary\", \"pre\", \"baseline\", or \"dx\"; label as 'post' if the string contains any of \"post\", \"after\", or \"resection\". Exclude cells that are not assigned to either group. For each value of the column 'seurat_clusters', compute the fraction of cells that are baseline and the fraction that are post. Identify the cluster with the greatest absolute difference between its baseline fraction and post fraction. Return the identifier of that cluster as a single label (shape: label).\n",
      "Answer: 2\n",
      "\n",
      "Task 1\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper2/data/dataset2.h5ad\n",
      "Description: After Harmony batch correction, compute for each Seurat cluster (adata.obs['seurat_clusters']) the fraction of cells belonging to the most abundant original batch (adata.obs['orig.ident']). Identify the cluster with the highest such fraction (i.e., the poorest batch mixing). Answer format: a single cluster label as a string (shape: label).\n",
      "Answer: 2\n",
      "\n",
      "Group anbl0532-like therapy with cisplatin & etoposide\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper2/data/dataset3.h5ad\n",
      "Description: Among the macrophage cells, which Treatment has the highest proportion of cells with Response == \"Complete\"? If multiple treatments share the maximal proportion, return the alphabetically first one. Answer format: a single treatment name (shape: label).\n",
      "Answer: ANBL0532-like therapy with cisplatin & etoposide\n",
      "\n",
      "Task 1\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper2/data/dataset3.h5ad\n",
      "Description: From the 'Treatment' column, identify the treatment that exhibits the highest fraction of cells labeled as 'Complete' in the 'Response' column, considering only treatments that have at least 5 cells. If multiple treatments share the highest fraction, select the one that comes first alphabetically. Answer format: a single label (exact treatment string) (shape: label).\n",
      "Answer: ANBL0532-like therapy with cisplatin & etoposide\n",
      "\n",
      "Group 3313\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper5/data/dataset2.h5ad\n",
      "Description: Identify the donor (donor_id) that contributes the highest proportion of cells annotated as \"mesothelial cell\" relative to its total number of cells in the dataset. Answer format: a single donor ID string (shape: label).\n",
      "Answer: 3313\n",
      "\n",
      "Task 1\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper5/data/dataset2.h5ad\n",
      "Description: Using the provided snRNA‑seq AnnData object, identify the cell type that has the highest overall abundance across all cells. Then, among all donors (column 'donor_id'), determine which donor exhibits the highest proportion of that cell type (proportion = cells of that type / total cells for the donor). Answer format: a single donor identifier string (shape: label).\n",
      "Answer: 3313\n",
      "\n",
      "Group neuron\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper7/data/dataset2.h5ad\n",
      "Description: Using the provided scRNA‑seq dataset, identify the cell type that exhibits the highest average expression of the top marker gene (MAB21L2) reported in the pre‑computed differential expression results. Answer format: a single lowercase cell type label (shape: label).\n",
      "Answer: neuron\n",
      "\n",
      "Task 1\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper6/data/dataset1.h5ad\n",
      "Description: After removing cells with a Scrublet doublet score > 0.5, compute the median RNA UMI count (nCount_RNA) for each distinct cell class (the values in adata.obs['Class']). Which class has the highest median? Answer format: a single label (exact string as it appears in the Class column).\n",
      "Answer: Neuron\n",
      "\n",
      "Group mis-c\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper4/data/lymph_dataset.h5ad\n",
      "Description: Determine which Condition has the highest proportion of cells annotated as \"natural killer cell\" in the dataset. Answer format: a single condition label (shape: label).\n",
      "Answer: MIS-C\n",
      "\n",
      "Task 1\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper4/data/my_dataset.h5ad\n",
      "Description: From the provided single-cell dataset, consider only cells annotated as 'macrophage' or 'cycling macrophage'. For each Condition, compute the proportion of cells that are labeled 'cycling macrophage' among these selected cells. Identify the Condition with the highest such proportion. Answer format: a single Condition label (shape: label).\n",
      "Answer: MIS-C\n",
      "\n",
      "Task 2\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper4/data/lymph_dataset.h5ad\n",
      "Description: Using the provided single-cell RNA‑seq dataset of human cardiac lymphocytes, consider the cell type annotations in the column 'author_cell_type'. Define NK cells as those labeled 'NK_CD16hi' or 'NK_CD56hi', and define CD8+ T cells as any cell whose annotation starts with 'CD8T_'. For each disease condition (values in the 'Condition' column), compute the average expression of the gene IFNG (interferon‑γ) separately in NK cells and in CD8+ T cells. Then calculate the difference (average IFNG in NK cells minus average IFNG in CD8+ T cells) for each condition. Identify the condition with the largest positive difference (i.e., the strongest relative up‑regulation of IFNG in NK cells compared to CD8+ T cells). Answer format: a single label exactly matching the condition name as it appears in the 'Condition' column (shape: label).\n",
      "Answer: MIS-C\n",
      "\n",
      "Group adh+fibro\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper7/data/dataset2.h5ad\n",
      "Description: Using the provided scRNA‑seq dataset, compare the earliest developmental stage (adj_stage = Pcw5.1) to the latest stage (adj_stage = Pcw9.3). For each stage, compute the proportion of cells belonging to each cell type (as given in the 'celltype' column). Identify the cell type whose proportion increase (latest stage proportion minus earliest stage proportion) is the greatest. Answer format: a single cell type label exactly as it appears in the 'celltype' column (shape: label).\n",
      "Answer: ADH+Fibro\n",
      "\n",
      "Task 1\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper7/data/dataset2.h5ad\n",
      "Description: Identify the cell type that shows the greatest increase in relative abundance (proportion of cells) from the earliest developmental stage (Pcw5.1) to the latest developmental stage (Pcw9.3) in the dataset. Answer format: a single cell type label exactly as it appears in the 'celltype' column (shape: label).\n",
      "Answer: ADH+Fibro\n",
      "\n",
      "Group pf\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/cell/paper2/data.h5ad\n",
      "Description: In the dataset, consider only Hofbauer cells sampled at 24 hours post‑infection. For each infection condition (Lm, Tg, Pf) perform a differential expression analysis versus the uninfected (UI) cells using Scanpy's Wilcoxon rank‑sum test (all genes, no preprocessing). Count the number of genes that satisfy adjusted p‑value < 0.05 and log2‑fold‑change > 0.25. Which infection condition has the largest such count? Answer format: a single infection label (one of UI, Lm, Tg, Pf).\n",
      "Answer: Pf\n",
      "\n",
      "Task 1\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/cell/paper2/data.h5ad\n",
      "Description: In the dataset, consider only the cells annotated as \"Hofbauer cell\". For each infection condition (UI, Lm, Pf, Tg), compute the average raw (unnormalized) count of the gene IL6 across those Hofbauer cells. Identify the infection condition with the highest mean IL6 expression. Answer format: a single two‑letter label exactly as it appears in the \"infection\" column (shape: label).\n",
      "Answer: Pf\n",
      "\n",
      "Task 2\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/cell/paper2/data.h5ad\n",
      "Description: Within the cell type 'Hofbauer cell', determine which infection condition ('UI', 'Tg', 'Pf', or 'Lm') exhibits the highest average expression of the gene CXCL10. Report the answer as the exact infection label (case-sensitive) with shape: label.\n",
      "Answer: Pf\n",
      "\n",
      "Group myh1+myh2+myh7+\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/nature/paper2/processed_myonuclei_data.h5ad\n",
      "Description: Identify the myofiber_type annotation that exhibits the highest average percent_mito across all nuclei in the dataset. Answer format: a single label exactly as it appears in the 'myofiber_type' column (shape: label).\n",
      "Answer: MYH1+MYH2+MYH7+\n",
      "\n",
      "Task 1\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/nature/paper2/processed_myonuclei_data.h5ad\n",
      "Description: Identify the myofiber_type that exhibits the strongest donor-specific bias, defined as the highest proportion of its nuclei contributed by a single donor. Answer format: a single label exactly as it appears in the 'myofiber_type' column (shape: label).\n",
      "Answer: MYH1+MYH2+MYH7+\n",
      "\n",
      "Group 11\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/nature/paper6/data.h5ad\n",
      "Description: Using the provided AnnData object, filter out cells flagged as doublets (`predicted_doublet == True`) and cells with mitochondrial transcript proportion > 10%. Then, compute the inverse Local Inverse Simpson’s Index (iLISI) for donor mixing within each Leiden cluster (using the `X_pca` embedding, k=30 nearest neighbors). Report the Leiden cluster identifier that has the lowest iLISI (i.e., the cluster with the poorest donor mixing). Answer format: a single integer (shape: label).\n",
      "Answer: 11\n",
      "\n",
      "Task 1\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper2/data/dataset1.h5ad\n",
      "Description: Using the provided snRNA‑seq dataset, compare the relative abundance of Seurat clusters between the two treatment stages indicated by the 'Stage_Code' column (DX = pre‑chemotherapy, PTX = post‑chemotherapy). For each cluster, compute the proportion of cells within each stage, then calculate the log2 fold‑change (PTX proportion / DX proportion). Identify the cluster that shows the greatest positive log2 fold‑change (i.e., the strongest increase after chemotherapy). Answer format: a single cluster identifier as a label (shape: label).\n",
      "Answer: 11\n",
      "\n",
      "Group fibroblast\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Dataset path: /mlbio_scratch/videnovi/papers_BIBA/paper2/data/dataset2.h5ad\n",
      "Description: In the provided snRNA‑seq dataset, use the binary metadata column **MYCN_Status** (values: 'AMP' and 'NON') as the grouping variable. Identify the gene that has the highest value in the column `vf_vst_counts_variance.standardized` among all genes with a non‑missing `gene_name`. For each distinct `cell_type`, compute the average expression of this gene in the 'AMP' and 'NON' groups, then calculate the log2 fold‑change (NON over AMP) adding a small epsilon (1e‑6) to avoid division by zero. Return the **cell_type** that shows the largest positive log2 fold‑change. Answer format: a single lowercase label with underscores (shape: label).\n",
      "Answer: fibroblast\n",
      "\n",
      "Task 1\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/cell/paper2/data.h5ad\n",
      "Description: Identify the cell type that exhibits the greatest enrichment (highest fold change of proportion) in Plasmodium falciparum (Pf) infected samples compared to uninfected (UI) samples at the 24 hour time point. Answer format: a single lowercase label without spaces (shape: label).\n",
      "Answer: fibroblast\n",
      "\n",
      "Group small intestine peyer's patch\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/nature/paper1/cxg_data.h5ad\n",
      "Description: Using the provided scRNA‑seq dataset, compute for each tissue the proportion of cells annotated as 'regulatory T cell' (number of regulatory T cells divided by total cells in that tissue). Identify the tissue with the highest proportion. Answer format: a single tissue label exactly as it appears in the 'tissue' column (shape: label).\n",
      "Answer: small intestine Peyer's patch\n",
      "\n",
      "Task 1\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/nature/paper1/cxg_data.h5ad\n",
      "Description: Using the provided scRNA‑seq dataset, identify regulatory T cells (Tregs) by selecting cells whose 'final_annotation' contains the substring \"Treg\" (case‑insensitive) or whose 'cell_type' equals \"regulatory T cell\". For each tissue (values in the 'tissue' column), compute the proportion of Treg cells among all cells in that tissue. Report the tissue with the highest Treg proportion. Answer format: a single tissue label exactly as it appears in the 'tissue' column (shape: label).\n",
      "Answer: small intestine Peyer's patch\n",
      "\n",
      "Task 2\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/nature/paper1/cxg_data.h5ad\n",
      "Description: Among the four sampled tissues, which tissue exhibits the highest proportion of regulatory T cell cells? Report the tissue name exactly as it appears in the 'tissue' column (case‑sensitive, including spaces and punctuation). Answer format: a single label (shape: label).\n",
      "Answer: small intestine Peyer's patch\n",
      "\n",
      "Group mf_type-myh8+\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/nature/paper2/processed_myonucleicells_data.h5ad\n",
      "Description: Identify the annotation_level2 cell type whose proportion of cells originating from the 'cells' batch (i.e., fraction = #cells in batch 'cells' / total #cells for that type) is maximal. If multiple cell types share the maximal proportion, return the one that comes first alphabetically. Answer format: a single label (shape: label) exactly as it appears in the annotation_level2 column.\n",
      "Answer: MF_type-MYH8+\n",
      "\n",
      "Task 1\n",
      "Dataset path: /mlbio_scratch/panigrah/agents/data/nature/paper2/processed_myonucleicells_data.h5ad\n",
      "Description: Using the provided single-nucleus/cell dataset, consider the fine-grained cell-type annotation column 'annotation_level2'. Restrict the analysis to cell types that contain at least 100 cells. Compute the average mitochondrial RNA proportion ('percent_mito') for each remaining cell type and identify the one with the highest mean value. Answer format: a single cell-type label exactly as it appears in 'annotation_level2' (shape: label).\n",
      "Answer: MF_type-MYH8+\n",
      "\n"
     ]
    }
   ],
   "source": [
    "for gta_key in nonsingle_keys:\n",
    "    tasks = tasks_by_gta[gta_key]\n",
    "    print(f\"Group {gta_key}\")\n",
    "    print(100 * '-')\n",
    "    for tid, task in enumerate(tasks):\n",
    "        print(f\"Task {tid}\")\n",
    "        print(f\"Dataset path: {task['dataset_path']}\")\n",
    "        print(f\"Description: {task['task_description']}\")\n",
    "        print(f\"Answer: {task['ground_truth_answer']}\")\n",
    "        print()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "b2cf5638-9b80-4091-9280-7e0139067853",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Dict with the following \n",
    "# - keys: nonsingle GTA keys\n",
    "# - values: list of duplicate tasks ids where the FIRST task id \n",
    "#           refers to the representative of the whole group\n",
    "duplicates_by_gta = {\n",
    "    \"paep\": [[0, 1]],\n",
    "    \"lymphocyte\": [[2, 0], [3, 4], [1]],\n",
    "    \"monocyte\": [[0], [1]],\n",
    "    \"3357\": [[0, 1, 2]],\n",
    "    \"nmj accessory\": [[0, 2], [4, 1], [3]],\n",
    "    \"mf-i\": [[0], [1], [2]],\n",
    "    \"goblet cell\": [[3, 1, 2, 4], [0], [5]],\n",
    "    \"b_cell\": [[2, 0], [1]],\n",
    "    \"r24_p2538\": [[1, 0]],\n",
    "    \"tom_vct\": [[1, 0]],\n",
    "    \"hepatocyte\": [[0], [1]],\n",
    "    \"5\": [[0], [1]],\n",
    "    \"4\": [[0], [1], [2]],\n",
    "    \"control\": [[8, 0, 1, 3, 5, 7], [2], [4], [6]],\n",
    "    \"3427\": [[0], [1]],\n",
    "    \"xist\": [[0], [1], [2], [3], [4]],\n",
    "    \"b-lymphocytes\": [[1, 0, 2]],\n",
    "    \"alzheimer disease\": [[1, 0]],\n",
    "    \"interferon-response\": [[0], [1]],\n",
    "    \"csh1\": [[0, 1]],\n",
    "    \"anbl12p1-like chemotherapy\": [[0], [1], [2], [3], [4]],\n",
    "    \"mt-co3\": [[0], [1], [2]],\n",
    "    \"464c\": [[1, 0]],\n",
    "    \"nuclei\": [[0], [1]],\n",
    "    \"mf-isn(fg)\": [[0], [1], [2], [3], [4]],\n",
    "    \"9\": [[1, 0, 2, 3, 5], [4]],\n",
    "    \"b cell\": [[3, 0], [1], [2]],\n",
    "    \"13\": [[0], [1]],\n",
    "    \"endothelial_cell\": [[3, 0, 2], [1]],\n",
    "    \"adipocyte of omentum tissue\": [[0], [1], [2], [3]],\n",
    "    \"grin2a\": [[0], [1]],\n",
    "    \"caecum\": [[0], [1]],\n",
    "    \"6720483e21rik\": [[2, 1], [0]],\n",
    "    \"evt_late_3\": [[1, 0, 2, 4], [3]],\n",
    "    \"cardiac_neuron\": [[0], [1]],\n",
    "    \"mesenteric lymph node\": [[1, 0, 2]],\n",
    "    \"tnfrsf12a+muscs\": [[2, 1], [0]],\n",
    "    \"500c\": [[0], [1]],\n",
    "    \"8\": [[2, 0, 1, 3, 4]],\n",
    "    \"14\": [[1, 2], [0]],\n",
    "    \"3\": [[0, 1], [2]],\n",
    "    \"20\": [[1, 0]],\n",
    "    \"central memory cd4-positive, alpha-beta t cell\": [[0, 1]],\n",
    "    \"fp236383.1\": [[0], [1]],\n",
    "    \"young\": [[0], [1]],\n",
    "    \"mf-iisc(fg)\": [[0], [1]],\n",
    "    \"tg\": [[3, 1], [0], [2], [4]],\n",
    "    \"syncytiotrophoblast cell\": [[0], [1]],\n",
    "    \"anbl 09p1\": [[0], [1]],\n",
    "    \"mf_typei(cytoplasmic)\": [[0], [1]],\n",
    "    \"myh2+myh7+\": [[0], [1]],\n",
    "    \"2\": [[0], [1]],\n",
    "    \"anbl0532-like therapy with cisplatin & etoposide\": [[0], [1]],\n",
    "    \"3313\": [[0], [1]],\n",
    "    \"neuron\": [[0], [1]],\n",
    "    \"mis-c\": [[0], [1], [2]],\n",
    "    \"adh+fibro\": [[0, 1]],\n",
    "    \"pf\": [[0], [1], [2]],\n",
    "    \"myh1+myh2+myh7+\": [[0], [1]],\n",
    "    \"11\": [[0], [1]],\n",
    "    \"fibroblast\": [[0], [1]],\n",
    "    \"small intestine peyer's patch\": [[0, 2], [1]],\n",
    "    \"mf_type-myh8+\": [[0], [1]],\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "bff07eaf-7363-4d2a-b5b9-24ca325aa3a3",
   "metadata": {},
   "outputs": [],
   "source": [
    "assert sorted(nonsingle_keys) == sorted(duplicates_by_gta.keys())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "ca460fed-9e7a-4208-889d-166423053df2",
   "metadata": {},
   "outputs": [],
   "source": [
    "for gta_key in nonsingle_keys:\n",
    "    tasks = tasks_by_gta[gta_key]\n",
    "    dups_ids = duplicates_by_gta[gta_key]\n",
    "    assert list(range(len(tasks))) == sorted(sum(dups_ids, []))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "65c30389-d068-489c-9c88-6d3782292051",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Duplicate groups: 137\n"
     ]
    }
   ],
   "source": [
    "print(f\"Duplicate groups: {sum(len(dups_ids) for dups_ids in duplicates_by_gta.values())}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "625f481a-1d3c-47b4-89cd-0e0de109da64",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_clean_tasks(tasks_by_key, duplicates_by_key):\n",
    "    clean_tasks = []\n",
    "\n",
    "    for key, tasks in tasks_by_key.items():\n",
    "        task_ids = set(range(len(tasks)))\n",
    "        \n",
    "        for dup_ids in duplicates_by_key.get(key, []):\n",
    "            rem_tids = set(dup_ids[1:])  # the first index is the representative of a duplicates group\n",
    "            task_ids -= rem_tids\n",
    "\n",
    "        clean_tasks.extend([tasks[tid] for tid in task_ids])\n",
    "\n",
    "    return clean_tasks"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "31019470-ceec-448c-b21e-31f5a7ab3fba",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Tasks after GTA-wise filtration: 286\n"
     ]
    }
   ],
   "source": [
    "clean_tasks = get_clean_tasks(tasks_by_gta, duplicates_by_gta)\n",
    "print(f\"Tasks after GTA-wise filtration: {len(clean_tasks)}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "d794c732-3df0-4978-a544-2912d0a75f8f",
   "metadata": {},
   "outputs": [],
   "source": [
    "datasets_list = load_data(\"/mlbio_scratch/kodryan/Biomni/data/all_sc_datasets.json\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "cbf32df5-2499-4f96-828e-8dab74968bd7",
   "metadata": {},
   "outputs": [],
   "source": [
    "tasks_by_dataset = {ds: [task for task in clean_tasks if task['dataset_path'] == ds] for ds in datasets_list}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "eef6758a-b2e3-446f-a0f1-2179e6ac5ca7",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dataset 1: /mlbio_scratch/videnovi/papers_BIBA/paper2/data/dataset1.h5ad\n",
      "# tasks: 15\n",
      "\n",
      "Dataset 2: /mlbio_scratch/videnovi/papers_BIBA/paper2/data/dataset2.h5ad\n",
      "# tasks: 16\n",
      "\n",
      "Dataset 3: /mlbio_scratch/videnovi/papers_BIBA/paper2/data/dataset3.h5ad\n",
      "# tasks: 14\n",
      "\n",
      "Dataset 4: /mlbio_scratch/videnovi/papers_BIBA/paper4/data/dataset1.h5ad\n",
      "# tasks: 13\n",
      "\n",
      "Dataset 5: /mlbio_scratch/videnovi/papers_BIBA/paper4/data/lymph_dataset.h5ad\n",
      "# tasks: 15\n",
      "\n",
      "Dataset 6: /mlbio_scratch/videnovi/papers_BIBA/paper4/data/my_dataset.h5ad\n",
      "# tasks: 11\n",
      "\n",
      "Dataset 7: /mlbio_scratch/videnovi/papers_BIBA/paper5/data/dataset2.h5ad\n",
      "# tasks: 12\n",
      "\n",
      "Dataset 8: /mlbio_scratch/videnovi/papers_BIBA/paper5/data/VA_adipocytes.h5ad\n",
      "# tasks: 16\n",
      "\n",
      "Dataset 9: /mlbio_scratch/videnovi/papers_BIBA/paper6/data/dataset1.h5ad\n",
      "# tasks: 16\n",
      "\n",
      "Dataset 10: /mlbio_scratch/videnovi/papers_BIBA/paper7/data/dataset2.h5ad\n",
      "# tasks: 15\n",
      "\n",
      "Dataset 11: /mlbio_scratch/panigrah/agents/data/nature/paper1/cxg_data.h5ad\n",
      "# tasks: 13\n",
      "\n",
      "Dataset 12: /mlbio_scratch/panigrah/agents/data/nature/paper2/processed_MuSC_data.h5ad\n",
      "# tasks: 14\n",
      "\n",
      "Dataset 13: /mlbio_scratch/panigrah/agents/data/nature/paper2/processed_human_muscle_data.h5ad\n",
      "# tasks: 16\n",
      "\n",
      "Dataset 14: /mlbio_scratch/panigrah/agents/data/nature/paper2/processed_myonuclei_data.h5ad\n",
      "# tasks: 14\n",
      "\n",
      "Dataset 15: /mlbio_scratch/panigrah/agents/data/nature/paper2/processed_myonucleicells_data.h5ad\n",
      "# tasks: 16\n",
      "\n",
      "Dataset 16: /mlbio_scratch/panigrah/agents/data/nature/paper5/mouse_data.h5ad\n",
      "# tasks: 8\n",
      "\n",
      "Dataset 17: /mlbio_scratch/panigrah/agents/data/nature/paper6/data.h5ad\n",
      "# tasks: 12\n",
      "\n",
      "Dataset 18: /mlbio_scratch/panigrah/agents/data/cell/paper1/data.h5ad\n",
      "# tasks: 14\n",
      "\n",
      "Dataset 19: /mlbio_scratch/panigrah/agents/data/cell/paper2/data.h5ad\n",
      "# tasks: 15\n",
      "\n",
      "Dataset 20: /mlbio_scratch/panigrah/agents/data/cell/paper3/data.h5ad\n",
      "# tasks: 10\n",
      "\n",
      "Dataset 21: /mlbio_scratch/panigrah/agents/data/cell/paper5/data.h5ad\n",
      "# tasks: 11\n",
      "\n"
     ]
    }
   ],
   "source": [
    "for did, (dataset_path, tasks) in enumerate(tasks_by_dataset.items(), 1):\n",
    "    print(f\"Dataset {did}: {dataset_path}\")\n",
    "    print(f\"# tasks: {len(tasks)}\")\n",
    "    print()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "f0151cee-aff8-459c-b25c-625294c3ab58",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dataset: /mlbio_scratch/videnovi/papers_BIBA/paper2/data/dataset1.h5ad\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Description: Using the provided single-nucleus RNA‑seq dataset (dataset1.h5ad), compute the mean Euclidean distance between pre‑treatment cells (where the 'Response' column is missing/NaN) and post‑treatment cells (where 'Response' is not NaN) for each unique value in the 'malignancy' column, based on the first 30 principal components stored in adata.obsm['X_pca']. Report the malignancy label that has the largest mean pre‑post distance. Answer format: a single token exactly matching one entry of the 'malignancy' column (shape: label).\n",
      "Answer: Nonmalignant\n",
      "\n",
      "Task 1\n",
      "Description: Using the provided snRNA‑seq dataset, identify the cell type whose relative abundance increases the most after chemotherapy (Stage_Code = PTX) compared to diagnosis (Stage_Code = DX). For each cell type, compute its proportion of cells within each stage, calculate the log2 fold‑change (log2( proportion_PT X / proportion_DX )), and consider only cell types present in both stages. Report the exact cell_type label (as it appears in the 'cell_type' column) with the highest positive log2 fold‑change. Answer format: a single label string.\n",
      "Answer: hepatocyte\n",
      "\n",
      "Task 2\n",
      "Description: Using the provided snRNA‑seq dataset, compare gene expression between the two treatment stages (Stage_Code = DX for diagnosis and Stage_Code = PTX for post‑chemotherapy). For each cell type, compute the log2 fold‑change (log2FC) of every gene as log2((mean PTX + 1e‑6) / (mean DX + 1e‑6)) using raw counts. Identify, for each cell type, the gene with the highest positive log2FC, then determine which cell type has the largest such positive log2FC among all cell types. Answer format: a single cell‑type label (shape: label).\n",
      "Answer: hepatocyte\n",
      "\n",
      "Task 3\n",
      "Description: Among the unsupervised Seurat clusters (column 'seurat_clusters') in the provided single-nucleus RNA‑seq dataset, identify the cluster that exhibits the highest proportion of cells annotated with the clinical response \"Complete\" (consider only cells with a non‑missing Response value). Answer format: a single integer label representing the cluster ID (shape: label).\n",
      "Answer: 5\n",
      "\n",
      "Task 4\n",
      "Description: Using the provided snRNA‑seq dataset, consider only cells annotated as malignant neuroblastoma (malignancy == \"Malignant\" and cell_type == \"neuroblast (sensu Vertebrata)\"). Compare the pre‑treatment group (Stage_Code == \"DX\") with the post‑treatment group (Stage_Code == \"PTX\"). Compute the mean expression of each gene in each group, calculate log2 fold‑change (add a pseudocount of 1e-6 to avoid division by zero), and identify the gene with the largest positive log2 fold‑change (i.e., the most up‑regulated after treatment). Provide the answer as a single HGNC gene symbol (shape: gene_symbol).\n",
      "Answer: CYP11B2\n",
      "\n",
      "Task 5\n",
      "Description: Within the malignant cells of the dataset, identify the most abundant Seurat cluster (by cell count). In that dominant cluster, compare cells from the pre‑treatment stage (Stage_Code = \"DX\") to post‑treatment stage (Stage_Code = \"PTX\"). Using a Wilcoxon rank‑sum test (as implemented by Scanpy's rank_genes_groups) and considering genes with adjusted p‑value < 0.05, report the gene that shows the greatest positive log2 fold‑change (i.e., strongest up‑regulation) after chemotherapy. Answer format: a single HGNC gene symbol in uppercase (shape: gene_symbol).\n",
      "Answer: MT-CO3\n",
      "\n",
      "Task 6\n",
      "Description: Identify the cell type (as listed in the 'cell_type' column) that has the highest proportion of cells labeled as 'Complete' in the 'Response' column. Answer format: a single label exactly matching the cell_type entry (shape: label).\n",
      "Answer: cortical cell of adrenal gland\n",
      "\n",
      "Task 7\n",
      "Description: Using the provided snRNA‑seq dataset, compute for each integrated cluster (column 'integrated_snn_res.0.2') the proportion of cells labeled as 'Malignant' within the diagnosis stage (Stage_Code = DX) and the post‑chemotherapy stage (Stage_Code = PTX). Identify the cluster that shows the largest positive difference (post‑chemotherapy proportion minus diagnosis proportion). Answer format: a single cluster identifier as a string (shape: label).\n",
      "Answer: 14\n",
      "\n",
      "Task 8\n",
      "Description: Using the provided snRNA‑seq dataset, compute for each Seurat cluster (seurat_clusters) the fraction of cells annotated as 'Malignant' within the diagnosis (DX) and post‑chemotherapy (PTX) stages. Identify the cluster that shows the greatest positive increase in malignant fraction from DX to PTX (i.e., PTX fraction minus DX fraction is maximal). Answer format: a single cluster identifier as a label (e.g., \"14\").\n",
      "Answer: 14\n",
      "\n",
      "Task 9\n",
      "Description: In the provided snRNA‑seq dataset of high‑risk neuroblastoma, consider only cells annotated as 'Malignant'. Compare the two treatment stages: pre‑treatment (Stage_Code = DX) and post‑treatment (Stage_Code = PTX). For each gene, compute the average expression across cells in each stage, add a pseudocount of 1, and calculate the log2 fold‑change (log2FC = log2[(mean_PTX+1)/(mean_DX+1)]). Identify the gene with the largest positive log2FC (i.e., the strongest up‑regulation after chemotherapy). Answer format: a single HGNC gene symbol (uppercase).\n",
      "Answer: RALYL\n",
      "\n",
      "Task 10\n",
      "Description: Among the annotated cell types in the dataset, which cell type shows the largest positive change in the fraction of cells labeled as 'Malignant' when comparing the diagnostic stage (DX) to the post‑chemotherapy stage (PTX)? Report the cell type name exactly as it appears in the 'cell_type' column (shape: label).\n",
      "Answer: kidney cell\n",
      "\n",
      "Task 11\n",
      "Description: Using the provided snRNA‑seq dataset of high‑risk neuroblastoma, consider only cells annotated as malignant (malignancy == 'Malignant') and of cell type 'neuroblast (sensu Vertebrata)'. Infer treatment condition from the sample identifier: identifiers ending with '_REG1' correspond to the pre‑treatment group and those ending with '_REG2' to the post‑treatment group. Compute the average expression of each gene across cells in each group, calculate the log2 fold‑change (post / pre) with a pseudocount of 1e‑6, and identify the gene with the greatest absolute log2 fold‑change. Provide the HGNC gene symbol of that gene as the answer. Answer format: a single gene symbol (shape: gene_symbol).\n",
      "Answer: SYNJ2BP-COX16\n",
      "\n",
      "Task 12\n",
      "Description: Identify the cell type that exhibits the highest proportion of malignant cells in the dataset. Provide the exact cell type label as it appears in the 'cell_type' column of the AnnData object (shape: label).\n",
      "Answer: neuroblast (sensu Vertebrata)\n",
      "\n",
      "Task 13\n",
      "Description: Using the provided snRNA‑seq dataset, compare the relative abundance of Seurat clusters between the two treatment stages indicated by the 'Stage_Code' column (DX = pre‑chemotherapy, PTX = post‑chemotherapy). For each cluster, compute the proportion of cells within each stage, then calculate the log2 fold‑change (PTX proportion / DX proportion). Identify the cluster that shows the greatest positive log2 fold‑change (i.e., the strongest increase after chemotherapy). Answer format: a single cluster identifier as a label (shape: label).\n",
      "Answer: 11\n",
      "\n",
      "Task 14\n",
      "Description: Identify the cluster (as given in the 'seurat_clusters' column) that shows the largest absolute difference in mean log1p‑normalized expression of the gene MKI67 between diagnostic (DX) and post‑treatment (PTX) samples (Stage_Code). Report the cluster identifier as a string label (shape: label).\n",
      "Answer: 15\n",
      "\n",
      "Dataset: /mlbio_scratch/videnovi/papers_BIBA/paper2/data/dataset2.h5ad\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Description: Using the provided snRNA‑seq dataset, identify the batch (orig.ident) that exhibits the highest average expression of the oncogene MYCN. Report the answer as the exact batch identifier string (shape: label).\n",
      "Answer: NB_7767_3495_REG1\n",
      "\n",
      "Task 1\n",
      "Description: Using the provided single-nucleus RNA‑seq dataset (dataset2.h5ad), compute for each Seurat cluster (column 'seurat_clusters') the proportion of cells belonging to each treatment condition (column 'Treatment'). Then calculate, for each cluster‑treatment pair, an enrichment score defined as (cluster proportion) / (global proportion of that treatment across all cells). Identify the cluster whose maximum enrichment score (i.e., the most over‑represented treatment in that cluster) is the highest among all clusters. Report the identifier of that cluster exactly as it appears in the 'seurat_clusters' column. Answer format: a single label (shape: label).\n",
      "Answer: 5\n",
      "\n",
      "Task 2\n",
      "Description: Identify the seurat_clusters label (numeric cluster identifier) that exhibits the highest proportion of MYCN-amplified cells (MYCN_Status == 'AMP') among all clusters. Answer format: a single integer label (shape: label).\n",
      "Answer: 4\n",
      "\n",
      "Task 3\n",
      "Description: Identify the seurat_cluster (label) that exhibits the highest average expression of the MYCN oncogene across all cells in the dataset. Use the 'seurat_clusters' annotation and the expression matrix. Answer format: a single cluster label as a string (e.g., '4'), shape: label.\n",
      "Answer: 4\n",
      "\n",
      "Task 4\n",
      "Description: Within the snRNA‑seq dataset, cells are assigned to six Seurat clusters (0‑5). Considering only the cells in cluster 0, compute the average expression of the gene MYCN for each distinct value in the 'Treatment' metadata column. Identify the treatment that yields the highest mean MYCN expression. Answer format: a single label matching exactly the treatment string as it appears in adata.obs['Treatment'] (shape: label).\n",
      "Answer: ANBL12P1-like chemotherapy\n",
      "\n",
      "Task 5\n",
      "Description: Using the provided snRNA‑seq dataset of neoplastic cells, compute the average expression of the oncogene MYCN (gene symbol) in diagnostic (Stage_Code == \"DX\") cells and in each post‑treatment (Stage_Code == \"PTX\") regimen. For each post‑treatment regimen, calculate the log2‑fold‑change of the mean MYCN expression relative to the diagnostic mean (add a small epsilon = 1e‑6 to avoid division by zero). Identify the post‑treatment regimen with the largest positive log2‑fold‑change. Report the exact treatment name as it appears in the 'Treatment' column. Answer format: a single label (shape: label).\n",
      "Answer: ANBL12P1-like chemotherapy\n",
      "\n",
      "Task 6\n",
      "Description: Among the malignant neuroblast cells (cell_type == 'neuroblast (sensu Vertebrata)' and malignancy == 'Malignant'), compute the average expression of the oncogene MYCN for each Treatment group. Identify the Treatment label with the highest average MYCN expression. Answer format: a single treatment label string exactly as it appears in the data (shape: label).\n",
      "Answer: ANBL12P1-like chemotherapy\n",
      "\n",
      "Task 7\n",
      "Description: Identify the treatment condition (as listed in the 'Treatment' column of the provided AnnData object) that exhibits the highest average normalized expression of the gene MYCN across all cells. Provide the answer as a single exact label string (shape: label), matching the Treatment entry case‑sensitively and without extra spaces.\n",
      "Answer: ANBL12P1-like chemotherapy\n",
      "\n",
      "Task 8\n",
      "Description: Within the neuroblastoma cells (cell_type == 'neuroblast (sensu Vertebrata)') in the provided dataset, compute the average raw count expression of the oncogene MYCN for each distinct Treatment group. Return the Treatment name that yields the highest average MYCN expression. Answer format: a single label exactly matching the Treatment string (shape: label).\n",
      "Answer: ANBL12P1-like chemotherapy\n",
      "\n",
      "Task 9\n",
      "Description: Identify the Treatment group that has the highest proportion of cells labeled as 'MES' in the 'cell_state' column of the provided AnnData object. Answer format: a single Treatment label exactly as it appears in the data (shape: label).\n",
      "Answer: CHP667\n",
      "\n",
      "Task 10\n",
      "Description: Using the provided snRNA-seq dataset of high-risk neuroblastoma (dataset2.h5ad), consider only cells annotated as neuroblast (cell_type == \"neuroblast (sensu Vertebrata)\"). For each treatment stage (Stage_Code) – diagnosis (DX) and post‑chemotherapy (PTX) – compute the proportion of cells belonging to each cell_state. Identify the cell_state that shows the greatest positive change in proportion when comparing PTX to DX (i.e., PTX proportion minus DX proportion). Answer format: a single label exactly matching the cell_state name (shape: label).\n",
      "Answer: ADRN-Calcium\n",
      "\n",
      "Task 11\n",
      "Description: Using the snRNA‑seq AnnData file at /mlbio_scratch/videnovi/papers_BIBA/paper2/data/dataset2.h5ad, classify each cell into a 'baseline' or 'post' treatment group based on its 'Treatment' annotation (case‑insensitive): label as 'baseline' if the string contains any of the substrings \"primary\", \"pre\", \"baseline\", or \"dx\"; label as 'post' if the string contains any of \"post\", \"after\", or \"resection\". Exclude cells that are not assigned to either group. For each value of the column 'seurat_clusters', compute the fraction of cells that are baseline and the fraction that are post. Identify the cluster with the greatest absolute difference between its baseline fraction and post fraction. Return the identifier of that cluster as a single label (shape: label).\n",
      "Answer: 2\n",
      "\n",
      "Task 12\n",
      "Description: After Harmony batch correction, compute for each Seurat cluster (adata.obs['seurat_clusters']) the fraction of cells belonging to the most abundant original batch (adata.obs['orig.ident']). Identify the cluster with the highest such fraction (i.e., the poorest batch mixing). Answer format: a single cluster label as a string (shape: label).\n",
      "Answer: 2\n",
      "\n",
      "Task 13\n",
      "Description: Within the malignant neuroblast cells (cell_type == 'neuroblast (sensu Vertebrata)'), compare the two treatment groups: 'ANBL12P1-like' (pre‑chemotherapy) and 'ANBL12P1-like chemotherapy' (post‑chemotherapy). Perform a differential expression analysis using Scanpy's rank_genes_groups with method='t-test' on the existing normalized expression matrix (use_raw=False). Identify the gene with the highest positive log2‑fold‑change (most up‑regulated in the post‑chemotherapy group). Provide the answer as a single HGNC gene symbol in uppercase (shape: gene_symbol).\n",
      "Answer: ZIC2\n",
      "\n",
      "Task 14\n",
      "Description: In the provided snRNA‑seq dataset, use the binary metadata column **MYCN_Status** (values: 'AMP' and 'NON') as the grouping variable. Identify the gene that has the highest value in the column `vf_vst_counts_variance.standardized` among all genes with a non‑missing `gene_name`. For each distinct `cell_type`, compute the average expression of this gene in the 'AMP' and 'NON' groups, then calculate the log2 fold‑change (NON over AMP) adding a small epsilon (1e‑6) to avoid division by zero. Return the **cell_type** that shows the largest positive log2 fold‑change. Answer format: a single lowercase label with underscores (shape: label).\n",
      "Answer: fibroblast\n",
      "\n",
      "Task 15\n",
      "Description: Using the provided snRNA‑seq dataset, restrict analysis to cells annotated as \"neuroblast (sensu Vertebrata)\". Compare the two treatment stages indicated by the column 'Stage_Code' (DX = pre‑treatment, PTX = post‑chemotherapy). Perform a Wilcoxon rank‑sum differential expression test between PTX and DX groups and identify the gene with the highest positive log₂ fold‑change (most up‑regulated in PTX). Return the gene symbol as a single uppercase HGNC symbol. Answer format: a single gene symbol (shape: gene_symbol).\n",
      "Answer: NRG1\n",
      "\n",
      "Dataset: /mlbio_scratch/videnovi/papers_BIBA/paper2/data/dataset3.h5ad\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Description: In the snRNA-seq dataset of macrophages from high‑risk neuroblastoma before (DX) and after (PTX) chemotherapy, identify the gene that shows the strongest statistically significant up‑regulation (largest positive log₂ fold‑change, FDR < 0.05) in post‑treatment macrophages compared to diagnostic macrophages. Answer format: a single HGNC gene symbol (shape: gene_symbol).\n",
      "Answer: PAEP\n",
      "\n",
      "Task 1\n",
      "Description: Using the provided snRNA‑seq dataset of macrophages, perform a differential expression analysis between post‑chemotherapy (Stage_Code = PTX) and pre‑chemotherapy (Stage_Code = DX) cells **within each Seurat cluster** (seurat_clusters). For each cluster, count the number of genes that are significantly up‑regulated in PTX compared to DX, defined as adjusted p‑value < 0.05 and log2 fold‑change > 0.5 (t‑test). Identify the cluster with the **largest** such count. Answer format: a single integer label corresponding to the Seurat cluster identifier (shape: label).\n",
      "Answer: 4\n",
      "\n",
      "Task 2\n",
      "Description: Within the most abundant seurat cluster of macrophage nuclei, perform differential expression between the post‑treatment (PTX) and diagnosis (DX) groups. Identify the gene that is most significantly up‑regulated in PTX versus DX (i.e., the up‑regulated gene with the smallest Benjamini‑Hochberg adjusted p‑value). Answer format: a single HGNC gene symbol (shape: gene_symbol).\n",
      "Answer: MT-CO3\n",
      "\n",
      "Task 3\n",
      "Description: Using the provided snRNA‑seq dataset of macrophages, compute the proportion of cells belonging to each 'global_cluster' within the diagnostic stage (Stage_Code = 'DX') and the post‑treatment stage (Stage_Code = 'PTX'). Calculate the fold‑change as (PTX proportion + 1e‑6) / (DX proportion + 1e‑6). Identify the cluster with the highest fold‑change (i.e., the greatest increase in relative abundance after treatment). Answer format: a single label string exactly matching the 'global_cluster' identifier (shape: label).\n",
      "Answer: 13\n",
      "\n",
      "Task 4\n",
      "Description: Using the provided single-nucleus RNA‑seq dataset of macrophages, compute the average expression of the canonical macrophage marker gene CD68 (Ensembl ID ENSG00000129226) across each distinct value of the 'Treatment' column (the expression matrix is already Harmony‑corrected). Return the exact Treatment label that shows the highest mean CD68 expression. Answer format: a single label (shape: label).\n",
      "Answer: CHP667, inc doxorubicin\n",
      "\n",
      "Task 5\n",
      "Description: From the provided snRNA‑seq dataset of macrophages in high‑risk neuroblastoma, cells have been assigned to Seurat clusters (seurat_clusters) and annotated with treatment response categories (Response: Complete, Partial, Progressive, Stable Disease). Identify the Seurat cluster (label as it appears in the seurat_clusters column) that shows the highest enrichment for the 'Complete' response relative to its overall frequency in the dataset. Enrichment is defined as (proportion of 'Complete' cells within the cluster) divided by (overall proportion of 'Complete' cells across all cells). Answer format: a single integer label (shape: label).\n",
      "Answer: 3\n",
      "\n",
      "Task 6\n",
      "Description: Among the macrophage cells in the dataset, first exclude any samples (orig.ident) that contain fewer than 50 cells. Then compute the median `phagocytosis_score1` for each remaining Treatment group. Which Treatment has the highest median `phagocytosis_score1`? Answer format: the exact Treatment string as it appears in the `Treatment` column (shape: label).\n",
      "Answer: ANBL 09P1\n",
      "\n",
      "Task 7\n",
      "Description: Identify the Treatment group (value in the 'Treatment' column) whose macrophage nuclei have the highest average phagocytosis_score1. Answer format: a single treatment label exactly as it appears in the Treatment column (shape: label).\n",
      "Answer: ANBL 09P1\n",
      "\n",
      "Task 8\n",
      "Description: Among the macrophage cells, which Treatment has the highest proportion of cells with Response == \"Complete\"? If multiple treatments share the maximal proportion, return the alphabetically first one. Answer format: a single treatment name (shape: label).\n",
      "Answer: ANBL0532-like therapy with cisplatin & etoposide\n",
      "\n",
      "Task 9\n",
      "Description: From the 'Treatment' column, identify the treatment that exhibits the highest fraction of cells labeled as 'Complete' in the 'Response' column, considering only treatments that have at least 5 cells. If multiple treatments share the highest fraction, select the one that comes first alphabetically. Answer format: a single label (exact treatment string) (shape: label).\n",
      "Answer: ANBL0532-like therapy with cisplatin & etoposide\n",
      "\n",
      "Task 10\n",
      "Description: Using the provided snRNA‑seq dataset of macrophages (dataset3.h5ad), determine which treatment group exhibits the highest average expression of the gene that is most up‑regulated between PTX and DX stages (the top gene from a Wilcoxon DE analysis). Report the treatment name exactly as it appears in the 'Treatment' column. Answer format: a single label (shape: label).\n",
      "Answer: primary tumor resection + laparotomy post 3 cycles of induction chemo per ANBL12P1\n",
      "\n",
      "Task 11\n",
      "Description: Identify the gene that exhibits the highest positive log2 fold-change when comparing post‑treatment (PTX) macrophage samples to diagnostic (DX) macrophage samples in the provided snRNA‑seq dataset. Answer format: a single HGNC gene symbol in uppercase (shape: gene_symbol).\n",
      "Answer: MT3\n",
      "\n",
      "Task 12\n",
      "Description: Identify the treatment (value in the 'Treatment' column) that contains the largest number of cells annotated as 'Complete' in the 'Response' column. Answer format: a single treatment name (shape: label).\n",
      "Answer: ANBL12P1-like\n",
      "\n",
      "Task 13\n",
      "Description: Using the provided AnnData object, identify the donor that exhibits the strongest residual batch effect within the most abundant macrophage sub‑cluster. Follow these exact steps: 1) Subset cells whose 'cell_type' annotation contains the word \"macrophage\" (case‑insensitive). 2) Determine the most abundant sub‑cluster using the column 'RNA_snn_res.0.2' (if absent, use 'seurat_clusters'); select the cluster with the highest cell count. 3) For cells in that cluster, compute the batch‑mixing LISI (Local Inverse Simpson Index) on the Harmony embedding stored in 'X_harmony' as follows: - Use Euclidean distance to find each cell's 30 nearest neighbors (exclude the cell itself). - For each cell, calculate the proportion of neighbors from each donor (column 'donor_id'), compute the Simpson index (sum of squared proportions), then take its inverse to obtain the LISI value for that cell. - Average LISI values across all cells belonging to each donor. 4) Report the donor ID (value of 'donor_id') with the lowest mean LISI (i.e., poorest mixing). Answer format: a single label (exact donor_id string).\n",
      "Answer: CH_18\n",
      "\n",
      "Dataset: /mlbio_scratch/videnovi/papers_BIBA/paper4/data/dataset1.h5ad\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Description: Using the provided single‑cell dataset, compute the proportion of each cell type within the Control group and within all disease groups combined (any Condition not equal to 'Control'). Calculate the log2 fold‑change of disease proportion over control proportion for each cell type (add a pseudocount of 1e-6 to avoid division by zero). Identify the cell type with the highest log2 fold‑change. Answer format: a single cell‑type label exactly as it appears in the 'cell_type' column (shape: label).\n",
      "Answer: lymphocyte\n",
      "\n",
      "Task 1\n",
      "Description: Compute the proportion of each cell type (from the 'cell_type' column) within each condition in the dataset. Then calculate the fold‑change of the proportion in the 'MIS-C' condition relative to the 'Control' condition (MIS-C proportion divided by Control proportion). Identify the cell type with the highest fold‑change. Answer format: a single cell type label exactly as it appears in the 'cell_type' column (case‑sensitive, shape: label).\n",
      "Answer: lymphocyte\n",
      "\n",
      "Task 2\n",
      "Description: Using the provided AnnData object, compute for each cell type the proportion of cells belonging to the disease 'myocarditis' and the proportion belonging to 'healthy'. Calculate the log2 fold‑change of these proportions (log2(prop_myocarditis / prop_healthy)) for each cell type. Identify the cell type with the highest positive log2 fold‑change (i.e., the strongest enrichment in myocarditis relative to healthy). Answer format: a single cell type label exactly as it appears in the 'cell_type' column (shape: label).\n",
      "Answer: lymphocyte\n",
      "\n",
      "Task 3\n",
      "Description: Identify the cell type that shows the largest average absolute difference in log‑transformed expression between the Control and Post‑COVID‑19 conditions. Use the raw UMI counts, apply log1p to non‑zero entries only, compute the mean log expression per gene for each condition within each cell type (requiring at least 20 cells in both conditions), then calculate the average of the absolute differences across all genes. Return the cell‑type name as a single lowercase label with underscores (shape: label).\n",
      "Answer: cardiac_endothelial_cell\n",
      "\n",
      "Task 4\n",
      "Description: Using the provided single‑cell RNA‑seq dataset, compute the proportion of each cell type (as given in the 'author_cell_type' column) within the Control and Post‑COVID‑19 conditions. Calculate the log2 fold‑change of these proportions (log2[(Post‑COVID‑19 proportion + 1e-6) / (Control proportion + 1e-6)]). Identify the cell type with the highest positive log2 fold‑change (i.e., the greatest increase in relative abundance in Post‑COVID‑19 compared to Control). Answer format: a single cell‑type label exactly as it appears in the 'author_cell_type' column (shape: label).\n",
      "Answer: PKHD1L1+ EC\n",
      "\n",
      "Task 5\n",
      "Description: Identify the cell type that exhibits the greatest up‑regulation of the chemokine gene CXCL10 when comparing MIS‑C (Multisystem Inflammatory Syndrome in Children) to Control samples. Use the mean expression difference (MIS‑C mean minus Control mean) calculated from the provided normalized expression matrix. Answer format: a single cell‑type label in lowercase with underscores (shape: label).\n",
      "Answer: cardiac_neuron\n",
      "\n",
      "Task 6\n",
      "Description: Using the provided scRNA‑seq dataset of human cardiac tissue, compare the MIS‑C condition to the Control condition within each annotated cell type. For each cell type, count the number of genes whose log2 fold‑change (MIS‑C vs Control) exceeds 1 (i.e., at least a two‑fold increase). Identify the cell type with the greatest such count. Answer format: a single cell‑type label exactly as it appears in the 'cell_type' column (shape: label).\n",
      "Answer: cardiac neuron\n",
      "\n",
      "Task 7\n",
      "Description: Identify the Leiden cluster that shows the strongest enrichment of the MIS-C condition relative to the Control condition, measured as the highest positive log2 fold‑change of the proportion of MIS-C cells over Control cells. Answer format: a single cluster label as a string (shape: label).\n",
      "Answer: 20\n",
      "\n",
      "Task 8\n",
      "Description: Using the precomputed differential expression results stored in adata.uns['rank_genes_groups'] (which contain the top 11 genes and their log2 fold‑change values for each Leiden cluster), determine the gene that exhibits the largest absolute log‑fold‑change (|logFC|) across all clusters. Return the gene symbol as a single HGNC uppercase identifier (shape: gene_symbol).\n",
      "Answer: CPA3\n",
      "\n",
      "Task 9\n",
      "Description: Identify the cell type (author-provided annotation in the 'cell_type' column) whose Leiden cluster exhibits the highest average log2 fold-change of its top-10 marker genes according to the pre-computed differential expression results stored in adata.uns['rank_genes_groups']. Answer format: a single lowercase label with underscores (shape: label).\n",
      "Answer: mural_cell\n",
      "\n",
      "Task 10\n",
      "Description: In the provided AnnData object (path: /mlbio_scratch/videnovi/papers_BIBA/paper4/data/dataset1.h5ad), differential expression results are stored in `adata.uns['rank_genes_groups']`. This object contains a field `names` (a list of tuples, each tuple holding the top 11 marker gene symbols for a DE cluster) and a structured array `scores` (with fields '0'–'10') that provides a numeric score for each corresponding gene in the tuple. For every gene that appears in any of the `names` tuples, compute the average of its scores across all clusters where it is present. Which gene has the highest average score? Answer format: a single HGNC gene symbol in uppercase (shape: gene_symbol).\n",
      "Answer: MYBPC3\n",
      "\n",
      "Task 11\n",
      "Description: Using the AnnData file at /mlbio_scratch/videnovi/papers_BIBA/paper4/data/dataset1.h5ad, consider only cells annotated as \"cardiac muscle cell\" in the \"cell_type\" column. From the gene metadata (adata.var), select the five genes with the highest normalized dispersion (the column \"dispersions_norm\"). For each disease condition (\"healthy\" and \"myocarditis\" as recorded in the \"disease\" column), compute the average of the log‑normalized expression values of these five genes across all selected cells. Which disease condition shows the higher average expression? Answer format: a single label matching exactly one of the values in the \"disease\" column (e.g., \"healthy\" or \"myocarditis\").\n",
      "Answer: myocarditis\n",
      "\n",
      "Task 12\n",
      "Description: Using the precomputed differential expression results stored in adata.uns['rank_genes_groups'] (groupby='leiden_02'), compute for each Leiden cluster (0‑10) the mean of the absolute log2‑fold‑change (logFC) of all genes that have an adjusted p‑value < 0.05. Identify the cluster with the highest mean absolute logFC. Answer format: a single label (the cluster identifier) with no spaces (shape: label).\n",
      "Answer: 6\n",
      "\n",
      "Dataset: /mlbio_scratch/videnovi/papers_BIBA/paper4/data/lymph_dataset.h5ad\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Description: Identify the cell type that constitutes the highest proportion of cells within the 'MIS-C' condition. Answer format: a single lowercase label with underscores (shape: label).\n",
      "Answer: natural_killer_cell\n",
      "\n",
      "Task 1\n",
      "Description: Within the natural killer cell population, identify the gene that shows the strongest statistically significant up‑regulation in the Post‑COVID‑19 condition compared to the Control condition. Perform differential expression using Scanpy's `rank_genes_groups` with the Wilcoxon test (default settings, considering all genes). Report the gene with the smallest adjusted p‑value; if multiple genes share this p‑value, choose the one with the highest log2‑fold‑change. Answer format: a single uppercase HGNC gene symbol (shape: gene_symbol).\n",
      "Answer: RAP1GAP2\n",
      "\n",
      "Task 2\n",
      "Description: Among cells annotated as cytotoxic T cell, which Condition shows the highest average expression of the gene CXCL9? Answer format: a single condition name exactly as it appears in the Condition column (shape: label).\n",
      "Answer: Control\n",
      "\n",
      "Task 3\n",
      "Description: Identify the disease condition that exhibits the highest proportion of the cell type whose proportion across conditions shows the greatest variance. Answer format: a single label exactly matching the condition name as it appears in the 'Condition' column (shape: label).\n",
      "Answer: Control\n",
      "\n",
      "Task 4\n",
      "Description: Within the B cell population, identify which experimental condition shows the highest average expression of the cytokine gene CXCL10. Provide the condition name exactly as it appears in the 'Condition' column (case‑sensitive). Answer format: a single label (shape: label).\n",
      "Answer: Control\n",
      "\n",
      "Task 5\n",
      "Description: Using the provided single-cell RNA‑seq dataset, compute the average expression of the gene CXCL10 for each cell type within the 'Post-COVID-19' and 'Control' conditions. Calculate the log2 fold‑change (log2[(mean_Post‑COVID‑19 + 1) / (mean_Control + 1)]) for each cell type. Report the cell type that exhibits the highest log2 fold‑change. Answer format: a single cell type label exactly as it appears in the 'cell_type' column of the AnnData object (shape: label).\n",
      "Answer: plasma cell\n",
      "\n",
      "Task 6\n",
      "Description: Determine which cell type (as given in the 'author_cell_type' annotation) has the highest proportion of cells from myocarditis samples (disease == 'myocarditis'). Answer format: a single cell‑type label exactly as it appears in the 'author_cell_type' column (shape: label).\n",
      "Answer: CD4T_cm\n",
      "\n",
      "Task 7\n",
      "Description: Identify the cell type whose proportion is most increased in myocarditis samples compared to control (normal) samples, using the provided AnnData object. Compute the proportion of each cell type within each disease group, calculate the log2 fold‑change (myocarditis / normal), and select the cell type with the highest positive log2 FC. Answer format: a single label exactly matching the 'cell_type' entry (shape: label).\n",
      "Answer: central memory CD4-positive, alpha-beta T cell\n",
      "\n",
      "Task 8\n",
      "Description: Within the effector memory CD8-positive, alpha-beta T cell population, compare cells from the 'Post-COVID-19' condition to those from the 'Control' condition. Perform a standard differential expression analysis (Wilcoxon rank‑sum test) on log‑normalized counts and identify the gene that is significantly up‑regulated in Post‑COVID‑19 (adjusted p‑value < 0.05) with the largest positive log2‑fold‑change. Answer format: a single HGNC gene symbol in uppercase (shape: gene_symbol).\n",
      "Answer: CXCL10\n",
      "\n",
      "Task 9\n",
      "Description: Identify the cell type that shows the greatest proportional enrichment in the Post‑COVID‑19 condition compared to the Control condition. Compute the proportion of each cell type within each condition, calculate the fold‑change (Post‑COVID‑19 proportion divided by Control proportion), and report the cell type with the highest fold‑change. Answer format: a single cell type label exactly as it appears in the 'cell_type' column (shape: label).\n",
      "Answer: cytotoxic T cell\n",
      "\n",
      "Task 10\n",
      "Description: Determine which Condition has the highest proportion of cells annotated as \"natural killer cell\" in the dataset. Answer format: a single condition label (shape: label).\n",
      "Answer: MIS-C\n",
      "\n",
      "Task 11\n",
      "Description: Using the provided single-cell RNA‑seq dataset of human cardiac lymphocytes, consider the cell type annotations in the column 'author_cell_type'. Define NK cells as those labeled 'NK_CD16hi' or 'NK_CD56hi', and define CD8+ T cells as any cell whose annotation starts with 'CD8T_'. For each disease condition (values in the 'Condition' column), compute the average expression of the gene IFNG (interferon‑γ) separately in NK cells and in CD8+ T cells. Then calculate the difference (average IFNG in NK cells minus average IFNG in CD8+ T cells) for each condition. Identify the condition with the largest positive difference (i.e., the strongest relative up‑regulation of IFNG in NK cells compared to CD8+ T cells). Answer format: a single label exactly matching the condition name as it appears in the 'Condition' column (shape: label).\n",
      "Answer: MIS-C\n",
      "\n",
      "Task 12\n",
      "Description: Within the cell type \"effector memory CD8-positive, alpha-beta T cell\", compute the average expression of each gene for the Post-COVID-19 and Control conditions (using the provided AnnData matrix). Calculate the log2 fold change as log2((mean_PostCOVID + 1e-6)/(mean_Control + 1e-6)). Identify the gene with the highest positive log2 fold change. Answer format: a single HGNC gene symbol (shape: gene_symbol).\n",
      "Answer: MCM8-AS1\n",
      "\n",
      "Task 13\n",
      "Description: Identify the cell type that exhibits the highest number of genes with adjusted p‑value < 0.05 when comparing the Post‑COVID‑19 condition versus the Control condition (using Wilcoxon rank‑sum test on raw counts). Answer format: a single label (shape: label) exactly as it appears in the 'cell_type' column.\n",
      "Answer: effector memory CD8-positive, alpha-beta T cell\n",
      "\n",
      "Task 14\n",
      "Description: Using the provided AnnData object, compute for each disease condition (excluding 'Control') the proportional representation of each cell type (using the 'author_cell_type' annotation) relative to the total cells in that condition. Then calculate the fold‑change of each cell‑type proportion versus the corresponding proportion in the 'Control' condition. Identify the condition‑cell_type pair with the highest fold‑change enrichment. Report the answer as a single string formatted as CONDITION::CELL_TYPE (shape: pair).\n",
      "Answer: MIS-C::T_prolif\n",
      "\n",
      "Dataset: /mlbio_scratch/videnovi/papers_BIBA/paper4/data/my_dataset.h5ad\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Description: Identify the cell type that is most enriched in the MIS-C condition relative to the Control condition, using an odds ratio calculated from a 2 × 2 contingency table (add a 0.5 pseudocount to each cell to avoid division by zero). Answer format: a single lowercase label with underscores (shape: label).\n",
      "Answer: monocyte\n",
      "\n",
      "Task 1\n",
      "Description: Using the provided AnnData object, calculate the proportion of cells labeled as 'MIS-C' within each coarse cell type found in the 'cell_type' column. Identify the cell type that has the highest proportion of MIS-C cells. Answer format: a single cell type label exactly as it appears in the 'cell_type' column (shape: label).\n",
      "Answer: monocyte\n",
      "\n",
      "Task 2\n",
      "Description: Using the provided AnnData object, calculate for each Condition the proportion of cells annotated as \"cycling macrophage\" (i.e., number of cycling macrophage cells divided by the total number of cells in that Condition). Identify the Condition with the highest proportion. Answer format: a single condition label exactly as it appears in the 'Condition' column (shape: label).\n",
      "Answer: Control\n",
      "\n",
      "Task 3\n",
      "Description: Within the macrophage population, identify the gene that shows the strongest up‑regulation in the MIS‑C condition compared to the Control condition after filtering for genes expressed in ≥10 % of cells in either group. Answer format: a single HGNC gene symbol (shape: gene_symbol).\n",
      "Answer: NAALADL2\n",
      "\n",
      "Task 4\n",
      "Description: Within the macrophage cell type, identify the gene that exhibits the highest positive log2 fold‑change when comparing myocarditis samples to normal controls. Answer format: a single HGNC gene symbol (shape: gene_symbol).\n",
      "Answer: CXCL9\n",
      "\n",
      "Task 5\n",
      "Description: Using the precomputed differential expression results stored in adata.uns['rank_genes_groups'], identify the gene among the top 100 DE genes that exhibits the largest positive log2‑fold‑change when comparing myocarditis vs normal samples. Answer format: a single HGNC gene symbol in uppercase (shape: gene_symbol).\n",
      "Answer: FP236383.1\n",
      "\n",
      "Task 6\n",
      "Description: Identify the gene with the highest absolute log2 fold change in the differential expression analysis comparing Leiden_12 cluster 0 (reference) versus all other clusters. Answer format: a single HGNC gene symbol (shape: gene_symbol).\n",
      "Answer: FP236383.1\n",
      "\n",
      "Task 7\n",
      "Description: Within the myeloid subpopulation 'MP_LYVE1hi/MHCII_lo', compute the average log1p‑normalized expression for each gene in the 'MIS-C' condition and in the 'Control' condition. Calculate the log2 fold‑change (log2FC = log2[(mean_MIS‑C + 1e-9)/(mean_Control + 1e-9)]). Identify the gene with the largest positive log2FC. Answer format: a single HGNC gene symbol (shape: gene_symbol).\n",
      "Answer: KCNE1\n",
      "\n",
      "Task 8\n",
      "Description: From the provided single-cell dataset, consider only cells annotated as 'macrophage' or 'cycling macrophage'. For each Condition, compute the proportion of cells that are labeled 'cycling macrophage' among these selected cells. Identify the Condition with the highest such proportion. Answer format: a single Condition label (shape: label).\n",
      "Answer: MIS-C\n",
      "\n",
      "Task 9\n",
      "Description: From the provided single‑cell dataset, compute the proportion of each myeloid cell type (column 'author_cell_type') within the MIS‑C condition and across all cells. Calculate the enrichment ratio for each cell type as (MIS‑C proportion) / (overall proportion), excluding cell types with fewer than 20 cells in the whole dataset. Identify the cell type with the highest enrichment ratio. Answer format: a single label (exact cell type name as it appears in 'author_cell_type').\n",
      "Answer: MO_CD16\n",
      "\n",
      "Task 10\n",
      "Description: Identify the cell type (as labeled in the 'author_cell_type' column) that exhibits the highest number of significantly up‑regulated genes (adjusted p‑value < 0.05 and log2 fold‑change > 0) when comparing myocarditis versus normal control samples. Answer format: a single label exactly matching the cell‑type string (shape: label).\n",
      "Answer: MP_LYVE1lo/MHCII_hi\n",
      "\n",
      "Dataset: /mlbio_scratch/videnovi/papers_BIBA/paper5/data/dataset2.h5ad\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Description: From the provided snRNA‑seq dataset of human visceral adipose tissue, compute for each donor (column 'donor_id') the proportion of nuclei annotated as \"macrophage\" in the 'cell_type' column. Identify the donor with the highest macrophage proportion. Answer format: a single donor identifier (shape: label).\n",
      "Answer: 3357\n",
      "\n",
      "Task 1\n",
      "Description: Using the provided snRNA‑seq AnnData object, compute a neighbourhood graph (e.g., Scanpy's default parameters: n_neighbors=15, n_pcs=30 after log‑normalization and HVG selection). For each cell, calculate an integration LISI (iLISI) score based on the 'donor_id' batch variable by measuring the Shannon entropy of donor labels within each cell's neighbourhood and converting it to exp(entropy). Then, average the iLISI scores for each cell type given in the 'cell_type' column. Identify the cell type with the lowest average iLISI (i.e., the most batch‑biased cell type). Answer format: a single cell‑type label (shape: label).\n",
      "Answer: B cell\n",
      "\n",
      "Task 2\n",
      "Description: Among all cell types annotated in the dataset (column 'cell_type'), which cell type exhibits the highest average expression of the mitochondrial gene MT-ND1? Answer format: a single lowercase label exactly as it appears in the 'cell_type' column (shape: label).\n",
      "Answer: endothelial cell\n",
      "\n",
      "Task 3\n",
      "Description: Using the provided AnnData object, compute the average mitochondrial transcript percentage (percent.mt) for each cell type listed in the 'cell_type' column of adata.obs. Report the cell type that has the highest mean percent.mt. Answer format: a single cell type label exactly as it appears in the 'cell_type' column (shape: label).\n",
      "Answer: endothelial cell\n",
      "\n",
      "Task 4\n",
      "Description: Using the provided snRNA‑seq dataset of human visceral adipose tissue, compute the average log‑normalized expression of the mature adipocyte marker genes ADIPOQ, PLIN1, FABP4 and LPL for each broad cell type annotation (the 'cell_type' column in .obs). Identify the cell type with the highest mean expression across these four markers. Return the cell type label exactly as it appears in the 'cell_type' column. Answer format: a single label (shape: label).\n",
      "Answer: adipocyte of omentum tissue\n",
      "\n",
      "Task 5\n",
      "Description: Identify the cell type that shows the highest average raw count expression of the adipogenic master regulator gene PPARG across all cells in the dataset. Report the answer as the exact cell type label found in the 'cell_type' column (case‑sensitive, including spaces). Answer format: a single label (shape: label).\n",
      "Answer: adipocyte of omentum tissue\n",
      "\n",
      "Task 6\n",
      "Description: Using the provided snRNA-seq dataset of human visceral adipose tissue, compute the average log‑CP10K expression of the gene PPARG within each annotated cell type (the 'cell_type' column). Normalize raw counts per cell to counts per 10 k, apply log1p, then calculate the mean of this transformed expression for each cell type. Return the cell type with the highest average expression. Answer format: a single cell‑type label exactly as it appears in adata.obs['cell_type'] (shape: label).\n",
      "Answer: adipocyte of omentum tissue\n",
      "\n",
      "Task 7\n",
      "Description: For each donor–cell_type combination in the dataset, compute the proportion of that donor's cells among all cells of the given cell_type (i.e., donor‑type count ÷ total count of that cell_type). Identify the donor‑cell_type pair with the highest proportion and report it as a pair formatted exactly as CELLTYPE::DONORID (shape: pair).\n",
      "Answer: B cell::3350\n",
      "\n",
      "Task 8\n",
      "Description: Identify the Seurat cluster that has the highest number of macrophage cells among clusters that consist exclusively of macrophages (i.e., 100% macrophage purity). Answer format: a single cluster identifier as a string (shape: label).\n",
      "Answer: 3\n",
      "\n",
      "Task 9\n",
      "Description: Identify the donor (donor_id) that contributes the highest proportion of cells annotated as \"mesothelial cell\" relative to its total number of cells in the dataset. Answer format: a single donor ID string (shape: label).\n",
      "Answer: 3313\n",
      "\n",
      "Task 10\n",
      "Description: Using the provided snRNA‑seq AnnData object, identify the cell type that has the highest overall abundance across all cells. Then, among all donors (column 'donor_id'), determine which donor exhibits the highest proportion of that cell type (proportion = cells of that type / total cells for the donor). Answer format: a single donor identifier string (shape: label).\n",
      "Answer: 3313\n",
      "\n",
      "Task 11\n",
      "Description: Identify the cell type that exhibits the greatest donor-to-donor variability in median RNA counts (nCount_RNA). For each cell type, compute the median nCount_RNA for each donor, then calculate the coefficient of variation (CV = standard deviation / mean) of those donor‑wise medians (use the population standard deviation, i.e., ddof=0). Report the cell type name exactly as it appears in the 'cell_type' column of the metadata. Answer format: a single label (case‑sensitive).\n",
      "Answer: T cell\n",
      "\n",
      "Dataset: /mlbio_scratch/videnovi/papers_BIBA/paper5/data/VA_adipocytes.h5ad\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Description: Using the provided single-nucleus RNA‑seq dataset of human visceral adipose tissue adipocytes, compute the average expression of each gene across all cells from male donors and across all cells from female donors. Calculate the log2 fold‑change (male vs female) for each gene, adding a pseudocount of 1e‑6 to avoid division by zero. Identify the gene with the highest positive log2 fold‑change. Answer format: a single gene identifier (uppercase, no spaces), shape: gene_symbol.\n",
      "Answer: ENSG00000286009\n",
      "\n",
      "Task 1\n",
      "Description: Identify the donor (donor_id) that exhibits the highest average expression of the adipocyte marker gene ADIPOQ across all its cells in the provided dataset. Answer format: a single donor identifier (shape: label).\n",
      "Answer: 3427\n",
      "\n",
      "Task 2\n",
      "Description: Using the provided UMAP embedding (X_umap) for each cell, compute the variance of the UMAP1 and UMAP2 coordinates separately for each donor (column 'donor_id' in .obs). Then calculate the average of these two variances for each donor. Identify the donor with the highest average variance (i.e., the most dispersed cells in the UMAP space). Answer format: a single donor identifier as a label (shape: label).\n",
      "Answer: 3427\n",
      "\n",
      "Task 3\n",
      "Description: Perform the following analysis on the AnnData file /mlbio_scratch/videnovi/papers_BIBA/paper5/data/VA_adipocytes.h5ad: (1) normalize raw counts to 10,000 per cell and log1p transform; (2) select the top 2,000 highly variable genes (Seurat flavor); (3) scale the data (max value 10) and compute PCA with 20 components; (4) for each of the first 10 principal components, calculate the proportion of variance explained (PVE) by the donor identifier (donor_id) as the between‑donor sum of squares divided by the total sum of squares; (5) identify the maximum PVE among PCs 1‑10 and report it as a percentage rounded to two decimal places. Answer format: a single numeric value (shape: number).\n",
      "Answer: 69.51\n",
      "\n",
      "Task 4\n",
      "Description: From the provided single-nucleus RNA‑seq dataset of human visceral adipose tissue, identify the Seurat cluster (column 'seurat_clusters' in .obs) that exhibits the highest median mitochondrial gene expression percentage (column 'percent.mt'). Report the cluster identifier exactly as it appears (shape: label).\n",
      "Answer: 9\n",
      "\n",
      "Task 5\n",
      "Description: Using the AnnData file at /mlbio_scratch/videnovi/papers_BIBA/paper5/data/VA_adipocytes.h5ad, examine the observation columns `seurat_clusters` (integer cluster IDs) and `donor_id`. For each cluster, count how many distinct donors are present. Identify the cluster(s) with the fewest distinct donors. If multiple clusters tie for this minimum, choose the one whose most frequent donor constitutes the highest proportion of cells in that cluster. Provide the chosen cluster label as a single integer (shape: label).\n",
      "Answer: 13\n",
      "\n",
      "Task 6\n",
      "Description: Using the AnnData file /mlbio_scratch/videnovi/papers_BIBA/paper5/data/VA_adipocytes.h5ad, perform the following analysis: (1) filter out cells with <500 total counts and genes expressed in <10 cells; (2) normalize each cell to 10,000 counts and log1p-transform; (3) select the 2,000 most highly variable genes (Seurat v3 method); (4) compute 30 principal components and build a 30‑nearest‑neighbor graph (using these PCs); (5) for each cell, count how many distinct donor IDs (column 'donor_id') are present among its 30 nearest neighbors; (6) for each annotated cell type (column 'cell_type'), calculate the average of these unique‑donor counts; (7) identify the cell type with the lowest average unique‑donor neighbor count (i.e., poorest donor mixing). Answer format: a single label string exactly matching the entry in the 'cell_type' column (case‑sensitive).\n",
      "Answer: adipocyte of omentum tissue\n",
      "\n",
      "Task 7\n",
      "Description: Using the provided snRNA‑seq dataset of human visceral adipose tissue adipocytes, compute log‑normalized expression (total count normalized to 10,000 and log1p transformed) for the adipocyte marker gene ADIPOQ (gene symbol). Then calculate the average ADIPOQ expression for each Seurat cluster (as given in the 'seurat_clusters' column). Identify the cluster with the highest average ADIPOQ expression. Answer format: a single cluster identifier string (shape: label).\n",
      "Answer: 1\n",
      "\n",
      "Task 8\n",
      "Description: Using the provided UMAP embedding (adata.obsm['X_umap']) for the visceral adipocyte dataset, compute the silhouette score for each cell based on the 'donor_id' batch labels. Then calculate the mean silhouette score for each donor. Identify the donor whose mean silhouette score is the lowest (i.e., shows the strongest residual batch mixing). Answer format: a single donor ID as a label (shape: label).\n",
      "Answer: 3384\n",
      "\n",
      "Task 9\n",
      "Description: Identify the adipocyte sub‑cluster (author_cell_type) that has the highest average expression of the adipocyte marker gene PLIN1 across all cells in the dataset. Answer format: a single label matching one of the author_cell_type values (e.g., VA8).\n",
      "Answer: VA8\n",
      "\n",
      "Task 10\n",
      "Description: In the provided snRNA‑seq dataset of human visceral adipose tissue, the column 'author_cell_type' defines eight adipocyte sub‑clusters (VA1‑VA8). Compute the average log‑normalized expression (log1p of raw counts) of the adipocyte marker gene ADIPOQ for each sub‑cluster and identify which sub‑cluster has the highest mean expression. Answer format: a single cluster label (shape: label, exact case as in the data).\n",
      "Answer: VA1\n",
      "\n",
      "Task 11\n",
      "Description: Using the provided AnnData object, compute the proportion of each Seurat cluster (column 'seurat_clusters') within each donor (column 'donor_id'). Identify the cluster that exhibits the largest range of proportions across donors. Then report the donor ID that has the highest proportion of that cluster. Answer format: a single donor ID as a string (shape: label).\n",
      "Answer: 3382\n",
      "\n",
      "Task 12\n",
      "Description: After normalizing the raw counts to counts-per-10,000 and applying a log1p transformation, compute the mean expression of the adipocyte marker gene PLIN1 for each donor (column 'donor_id' in adata.obs). Return the donor identifier (exact string as it appears in 'donor_id') that has the highest average PLIN1 expression. Answer format: a single label (e.g., 3431).\n",
      "Answer: 3431\n",
      "\n",
      "Task 13\n",
      "Description: Using the provided snRNA‑seq data of human visceral adipose tissue adipocytes, compute log‑normalized expression (library‑size scaling to 10,000 reads per cell followed by log1p). Then calculate the Pearson correlation between each gene’s expression and the donors’ BMI (column 'donor_BMI_at_collection'), using only cells with a non‑missing BMI value. Report the HGNC gene symbol of the gene with the strongest positive correlation (i.e., highest Pearson r). Answer format: a single uppercase gene symbol (shape: gene_symbol).\n",
      "Answer: ANXA1\n",
      "\n",
      "Task 14\n",
      "Description: Identify the cluster (among the provided 'seurat_clusters') that exhibits the highest average expression of the adipocyte marker gene ADIPOQ. Within that ADIPOQ‑high cluster, determine which gene (excluding ADIPOQ itself) has the highest mean expression across all nuclei. Respond with the gene's HGNC symbol (shape: gene_symbol, uppercase, no spaces).\n",
      "Answer: MALAT1\n",
      "\n",
      "Task 15\n",
      "Description: Using the provided AnnData object, compute for each Seurat cluster (column 'seurat_clusters') the proportion of cells contributed by each donor (column 'donor_id'), then divide this by the donor's overall proportion of cells in the whole dataset to obtain an enrichment ratio. Identify the donor‑cluster pair with the highest enrichment ratio. Report the answer as a pair formatted as donor_id::cluster (shape: pair).\n",
      "Answer: 3350::9\n",
      "\n",
      "Dataset: /mlbio_scratch/videnovi/papers_BIBA/paper6/data/dataset1.h5ad\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Description: For each distinct value in the 'Subclass' column of the provided AnnData object, compute the Spearman correlation between the RNA UMI count per cell (obs['nCount_RNA']) and the developmental age (obs['Estimated_postconceptional_age_in_days']). Identify the Subclass that exhibits the strongest positive monotonic relationship (i.e., the highest Spearman rho value greater than zero). Return the name of that Subclass exactly as it appears in the 'Subclass' column. Answer format: a single label (shape: label).\n",
      "Answer: GABAergic neuron\n",
      "\n",
      "Task 1\n",
      "Description: Using the provided snMultiome dataset, determine which cell type exhibits the largest increase in average number of detected genes (nFeature_RNA) between the earliest and latest postconceptional ages. Report the cell type name exactly as it appears in the 'cell_type' column. Answer format: a single label (shape: label).\n",
      "Answer: L5 extratelencephalic projecting glutamatergic cortical neuron\n",
      "\n",
      "Task 2\n",
      "Description: Within the Immune cell population of the provided snMultiome dataset, identify the gene (HGNC symbol, uppercase) that exhibits the strongest positive monotonic relationship with post‑conceptional age (days) after controlling for multiple testing using the Benjamini‑Hochberg procedure (FDR < 0.05). Report the gene symbol as a single token (shape: gene_symbol).\n",
      "Answer: ARHGAP24\n",
      "\n",
      "Task 3\n",
      "Description: In the provided snMultiome dataset, restrict to cells where the 'Class' annotation equals 'Neuron'. For each gene, compute the Spearman correlation between its expression values (across these neuronal cells) and the numeric variable 'Estimated_postconceptional_age_in_days'. Identify the gene with the highest positive correlation (i.e., the strongest monotonic increase with age). Return the gene symbol as a single uppercase HGNC gene symbol (shape: gene_symbol).\n",
      "Answer: GRIN2A\n",
      "\n",
      "Task 4\n",
      "Description: Identify the neuronal gene whose expression has the strongest positive Spearman correlation with the estimated post‑conceptional age (in days) across all cells labeled as Class = 'Neuron', after applying Benjamini‑Hochberg FDR correction (α = 0.05). Provide the answer as a single HGNC gene symbol (shape: gene_symbol).\n",
      "Answer: GRIN2A\n",
      "\n",
      "Task 5\n",
      "Description: Among cell types annotated in the 'Type' column that have at least 200 cells, compute the Pearson correlation between Estimated_postconceptional_age_in_days and ATAC_fragments_in_peaks for each type. Identify the cell type with the highest positive correlation. Answer format: a single label exactly as it appears in the 'Type' column (shape: label).\n",
      "Answer: IN-MGE-PV\n",
      "\n",
      "Task 6\n",
      "Description: From the provided snMultiome dataset of the developing human neocortex, consider only cell types (column 'Type' in adata.obs) that contain at least 200 cells. Compute the average total RNA counts per cell (column 'nCount_RNA') for each of these cell types and identify the cell type with the highest average. Answer format: a single cell‑type label (shape: label).\n",
      "Answer: EN-L2_3-IT\n",
      "\n",
      "Task 7\n",
      "Description: Among the six broad cell classes (Class) present in the dataset, identify the class that has the highest median value of the ATAC QC metric 'ATAC_fragments_in_peaks'. Answer format: a single label exactly matching one of the class names as they appear in adata.obs['Class'] (shape: label).\n",
      "Answer: Progenitor\n",
      "\n",
      "Task 8\n",
      "Description: Considering only cells with Class == 'Progenitor', which developmental Group (First_trimester, Second_trimester, Third_trimester, Infancy, Adolescence) has the highest proportion of cells annotated as Subclass == 'Radial glia'? Answer format: a single lowercase token from the set {first_trimester, second_trimester, third_trimester, infancy, adolescence} (shape: label).\n",
      "Answer: first_trimester\n",
      "\n",
      "Task 9\n",
      "Description: Using the provided snMultiome dataset of the developing human neocortex, compute for each fine‑grained cell type (the 'Type' annotation) the proportion of cells belonging to that type within each Sample_ID. Then calculate the Pearson correlation between each cell type's proportion across samples and the corresponding Estimated_postconceptional_age_in_days (numeric age). Identify the cell type whose proportion shows the strongest positive correlation with age (i.e., the highest positive Pearson correlation coefficient). Answer format: a single cell type label exactly as it appears in the 'Type' column (shape: label).\n",
      "Answer: Oligodendrocyte\n",
      "\n",
      "Task 10\n",
      "Description: After removing cells with a Scrublet doublet score > 0.5, compute the median RNA UMI count (nCount_RNA) for each distinct cell class (the values in adata.obs['Class']). Which class has the highest median? Answer format: a single label (exact string as it appears in the Class column).\n",
      "Answer: Neuron\n",
      "\n",
      "Task 11\n",
      "Description: Using the provided snMultiome dataset of the developing human neocortex, restrict analysis to cells with the maximum value of 'Estimated_postconceptional_age_in_days'. For the gene NEUROD1, compute the mean expression (raw counts) within each distinct 'cell_type' annotation among these cells. Return the cell_type that has the highest mean NEUROD1 expression. Answer format: a single label exactly as it appears in the 'cell_type' column (case‑sensitive, spaces allowed).\n",
      "Answer: L4 intratelencephalic projecting glutamatergic neuron\n",
      "\n",
      "Task 12\n",
      "Description: Identify the gene–subclass pair that shows the strongest positive monotonic association with developmental age (Estimated_postconceptional_age_in_days) across cells. For each Subclass, compute the Spearman correlation between the expression of each gene and the age vector for cells belonging to that Subclass. Return the pair with the highest positive correlation, formatted as GENE_SYMBOL::Subclass (gene symbol in uppercase, Subclass exactly as listed in the 'Subclass' column). Answer shape: pair.\n",
      "Answer: NRGN::Glutamatergic neuron\n",
      "\n",
      "Task 13\n",
      "Description: For each donor in the dataset, compute the proportion of cells labeled as \"EN-L5-ET\" (i.e., the number of cells with Type == EN-L5-ET divided by the total number of cells for that donor). Consider only donors that have at least 3,000 cells in total. Identify the donor with the highest such proportion and report the result as a pair formatted as DONOR_ID::PROPORTION, where PROPORTION is rounded to four decimal places (e.g., ARK35::0.0314). Answer shape: pair.\n",
      "Answer: ARK35::0.0314\n",
      "\n",
      "Task 14\n",
      "Description: Identify the cell type (as labeled in the 'Type_updated' column of the provided AnnData object) that exhibits the strongest positive Pearson correlation between developmental age (Estimated_postconceptional_age_in_days) and the expression of the neurodevelopmental marker gene SOX2. Answer format: a single label exactly matching the cell type name (shape: label).\n",
      "Answer: Unknown\n",
      "\n",
      "Task 15\n",
      "Description: Using the provided snMultiome dataset of the developing human neocortex, consider the cell type annotation column 'Class'. For each post‑conceptional age (rounded to the nearest whole day) compute the enrichment of the class \"Neuron\" relative to all other ages using a 2 × 2 contingency table with Haldane‑Anscombe (+0.5) correction and report the log₂ odds ratio. Identify the age (in days) that shows the strongest enrichment (i.e., the highest log₂ odds ratio). Answer format: a single integer (shape: number).\n",
      "Answer: 147\n",
      "\n",
      "Dataset: /mlbio_scratch/videnovi/papers_BIBA/paper7/data/dataset2.h5ad\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Description: Within the 'Mes1' cell type, compare gene expression between cells from the 'Distal' region and cells from the 'Proximal' region. Identify the gene that shows the strongest positive log2 fold-change (log2FC) when Distal expression is divided by Proximal expression (add a pseudocount of 1e-6 to avoid division by zero). Provide the answer as a single HGNC gene symbol (shape: gene_symbol, uppercase).\n",
      "Answer: HOXD13\n",
      "\n",
      "Task 1\n",
      "Description: Using the provided scRNA‑seq AnnData object, locate the gene HAND2 and compute the average HAND2 expression for each combination of cell type (adata.obs['celltype']) and limb region (adata.obs['region']). Identify the (cell type, region) pair with the highest mean HAND2 expression. Return the answer as a pair formatted exactly as CELLTYPE::REGION (no spaces).\n",
      "Answer: ChondroProg::WholeLimb\n",
      "\n",
      "Task 2\n",
      "Description: Identify the cell type whose donor composition is most balanced, defined as the cell type with the smallest maximum donor fraction across all donors. Answer format: a single cell type label (shape: label).\n",
      "Answer: Macro\n",
      "\n",
      "Task 3\n",
      "Description: Using the provided AnnData object, compute for each anatomical region (the `region` column) the proportion of cells annotated as `Mes1` in the `celltype` column. Report the region that has the highest such proportion. Answer format: a single region name exactly as it appears in the `region` column (shape: label).\n",
      "Answer: WholeLimb\n",
      "\n",
      "Task 4\n",
      "Description: Using the provided scRNA‑seq AnnData object, compute the proportion of each cell type (column 'cell_type') within each tissue ('tissue' = 'forelimb' or 'hindlimb'), calculate the log2 fold‑change of the forelimb proportion over the hindlimb proportion, and report the cell type with the highest positive log2 fold‑change. Answer format: a single cell‑type label exactly as it appears in the 'cell_type' column (shape: label).\n",
      "Answer: primitive red blood cell\n",
      "\n",
      "Task 5\n",
      "Description: Identify the cell type that exhibits the highest enrichment in the Distal region relative to the WholeLimb region, defined as the ratio of its proportion of cells in Distal to its proportion in WholeLimb (if the proportion in WholeLimb is zero, treat the ratio as 0). Answer format: a single cell type name matching the exact label in the 'celltype' column (shape: celltype).\n",
      "Answer: Mes3\n",
      "\n",
      "Task 6\n",
      "Description: Using the differential expression results stored in adata.uns['rank_genes_groups'], identify the gene with the highest log2 fold‑change among the top 100 marker genes for the comparison group labeled \"0,1\". Answer format: a single HGNC gene symbol (shape: gene_symbol).\n",
      "Answer: HOXA10\n",
      "\n",
      "Task 7\n",
      "Description: In the provided scRNA‑seq dataset, compute the proportion of each cell type within each donor. Identify the cell type whose donor‑wise proportion variance is the greatest. Then, among all donors, find the donor that has the highest proportion of that cell type. Report the donor identifier as a single label (no spaces, case‑sensitive).\n",
      "Answer: 5386STDY7537944\n",
      "\n",
      "Task 8\n",
      "Description: Using the provided scRNA‑seq dataset, identify the cell type that exhibits the highest average expression of the top marker gene (MAB21L2) reported in the pre‑computed differential expression results. Answer format: a single lowercase cell type label (shape: label).\n",
      "Answer: neuron\n",
      "\n",
      "Task 9\n",
      "Description: Using the provided scRNA‑seq dataset, compare the earliest developmental stage (adj_stage = Pcw5.1) to the latest stage (adj_stage = Pcw9.3). For each stage, compute the proportion of cells belonging to each cell type (as given in the 'celltype' column). Identify the cell type whose proportion increase (latest stage proportion minus earliest stage proportion) is the greatest. Answer format: a single cell type label exactly as it appears in the 'celltype' column (shape: label).\n",
      "Answer: ADH+Fibro\n",
      "\n",
      "Task 10\n",
      "Description: Using the provided scRNA-seq AnnData object, compute for each annotated cell type the proportion of its cells that originate from the 'Distal' region (i.e., number of cells with region == 'Distal' divided by the total cells of that cell type). Identify the cell type with the highest such proportion. If multiple cell types share the exact maximal proportion, break ties by alphabetical order. Answer format: a single cell‑type label exactly as it appears in the 'celltype' column (shape: label).\n",
      "Answer: RDH10+DistalMes\n",
      "\n",
      "Task 11\n",
      "Description: Identify the cell type that constitutes the highest proportion of cells within the 'Distal' region of the dataset. Answer format: a single cell type label exactly as it appears in the 'celltype' column (shape: label).\n",
      "Answer: TransMes\n",
      "\n",
      "Task 12\n",
      "Description: For each combination of region and cell type in the dataset, compute the proportion of cells belonging to that cell type within the region (i.e., count(celltype in region) / total cells in region). Identify the region–cell type pair with the highest proportion. Return the answer as a pair formatted as CELLTYPE::REGION (shape: pair).\n",
      "Answer: Mes1::WholeLimb\n",
      "\n",
      "Task 13\n",
      "Description: Using the provided AnnData file, compute the median `percent_mito` value for each donor (`donor_id`). Report the donor ID that has the highest median `percent_mito`. Answer format: a single donor identifier string exactly as it appears in the `donor_id` column (shape: label).\n",
      "Answer: 5478STDY7935101\n",
      "\n",
      "Task 14\n",
      "Description: Using the provided single‑cell RNA‑seq AnnData object, compute the variance of each gene across all cells and select the five genes with the highest variance (Ensembl IDs). For each of these five genes, calculate the mean expression within each region (the \"region\" column in adata.obs). Then, for each region, average the five mean expression values to obtain a region‑level score. Report the region name (exact string as it appears in the \"region\" column) that attains the highest score. Answer format: a single label (e.g., Thigh).\n",
      "Answer: Thigh\n",
      "\n",
      "Dataset: /mlbio_scratch/panigrah/agents/data/nature/paper1/cxg_data.h5ad\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Description: Among the cells belonging to the lineage 'ILCs-T.cells' in the provided dataset, identify the cell type label (column 'final_annotation') that shows the highest average raw expression of the gene IFNG. Answer format: a single label (shape: label), exactly as it appears in the 'final_annotation' column.\n",
      "Answer: CD4.memory\n",
      "\n",
      "Task 1\n",
      "Description: Using the PCA embedding (adata.obsm['X_pca']) compute, for each B-cell subpopulation listed in the 'final_annotation' column (i.e., entries containing \"Bcell\" or \"Bcells\"), the average Euclidean distance of its cells to the centroid of that subpopulation in PCA space. Identify the subpopulation with the largest average distance. Answer format: a single label exactly as it appears in 'final_annotation' (shape: label).\n",
      "Answer: Naive.Bcells.ISGhi\n",
      "\n",
      "Task 2\n",
      "Description: Identify the cell type (as listed in the 'final_annotation' column of the AnnData object) that has the highest average expression of the gene Cd14 across all cells. Provide the cell‑type label exactly as it appears (case‑sensitive). Answer format: a single label (shape: label).\n",
      "Answer: Mono.Il1b+\n",
      "\n",
      "Task 3\n",
      "Description: Within the effector regulatory T‑cell subset (annotation = \"eTregs\"), compute the average expression of the gene Cd69 for each tissue (mesenteric lymph node, caecum, small intestine Peyer's patch, lamina propria of small intestine). Return the tissue name that shows the highest mean Cd69 expression. Answer format: a single tissue label (shape: label).\n",
      "Answer: caecum\n",
      "\n",
      "Task 4\n",
      "Description: Among the four tissues represented in the dataset (mesenteric lymph node, caecum, lamina propria of small intestine, small intestine Peyer's patch), which tissue has the highest proportion of cells annotated as Naive.CD4+_CP? Answer format: a single lowercase tissue name (shape: label).\n",
      "Answer: caecum\n",
      "\n",
      "Task 5\n",
      "Description: Calculate the proportion of cells annotated as 'cTregs' within each tissue (mesenteric lymph node, caecum, lamina propria of small intestine, small intestine Peyer's patch). Report the tissue that has the highest cTregs proportion. Answer format: the exact tissue name as it appears in the 'tissue' column (shape: label).\n",
      "Answer: mesenteric lymph node\n",
      "\n",
      "Task 6\n",
      "Description: Using the provided scRNA‑seq dataset, compute for each cell type (column 'final_annotation') the proportion of cells within each tissue (column 'tissue') and the overall proportion across all tissues. For each cell type, calculate the enrichment ratio = (tissue proportion) / (global proportion) for every tissue, then take the maximum enrichment across tissues. Identify the cell type with the highest maximum enrichment ratio. Answer format: a single label matching the exact entry in the 'final_annotation' column (shape: label).\n",
      "Answer: ILC3s.Rorgt+\n",
      "\n",
      "Task 7\n",
      "Description: Using the provided scRNA‑seq dataset, perform differential expression between cells annotated as \"cTregs\" (test group) and cells annotated as \"GC.BC_LZ-resting\" (reference group). Filter out mitochondrial genes, normalize the data, select highly variable genes, and run a Wilcoxon rank‑sum test with Benjamini‑Hochberg correction. Identify the gene with the highest positive average log2 fold‑change (logFC) in the cTregs group among genes with adjusted p‑value < 0.05. Report the gene symbol (HGNC) of this top up‑regulated gene. Answer format: a single gene symbol (shape: gene_symbol).\n",
      "Answer: Cd3g\n",
      "\n",
      "Task 8\n",
      "Description: Using the provided scRNA‑seq dataset, compute for each cell type the proportion of cells that belong to that type within each tissue, and the overall proportion across all tissues. Then calculate the enrichment ratio (tissue proportion ÷ overall proportion) for each cell type in the caecum. Identify the cell type with the highest enrichment ratio in the caecum. Answer format: a single lowercase label (shape: label).\n",
      "Answer: b cell zone reticular cell\n",
      "\n",
      "Task 9\n",
      "Description: Within the macrophage cluster labeled \"Mac.Il1b+\", determine which gene exhibits the highest positive log2 fold change when comparing cells from the caecum to cells from the mesenteric lymph node. Report the answer as a single gene symbol (shape: gene_symbol).\n",
      "Answer: Zfp36\n",
      "\n",
      "Task 10\n",
      "Description: Using the provided scRNA‑seq dataset, compute for each tissue the proportion of cells annotated as 'regulatory T cell' (number of regulatory T cells divided by total cells in that tissue). Identify the tissue with the highest proportion. Answer format: a single tissue label exactly as it appears in the 'tissue' column (shape: label).\n",
      "Answer: small intestine Peyer's patch\n",
      "\n",
      "Task 11\n",
      "Description: Using the provided scRNA‑seq dataset, identify regulatory T cells (Tregs) by selecting cells whose 'final_annotation' contains the substring \"Treg\" (case‑insensitive) or whose 'cell_type' equals \"regulatory T cell\". For each tissue (values in the 'tissue' column), compute the proportion of Treg cells among all cells in that tissue. Report the tissue with the highest Treg proportion. Answer format: a single tissue label exactly as it appears in the 'tissue' column (shape: label).\n",
      "Answer: small intestine Peyer's patch\n",
      "\n",
      "Task 12\n",
      "Description: Determine the gene that exhibits the highest variance of expression across all cells in the dataset. Then, among the values in the 'final_annotation' column, identify which cell type label has the highest average expression of that gene. Provide the answer as a pair formatted as GENE_SYMBOL::CELL_TYPE_LABEL (no spaces).\n",
      "Answer: Cd74::DC.Il1b_high.Cd103+\n",
      "\n",
      "Dataset: /mlbio_scratch/panigrah/agents/data/nature/paper2/processed_MuSC_data.h5ad\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Description: Among the MuSC cell subtypes (annotation_level2 values: MYOG+MuSCs, MuSC, TNFRSF12A+MuSCs, ICAM1+MuSCs), identify the Age_group that has the highest average expression of the muscle stem cell marker gene Myod1. Answer format: a single age-group label exactly as it appears in the Age_group column (shape: label).\n",
      "Answer: 15-20\n",
      "\n",
      "Task 1\n",
      "Description: Within the ICAM1+MuSCs sub-population of the provided mouse muscle stem cell dataset, compare cells from the oldest age group (70-75) versus the youngest age group (15-20). Perform a Wilcoxon rank‑sum differential expression test on the expression matrix stored in the AnnData object, compute log2 fold‑changes, and adjust p‑values using the Benjamini‑Hochberg method. Among genes with adjusted p‑value < 0.05, identify the gene with the greatest absolute log2 fold‑change. Report the gene symbol as a single uppercase HGNC gene symbol (shape: gene_symbol).\n",
      "Answer: XIST\n",
      "\n",
      "Task 2\n",
      "Description: Within the dataset, select cells annotated as \"MuSC\" (annotation_level2 == \"MuSC\"). From these, keep only the two extreme age groups: 15-20 (young) and 70-75 (old). Normalize each cell to 10,000 counts, apply log1p transformation, and then perform differential expression using Scanpy's rank_genes_groups with groupby='Age_group', groups=['70-75'], reference='15-20', method='t-test_overestim_var'. Identify the gene with the largest positive log2 fold‑change (old vs young). Provide the gene symbol as a single uppercase string (shape: gene_symbol).\n",
      "Answer: XIST\n",
      "\n",
      "Task 3\n",
      "Description: Identify the donor (DonorID) that exhibits the highest fraction of cells annotated as 'MYOG+MuSCs' among all cells from that donor. Report the DonorID exactly as it appears (e.g., '464C'). Answer format: a single label (shape: label).\n",
      "Answer: 464C\n",
      "\n",
      "Task 4\n",
      "Description: After filtering cells to retain only those with percent_mito < 10 and scrublet_score < 0.2, compute the proportion of cells annotated as MYOG+MuSCs within each Age_group (column 'Age_group' in adata.obs). Which Age_group has the highest proportion? Provide the Age_group label exactly as it appears in the data (shape: label).\n",
      "Answer: 35-40\n",
      "\n",
      "Task 5\n",
      "Description: Among the MuSC sub‑states defined in the 'annotation_level2' column, identify the sub‑state that has the highest proportion of cells belonging to the 'old' age bin. Return the exact sub‑state label as it appears in the 'annotation_level2' column. Answer format: a single label (shape: label).\n",
      "Answer: TNFRSF12A+MuSCs\n",
      "\n",
      "Task 6\n",
      "Description: Within the MuSC dataset, cells are annotated with sub‑type labels in the column 'annotation_level2' and belong to two age categories: 'young' and 'old' (as indicated by the 'Age_bin' column). Compute the proportion of each sub‑type within each age category, then calculate the log2 fold‑change of the old proportion over the young proportion (add a negligible pseudocount to avoid division by zero). Identify the sub‑type that has the highest positive log2 fold‑change (i.e., is most enriched in old compared to young). Answer format: a single label exactly matching the value in 'annotation_level2' (case‑sensitive).\n",
      "Answer: TNFRSF12A+MuSCs\n",
      "\n",
      "Task 7\n",
      "Description: Determine which Age_bin ('young' or 'old') shows the higher proportion of cells annotated as ICAM1+MuSCs among all MuSCs. Answer format: a single lowercase label (shape: label).\n",
      "Answer: young\n",
      "\n",
      "Task 8\n",
      "Description: Identify the cell type (annotation_level2) that has the highest average mitochondrial read percentage (percent_mito) across all cells. Return the cell‑type name as a single lowercase label, converting any '+' to '_plus_' and removing spaces (shape: label).\n",
      "Answer: tnfrsf12a_plus_muscs\n",
      "\n",
      "Task 9\n",
      "Description: Using the AnnData file /mlbio_scratch/panigrah/agents/data/nature/paper2/processed_MuSC_data.h5ad, identify the gene that shows the largest absolute log2 fold‑change (old vs. young) when comparing the youngest and oldest age bins that each contain at least 20 cells within any cell type (annotation_level2). Compute mean expression per gene for the youngest and oldest age bins, calculate log2FC = log2(mean_old / mean_young) (add a tiny pseudocount if needed), take the absolute value, and find the gene with the maximum across all qualifying cell types. Report the answer as a single uppercase gene symbol (shape: gene_symbol).\n",
      "Answer: RPS4Y1\n",
      "\n",
      "Task 10\n",
      "Description: Identify the gene that exhibits the highest positive log2 fold change in the 'MYOG+MuSCs' sub‑population (annotation_level2) compared to all other MuSC sub‑populations in the provided dataset. Answer format: a single HGNC gene symbol (shape: gene_symbol, uppercase).\n",
      "Answer: MYEOV\n",
      "\n",
      "Task 11\n",
      "Description: Using the provided MuSC single‑cell dataset, perform Leiden clustering on the scVI latent space (resolution 0.5, 15 nearest neighbors, Euclidean metric). Identify the cluster that has the highest proportion of cells labeled as \"old\" in the \"Age_bin\" column. Then conduct differential expression (Wilcoxon rank‑sum test) between cells in this old‑enriched cluster and all other cells, using raw counts. Among genes with adjusted p‑value < 0.05 and positive log2 fold‑change, report the gene with the largest log2 fold‑change. Answer format: a single HGNC gene symbol (shape: gene_symbol).\n",
      "Answer: CD36\n",
      "\n",
      "Task 12\n",
      "Description: Using the provided AnnData object, compute the mean expression of each gene within each distinct 'annotation_level2' sub‑population. For each gene, calculate the standard deviation of these group means across all sub‑populations. Identify the gene with the highest standard deviation (i.e., the greatest variability of mean expression between sub‑populations). Answer format: a single uppercase HGNC gene symbol (shape: gene_symbol).\n",
      "Answer: FOS\n",
      "\n",
      "Task 13\n",
      "Description: Within the dataset, identify the annotation_level2 cell type that contains the most cells. Restrict analysis to cells of this cell type belonging to the two most frequent Age_group categories. Using the raw count matrix, compute the mean expression of each gene in each of the two age groups, add a pseudocount of 1 to each mean, and calculate the log2 fold change (log2((mean_group1+1)/(mean_group2+1))). Report the gene symbol (uppercase) of the gene with the largest absolute log2 fold change. Answer format: a single gene symbol (shape: gene_symbol).\n",
      "Answer: ZFAND5\n",
      "\n",
      "Dataset: /mlbio_scratch/panigrah/agents/data/nature/paper2/processed_human_muscle_data.h5ad\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Description: Within the MuSC (muscle stem cell) population (annotation_level0 == \"MuSC\"), perform a differential expression analysis comparing the \"old\" versus \"young\" age groups (Age_bin) using Scanpy's default Wilcoxon rank‑sum test (rank_genes_groups) on the raw count matrix without additional filtering. Report the gene symbol with the largest absolute log2‑fold‑change (|logFC|) between old and young. Answer format: a single HGNC gene symbol (shape: gene_symbol).\n",
      "Answer: AC022414.1\n",
      "\n",
      "Task 1\n",
      "Description: Using the provided AnnData file, calculate the proportion of each annotation_level2 cell type within each assay batch (batch column, values cells and nuclei). Determine which cell type shows the greatest absolute difference in proportion between the two batches. Report the exact cell-type name as a single label (case‑sensitive, include any hyphens).\n",
      "Answer: MF-I\n",
      "\n",
      "Task 2\n",
      "Description: Using the provided single‑cell RNA‑seq dataset, restrict the analysis to cells annotated as MuSC (muscle stem cells). Define two age groups: \"young\" = Age_group \"15-20\" and \"old\" = Age_group \"70-75\". Perform a differential expression test between old and young MuSC cells using Scanpy's rank_genes_groups with the Wilcoxon method, using the young group as reference and Benjamini‑Hochberg correction for multiple testing. Consider only genes with adjusted p‑value < 0.05. Identify the gene with the greatest absolute log2 fold‑change (old vs young) among these significant genes. Answer format: a single HGNC gene symbol in uppercase (shape: gene_symbol).\n",
      "Answer: XIST\n",
      "\n",
      "Task 3\n",
      "Description: Using the provided AnnData file, subset the cells annotated as 'MF-I' (annotation_level0). Within this subset, compare the oldest age group (Age_group = '70-75') to the youngest age group (Age_group = '15-20') and perform differential expression analysis (Wilcoxon test, default Scanpy settings). Report the HGNC gene symbol of the gene with the largest positive log₂ fold‑change that has an adjusted p‑value < 0.05. Answer format: a single gene symbol (shape: gene_symbol, uppercase).\n",
      "Answer: XIST\n",
      "\n",
      "Task 4\n",
      "Description: Using the provided AnnData object, identify the cell type with the highest number of cells in the 'annotation_level1' column. Subset the data to this cell type, normalize, log‑transform, select 2,000 highly variable genes, scale, and compute 30 principal components. Build a k‑nearest‑neighbour graph with k = 30 on the PCA space and compute the inverse Simpson index (iLISI) of the 'batch' labels among each cell's neighbours. Report the batch (value from the 'batch' column) that has the lowest mean iLISI across all cells of this cell type. Answer format: a single label (shape: label).\n",
      "Answer: nuclei\n",
      "\n",
      "Task 5\n",
      "Description: Identify the gene that exhibits the strongest positive monotonic relationship with donor age (Spearman correlation) among all cell types that contain at least 5,000 cells. Use the numeric midpoint of each Age_group as the age variable. Provide the answer as a single HGNC gene symbol (shape: gene_symbol).\n",
      "Answer: RPL10\n",
      "\n",
      "Task 6\n",
      "Description: Identify the DonorID that has the highest fraction of cells annotated as \"MF-I\" (annotation_level2) among all donors in the dataset. Answer format: a single label exactly as it appears in the DonorID column (shape: label).\n",
      "Answer: 500C\n",
      "\n",
      "Task 7\n",
      "Description: Using the provided AnnData object, compute the proportion of each cell type (annotation_level0) within the two age groups defined by the 'Age_bin' column (young and old). Identify the cell type that exhibits the greatest positive difference in proportion when comparing old versus young (i.e., proportion in old minus proportion in young is maximal). Answer format: a single label exactly as it appears in the 'annotation_level0' column (shape: label).\n",
      "Answer: MF-IIsc(fg)\n",
      "\n",
      "Task 8\n",
      "Description: Using the provided AnnData object, compute the mean UMI count (n_counts) for each major cell type (column 'annotation_level1') separately in the two age groups defined by 'Age_bin' (values 'old' and 'young'). Identify the cell type with the largest absolute difference between the mean n_counts of the old and young groups. Answer format: a single cell‑type label exactly as it appears in 'annotation_level1' (shape: label).\n",
      "Answer: Mesothelium\n",
      "\n",
      "Task 9\n",
      "Description: Identify the annotation_level0 cell type that exhibits the largest positive log2 fold-change (old vs young) for the gene COL1A1 expression. Answer format: a single label (the cell type name exactly as it appears in the 'annotation_level0' column).\n",
      "Answer: Adipocyte\n",
      "\n",
      "Task 10\n",
      "Description: Using the provided AnnData object, consider the two age groups defined in the 'Age_group' column: the youngest group (15-20) and the oldest group (70-75). For each fine-grained cell type label in 'annotation_level2', compute the proportion of cells belonging to that cell type within each age group (proportion = number of cells of that type in the group divided by the total number of cells in the group). Identify the cell type that exhibits the largest positive increase in proportion from the youngest to the oldest age group (i.e., proportion_old - proportion_young is maximal and > 0). Provide the answer as a single cell‑type label exactly as it appears in 'annotation_level2'. Answer format: a single token (shape: label).\n",
      "Answer: MF_typeII(fg)\n",
      "\n",
      "Task 11\n",
      "Description: Within the provided human muscle single-cell dataset, consider the cell type annotations in the column 'annotation_level2'. For each cell type that has at least 30 cells in both the 'old' and 'young' age bins (as defined by the 'Age_bin' column), compute the ratio of the number of 'old' cells to the number of 'young' cells. Identify the cell type with the highest old-to-young ratio. Answer format: a single cell‑type label exactly as it appears in 'annotation_level2' (shape: label).\n",
      "Answer: CD16-NK\n",
      "\n",
      "Task 12\n",
      "Description: Using the provided single‑cell dataset, compute for each donor the proportion of cells annotated as MuSC (annotation_level2 == \"MuSC\"). Then, for each Age_group, calculate the average of these donor‑level MuSC proportions. Identify the Age_group with the highest average MuSC proportion. Answer format: a single label matching exactly one of the Age_group strings present in the data (shape: label).\n",
      "Answer: 25-30\n",
      "\n",
      "Task 13\n",
      "Description: Identify the cell type (annotation_level2) that exhibits the largest absolute difference in average library size (n_counts) between the two batches ('cells' vs 'nuclei'). Report the cell type label exactly as it appears in the annotation_level2 column. Answer format: a single label (shape: label).\n",
      "Answer: B-plasma\n",
      "\n",
      "Task 14\n",
      "Description: Using the provided AnnData object, compute the Euclidean distance between the mean scVI embeddings (X_scVI) of the two batches ('cells' and 'nuclei') for each cell type in the 'annotation_level1' column. Identify the cell type that shows the smallest centroid distance, i.e., the least batch effect. Answer format: a single cell‑type label (shape: label).\n",
      "Answer: Tenocyte\n",
      "\n",
      "Task 15\n",
      "Description: Using the provided AnnData file, compute the mean mitochondrial gene expression percentage (percent_mito) for each cell type (annotation_level1) separately for male (Sex = 'M') and female (Sex = 'F') cells. Identify the cell type that exhibits the largest absolute difference in mean percent_mito between the two sexes. Answer format: a single cell‑type label exactly as it appears in the annotation_level1 column (shape: label).\n",
      "Answer: RBC\n",
      "\n",
      "Dataset: /mlbio_scratch/panigrah/agents/data/nature/paper2/processed_myonuclei_data.h5ad\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Description: Using the provided myonuclei single-nucleus RNA‑seq dataset, compute for each cell type in the 'annotation_level1' annotation the proportion of nuclei belonging to the 'old' age bin and the proportion belonging to the 'young' age bin. Add a pseudocount of 1 to the counts of each age bin for every cell type to avoid division‑by‑zero. Calculate the log2 enrichment as log2(old_proportion / young_proportion). Identify the cell type with the highest log2 enrichment (i.e., strongest relative enrichment in old nuclei). Answer format: a single label matching exactly one entry of the 'annotation_level1' column (case‑sensitive, including spaces).\n",
      "Answer: NMJ accessory\n",
      "\n",
      "Task 1\n",
      "Description: Using the provided myonuclei dataset, for each SampleID compute the proportion of nuclei belonging to each 'annotation_level1' cell type. Convert each sample's 'Age_group' (e.g., \"15-20\") to the numeric midpoint (e.g., 17.5). Calculate the Spearman rank correlation (ρ) between the cell‑type proportion and the numeric age midpoint across samples. Identify the cell type with the strongest positive age‑associated increase (largest positive ρ). Answer format: a single label exactly as it appears in the 'annotation_level1' column (shape: label).\n",
      "Answer: NMJ accessory\n",
      "\n",
      "Task 2\n",
      "Description: Using the provided muscle nuclei dataset, compute for each cell type annotation (annotation_level1) the proportion of nuclei belonging to that annotation within each age group. Convert the age groups to numeric midpoints (15-20 → 17.5, 20-25 → 22.5, 25-30 → 27.5, 35-40 → 37.5, 55-60 → 57.5, 60-65 → 62.5, 70-75 → 72.5). For each annotation, calculate the Spearman correlation (ρ) between the numeric age and its proportion across the age groups. Identify the annotation with the highest positive Spearman ρ. Answer format: a single label exactly as it appears in the \"annotation_level1\" column (shape: label).\n",
      "Answer: NMJ accessory\n",
      "\n",
      "Task 3\n",
      "Description: Using the provided AnnData object, compute the average silhouette score for each distinct `myofiber_type` (excluding the \"nan\" category) based on the scVI latent embedding stored in `obsm[\"X_scVI\"]`. Report the `myofiber_type` that attains the highest average silhouette score. Answer format: a single token exactly matching one of the `myofiber_type` categories (shape: label, case-sensitive, no spaces).\n",
      "Answer: MYH1+MYH7+\n",
      "\n",
      "Task 4\n",
      "Description: Identify the annotation_level1 cell type that exhibits the highest average expression of the gene MYH7 across all nuclei in the dataset. Answer format: a single cell type label exactly as it appears in the 'annotation_level1' column (shape: label).\n",
      "Answer: MF-Isn(fg)\n",
      "\n",
      "Task 5\n",
      "Description: Using the provided muscle nuclei dataset, compute for each fine-grained cell type (annotation_level1) the proportion of cells contributed by each donor (DonorID). Then calculate the coefficient of variation (CV = standard deviation / mean) of these donor-wise proportions across all donors. Report the cell type (annotation_level1 label) that exhibits the highest CV. Answer format: a single label (shape: label).\n",
      "Answer: MF-Isn(fg)\n",
      "\n",
      "Task 6\n",
      "Description: Apply the following quality control filters to the muscle nuclei dataset: keep cells with scrublet_score < 0.2, n_counts ≥ 1000, n_genes ≥ 500, and percent_mito < 0.05. Exclude cells where myofiber_type is missing ('nan'). Then, for each donor, compute the fraction of retained cells that are annotated as MYH7+. Identify the donor with the highest MYH7+ fraction. Answer format: a single donor identifier (shape: label).\n",
      "Answer: 500C\n",
      "\n",
      "Task 7\n",
      "Description: Identify the annotation_level0 cell type with the highest average mitochondrial transcript proportion (percent_mito) across all nuclei. Answer format: a single label exactly as it appears in the annotation_level0 column (shape: label).\n",
      "Answer: MF_typeI(cytoplasmic)\n",
      "\n",
      "Task 8\n",
      "Description: Using the scVI latent embeddings (X_scVI) provided in the AnnData object, compute for each top‑level cell type (annotation_level0) the average Euclidean distance of its nuclei to the centroid of that cell type in the scVI space. Identify the cell type with the smallest average distance (i.e., the most compact cluster). Answer format: a single cell‑type label exactly as it appears in annotation_level0 (shape: label).\n",
      "Answer: MF_typeII(cytoplasmic)\n",
      "\n",
      "Task 9\n",
      "Description: Using the provided myonuclei single‑cell dataset, determine which non‑missing myofiber_type category has the highest average expression of the gene MYH7. Report the myofiber_type label exactly as it appears in the data (case‑sensitive, including any '+' characters). Answer format: a single label (shape: label).\n",
      "Answer: MYH2+MYH7+\n",
      "\n",
      "Task 10\n",
      "Description: Among the myofiber_type categories present in the dataset (excluding the 'nan' category), which myofiber type shows the highest average raw expression level of the muscle gene MYH7? Answer format: a single myofiber_type label exactly as it appears in the 'myofiber_type' column (shape: label, case‑sensitive).\n",
      "Answer: MYH2+MYH7+\n",
      "\n",
      "Task 11\n",
      "Description: Identify the myofiber_type annotation that exhibits the highest average percent_mito across all nuclei in the dataset. Answer format: a single label exactly as it appears in the 'myofiber_type' column (shape: label).\n",
      "Answer: MYH1+MYH2+MYH7+\n",
      "\n",
      "Task 12\n",
      "Description: Identify the myofiber_type that exhibits the strongest donor-specific bias, defined as the highest proportion of its nuclei contributed by a single donor. Answer format: a single label exactly as it appears in the 'myofiber_type' column (shape: label).\n",
      "Answer: MYH1+MYH2+MYH7+\n",
      "\n",
      "Task 13\n",
      "Description: Using the provided `processed_myonuclei_data.h5ad`, compute the Euclidean distance between the mean scVI embeddings of the “young” and “old” samples for each `annotation_level0` cell type. Which cell type shows the largest distance? Answer format: a single label exactly as it appears in `annotation_level0` (shape: label).\n",
      "Answer: Specialised_MF_populations\n",
      "\n",
      "Dataset: /mlbio_scratch/panigrah/agents/data/nature/paper2/processed_myonucleicells_data.h5ad\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Description: From the AnnData file, calculate for each cell type in the column `annotation_level1` its proportion of cells within each batch (`batch` column, values 'cells' and 'nuclei'). Identify the cell type with the greatest absolute difference in proportion between the two batches. Provide the cell‑type label exactly as it appears in `annotation_level1`. Answer format: a single label (shape: label).\n",
      "Answer: MF-I\n",
      "\n",
      "Task 1\n",
      "Description: Using the provided AnnData object, compute the proportion of each broad cell type (annotation_level0) within each age bin (young vs old). Identify the cell type with the greatest absolute difference in proportion between the two age bins. Answer format: a single label exactly matching the cell type name as it appears in annotation_level0 (shape: label).\n",
      "Answer: MF-I\n",
      "\n",
      "Task 2\n",
      "Description: Using the scANVI embedding (X_scANVI) provided in the AnnData object, compute the 30‑nearest‑neighbor graph (Euclidean distance) for all cells. For each cell, calculate the fraction of its 30 neighbors that belong to the same batch (the 'batch' column in .obs). Then average this fraction across all cells within each batch. Report the batch that has the highest average same‑batch neighbor fraction (i.e., the batch showing the poorest mixing). Answer format: a single lowercase label (shape: label).\n",
      "Answer: nuclei\n",
      "\n",
      "Task 3\n",
      "Description: After removing cells with scrublet_score > 0.2, compute the average percent_mito for each cell type in the annotation_level0 column. Report the cell type (exact string as it appears in annotation_level0) that has the highest average percent_mito. Answer format: a single label (shape: label).\n",
      "Answer: MF-Isn(fg)\n",
      "\n",
      "Task 4\n",
      "Description: After removing cells with a Scrublet doublet score greater than 0.1, compute the mean percent_mito for each cell type defined in the column 'annotation_level1'. Which cell type has the highest average percent_mito? Answer format: a single label exactly as it appears in 'annotation_level1' (shape: label).\n",
      "Answer: MF-Isn(fg)\n",
      "\n",
      "Task 5\n",
      "Description: Using the provided AnnData file (/mlbio_scratch/panigrah/agents/data/nature/paper2/processed_myonucleicells_data.h5ad), compute the average raw counts expression of the gene MYH1 for each cell type defined in the 'annotation_level2' column of adata.obs. Identify the cell type with the highest mean MYH1 expression. Answer format: a single label that matches exactly one of the annotation_level2 categories (shape: label).\n",
      "Answer: MF-IIsn(fg)\n",
      "\n",
      "Task 6\n",
      "Description: Identify the batch (value in the 'batch' column of the AnnData object) that has the highest average mitochondrial transcript proportion ('percent_mito'). Answer format: a single batch identifier as a lowercase label (shape: label).\n",
      "Answer: cells\n",
      "\n",
      "Task 7\n",
      "Description: Using the provided single‑nucleus muscle dataset, perform a differential expression analysis between the two major cell types MF-I and MF-II (group: MF-I vs reference: MF-II) with Scanpy's default Wilcoxon rank‑sum test (rank_genes_groups). Identify the gene that shows the strongest positive log2 fold‑change (log2FC) in MF-I relative to MF-II. Answer format: a single HGNC gene symbol (shape: gene_symbol).\n",
      "Answer: GLIS3-AS1\n",
      "\n",
      "Task 8\n",
      "Description: Identify the cell type that exhibits the highest average expression of the gene MYH1 across the entire dataset. Within that cell type, compare the average MYH1 expression between the two age categories ('young' and 'old') as defined in the 'Age_bin' metadata. Report the age category that shows the higher mean MYH1 expression. Answer format: a single lowercase label (shape: label).\n",
      "Answer: young\n",
      "\n",
      "Task 9\n",
      "Description: Compute the overall proportion of nuclei (batch == 'nuclei') across all cells in the provided AnnData file. For each cell type in the 'annotation_level1' column, compute the proportion of nuclei among cells of that type. Calculate the absolute deviation of each cell type's nuclei proportion from the overall proportion. Identify the cell type(s) with the maximum deviation; if multiple cell types share this maximum, choose the one with the highest total cell count. Return the selected cell type label exactly as it appears in 'annotation_level1'. Answer format: a single label (shape: label).\n",
      "Answer: MF-IIsc(fg)\n",
      "\n",
      "Task 10\n",
      "Description: For each cell type defined in the 'annotation_level2' column, compute the average n_counts (total UMI counts) separately for female (Sex = F) and male (Sex = M) cells. Identify the cell type that shows the largest absolute difference between the two sex-specific averages (|mean_F - mean_M|). Provide the answer as a single cell type label exactly as it appears in 'annotation_level2' (shape: label).\n",
      "Answer: NMJ\n",
      "\n",
      "Task 11\n",
      "Description: Determine which Age_group has the highest proportion of cells annotated as MF_type-MYH8+ (annotation_level2). Report the Age_group label (e.g., \"70-75\"). Answer format: a single label (shape: label).\n",
      "Answer: 70-75\n",
      "\n",
      "Task 12\n",
      "Description: Among the annotation_level1 categories in the provided dataset, which category has the highest proportion of nucleus-derived measurements (i.e., nuclei / total cells) when considering both 'cells' and 'nuclei' batches? Answer format: a single label (shape: label).\n",
      "Answer: Hyb\n",
      "\n",
      "Task 13\n",
      "Description: Within the cell type labeled 'MF-I' (annotation_level1) in the provided dataset, compute the average mitochondrial gene percentage (percent_mito) for each Age_group and report the Age_group that has the highest average percent_mito. Answer format: a single Age_group label exactly as it appears in the data (shape: label).\n",
      "Answer: 20-25\n",
      "\n",
      "Task 14\n",
      "Description: Identify the annotation_level2 cell type whose proportion of cells originating from the 'cells' batch (i.e., fraction = #cells in batch 'cells' / total #cells for that type) is maximal. If multiple cell types share the maximal proportion, return the one that comes first alphabetically. Answer format: a single label (shape: label) exactly as it appears in the annotation_level2 column.\n",
      "Answer: MF_type-MYH8+\n",
      "\n",
      "Task 15\n",
      "Description: Using the provided single-nucleus/cell dataset, consider the fine-grained cell-type annotation column 'annotation_level2'. Restrict the analysis to cell types that contain at least 100 cells. Compute the average mitochondrial RNA proportion ('percent_mito') for each remaining cell type and identify the one with the highest mean value. Answer format: a single cell-type label exactly as it appears in 'annotation_level2' (shape: label).\n",
      "Answer: MF_type-MYH8+\n",
      "\n",
      "Dataset: /mlbio_scratch/panigrah/agents/data/nature/paper5/mouse_data.h5ad\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Description: Identify the epithelial cell type whose proportion of cells increases most strongly from the most proximal segment (segment 1) to the most distal segment (segment 30) of the mouse small intestine. For each segment, compute the proportion of each cell type (as defined in the 'cell.type' column) among all cells in that segment, fit a simple linear regression of proportion versus segment number (1‑30), and select the cell type with the largest positive slope. Answer format: a single cell type label exactly as it appears in the 'cell.type' column (case‑sensitive).\n",
      "Answer: Goblet cell\n",
      "\n",
      "Task 1\n",
      "Description: For each of the 30 intestinal segments (1‑30) compute the proportion of cells belonging to each cell type (as listed in the 'cell.type' column). Then calculate the Spearman correlation between segment number and the proportion of each cell type. Identify the cell type with the highest positive Spearman correlation (strongest monotonic increase along the proximal‑distal axis). Provide the answer as a single label exactly matching the value in the 'cell.type' column (case‑sensitive, including spaces).\n",
      "Answer: Goblet cell\n",
      "\n",
      "Task 2\n",
      "Description: Identify the cell type whose proportion across the 30 intestinal segments shows the strongest positive monotonic trend (i.e., the highest Pearson correlation between segment number and the cell-type proportion). Answer format: a single cell type label exactly as it appears in the 'cell.type' column (shape: label).\n",
      "Answer: Goblet cell\n",
      "\n",
      "Task 3\n",
      "Description: Using the provided single-cell dataset of mouse intestinal epithelial cells, calculate for each intestinal segment (1 to 30) the proportion of cells annotated as \"Stem cell\" (i.e., number of Stem cell entries divided by total cells in that segment). Identify the segment that has the highest Stem cell proportion. Answer format: a single integer representing the segment number (shape: number).\n",
      "Answer: 9\n",
      "\n",
      "Task 4\n",
      "Description: In the provided single-cell dataset, each cell is annotated with a 'cell.type' and a 'segment' (values 1-30) indicating its position along the mouse small intestine. For each segment, compute the proportion of cells belonging to each cell type (i.e., number of cells of that type divided by the total number of cells in the segment). Identify the cell type that achieves the highest proportion in any segment across the whole dataset. Report this cell type as a single lowercase label with underscores (shape: label).\n",
      "Answer: stem_cell\n",
      "\n",
      "Task 5\n",
      "Description: Identify the epithelial cell type that exhibits the strongest monotonic trend (largest absolute Spearman correlation) in its average per‑cell gene expression across the 30 anatomical segments (proximal segment 1 to distal segment 30) of the mouse small intestine. Report the answer as a single lowercase label with underscores (shape: label).\n",
      "Answer: mature_enterocyte\n",
      "\n",
      "Task 6\n",
      "Description: Within the goblet cell population, identify the intestinal segment that exhibits the highest average expression of the canonical goblet‑cell marker Muc2. Provide the segment identifier exactly as it appears in the 'segment' column. Answer format: a single label (e.g., 25).\n",
      "Answer: 25\n",
      "\n",
      "Task 7\n",
      "Description: Using the provided single‑cell dataset, compute the proportion of Goblet cells within each intestinal segment (the 'segment' column). Then calculate the enrichment ratio for each segment as (segment Goblet proportion) ÷ (overall Goblet proportion across all cells). Identify the segment with the highest enrichment ratio. Answer format: a single integer segment identifier (shape: number).\n",
      "Answer: 26\n",
      "\n",
      "Dataset: /mlbio_scratch/panigrah/agents/data/nature/paper6/data.h5ad\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Description: Using the provided single‑cell RNA‑seq atlas of the aging mouse ovary, compute for each annotated cell type the gene with the largest absolute log2 fold‑change (log2FC) between OLD and YOUNG mice (log2FC = log2(mean_OLD / mean_YOUNG), using CPM‑normalized log1p expression). Identify the cell type whose maximal absolute log2FC is the greatest among all cell types. Answer format: a single lowercase label with underscores (shape: label).\n",
      "Answer: b_cell\n",
      "\n",
      "Task 1\n",
      "Description: Identify the cell type that shows the greatest enrichment in the OLD mice compared to YOUNG mice, measured as the fold‑change of its proportion (OLD/YOUNG). Report the cell type name as a single label with spaces replaced by underscores (shape: label).\n",
      "Answer: B_cell\n",
      "\n",
      "Task 2\n",
      "Description: Using the provided single‑cell atlas of the aging mouse ovary, compute the proportion of each cell type (column 'cluster.names') within the two age groups labeled 'YOUNG' and 'OLD' in the 'age' column. Calculate the fold‑change of each cell type's proportion as (OLD proportion) / (YOUNG proportion). Identify the cell type with the largest fold‑change greater than 1 (i.e., the greatest relative increase from young to old). Answer format: a single cell‑type label (shape: label).\n",
      "Answer: B-Lymphocytes\n",
      "\n",
      "Task 3\n",
      "Description: In the provided single‑cell atlas of the aging mouse ovary, restrict the analysis to cells annotated as \"stromal cell\". Using the log‑normalized expression matrix, perform a differential expression comparison between cells from the OLD age group versus the YOUNG age group (Wilcoxon rank‑sum test, all genes). Identify the gene that exhibits the greatest absolute log2 fold‑change (|log2FC|) between the two age groups. Report the official mouse gene symbol of this gene. Answer format: a single gene symbol (shape: gene_symbol, uppercase).\n",
      "Answer: MT-CO3\n",
      "\n",
      "Task 4\n",
      "Description: Identify the cell type that exhibits the greatest absolute log2 fold-change in proportion between the OLD and YOUNG age groups. For each cell type, compute its proportion within each age (count of cells of that type divided by total cells of that age), calculate log2(proportion_OLD / proportion_YOUNG), take the absolute value, and report the cell type with the highest absolute value. Answer format: a single cell type label exactly as it appears in the `cell_type` column (shape: label).\n",
      "Answer: B cell\n",
      "\n",
      "Task 5\n",
      "Description: Based on the provided single-cell atlas of the aging mouse ovary, compute the proportion of each cell type within the two age groups (YOUNG and OLD). Then calculate the log2 fold-change of these proportions (OLD vs YOUNG). Identify the cell type that shows the greatest positive log2 fold-change (i.e., the strongest increase in relative abundance in OLD compared to YOUNG). Answer format: a single cell type label exactly as it appears in the 'cell_type' column (shape: label).\n",
      "Answer: B cell\n",
      "\n",
      "Task 6\n",
      "Description: Identify the cell type that shows the largest absolute change in its proportion between the OLD and YOUNG age groups in the provided mouse ovary single‑cell atlas. Answer format: a single lowercase label with words joined by underscores (shape: label).\n",
      "Answer: granulosa_cell\n",
      "\n",
      "Task 7\n",
      "Description: Using the provided AnnData object (data.h5ad), compute for each cell type the median of the absolute log2 fold‑change (log2FC) of gene expression between the YOUNG and OLD age groups (log2FC = mean expression in YOUNG – mean expression in OLD, using the stored expression matrix). Identify the cell type with the largest median absolute log2FC. Answer format: a single cell‑type label exactly as it appears in the 'cell_type' column of adata.obs (shape: label).\n",
      "Answer: oocyte\n",
      "\n",
      "Task 8\n",
      "Description: Given the single-cell atlas of the aging mouse ovary, compare the relative abundance of each cell type between the two age groups (YOUNG and OLD). Identify the cell type whose proportion is most increased in the OLD group relative to the YOUNG group (i.e., the largest positive difference OLD proportion – YOUNG proportion). Provide the answer as a single cell type label exactly as it appears in the 'cluster.names' annotation (shape: label).\n",
      "Answer: Stroma A\n",
      "\n",
      "Task 9\n",
      "Description: In the single-cell atlas of the aging mouse ovary, consider only the granulosa cells. Perform differential expression between the OLD and YOUNG age groups (OLD vs YOUNG) and identify the gene that shows the greatest positive log2 fold‑change (i.e., most up‑regulated in OLD). Provide the answer as a single HGNC gene symbol in uppercase. (shape: gene_symbol)\n",
      "Answer: LUZP4\n",
      "\n",
      "Task 10\n",
      "Description: Using the provided AnnData object, filter out cells flagged as doublets (`predicted_doublet == True`) and cells with mitochondrial transcript proportion > 10%. Then, compute the inverse Local Inverse Simpson’s Index (iLISI) for donor mixing within each Leiden cluster (using the `X_pca` embedding, k=30 nearest neighbors). Report the Leiden cluster identifier that has the lowest iLISI (i.e., the cluster with the poorest donor mixing). Answer format: a single integer (shape: label).\n",
      "Answer: 11\n",
      "\n",
      "Task 11\n",
      "Description: Using the provided single‑cell atlas of the aging mouse ovary, compute the average expression of the senescence marker gene Cdkn2a for each cell type within the YOUNG and OLD age groups. Then calculate the log2 fold change (OLD vs YOUNG) for each cell type (add a pseudocount of 1e-3 to avoid log of zero). Report the cell‑type name that shows the highest positive log2 fold change, formatted exactly as it appears in the `cell_type` column but converted to lower‑case with spaces replaced by underscores. Answer format: a single label (shape: label).\n",
      "Answer: phagocyte\n",
      "\n",
      "Dataset: /mlbio_scratch/panigrah/agents/data/cell/paper1/data.h5ad\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Description: From the provided human brain snRNA‑seq dataset, identify the donor (donor_id) that has the highest fraction of cells labeled as \"microglial cell\". Answer format: a single donor identifier string exactly as it appears in the 'donor_id' column (shape: label).\n",
      "Answer: R24_P2538\n",
      "\n",
      "Task 1\n",
      "Description: Using the provided single-nucleus RNA‑seq dataset, calculate for each disease category (as listed in the 'disease' column) the proportion of cells annotated as \"microglial cell\" (based on the 'cell_type' column). Identify the disease with the highest microglial cell proportion. Answer format: a single disease name exactly as it appears in the 'disease' column (shape: label).\n",
      "Answer: Alzheimer disease\n",
      "\n",
      "Task 2\n",
      "Description: Using the provided snRNA‑seq AnnData object, consider only cells annotated as the most abundant cell type (the cell type with the highest count in `adata.obs['cell_type']`). Compute the mean iLISI (inverse Simpson’s Index) for each donor within this subset, based on the existing PCA embedding (`obsm['X_pca']`). Identify the donor whose cells have the **lowest** mean iLISI (i.e., poorest mixing after integration). Report this lowest mean iLISI value rounded to three decimal places. Answer format: a single numeric value (shape: number, three‑decimal rounding).\n",
      "Answer: 1.237\n",
      "\n",
      "Task 3\n",
      "Description: Using the provided AnnData object, determine which cell type has the highest average expression of the marker gene GFAP. Use the 'cell_type' column in obs for cell type labels and report the label exactly as it appears (case‑sensitive). Answer format: a single label (shape: label).\n",
      "Answer: astrocyte\n",
      "\n",
      "Task 4\n",
      "Description: From the provided human brain single-nucleus RNA‑seq dataset, consider the neuronal marker genes SLC17A7, GAD1, SNAP25, SYT1, CAMK2A, GRIN1, and NEUROD1. For each gene that is present in the dataset, compute the median expression across all cells belonging to each donor (column 'donor_id' in adata.obs). Identify the gene whose donor‑wise median expressions have the largest variance across donors. Then, for that selected gene, determine which donor exhibits the highest median expression. Answer format: a single donor identifier (shape: label).\n",
      "Answer: R24_P2324\n",
      "\n",
      "Task 5\n",
      "Description: For each donor in the dataset, compute the proportion of each cell type among that donor's cells and identify the cell type with the highest proportion for that donor. Then, among all donors, find the donor whose most abundant cell type accounts for the greatest proportion of its cells. Report the answer as a pair formatted as DONOR_ID::CELL_TYPE (no spaces).\n",
      "Answer: R24_P2392::oligodendrocyte\n",
      "\n",
      "Task 6\n",
      "Description: Using the provided single-nucleus RNA‑seq AnnData object, compute for each gene the variance of its average expression across all annotated cell types (the 'cell_type' column in .obs). Identify the gene with the highest variance and report its HGNC gene symbol (the value in the 'feature_name' column of .var). Answer format: a single uppercase gene symbol (shape: gene_symbol).\n",
      "Answer: PCDH9\n",
      "\n",
      "Task 7\n",
      "Description: Using the provided human brain snRNA‑seq dataset, calculate the mean expression of the gene NPY (ENSG00000122585) for each cell type listed in the 'cell_type' column of adata.obs. Identify the cell type with the highest average NPY expression. Answer format: a single cell type label exactly as it appears in the 'cell_type' column (shape: label).\n",
      "Answer: inhibitory interneuron\n",
      "\n",
      "Task 8\n",
      "Description: Using the provided human brain snRNA‑seq dataset, compare the average expression between the cell types \"oligodendrocyte\" and \"astrocyte\". Compute the log2‑fold change (log2(mean expression in oligodendrocytes + 1e-6) – log2(mean expression in astrocytes + 1e-6)) for each protein‑coding gene (identified by feature_type == \"protein_coding\"). Report the HGNC gene symbol of the protein‑coding gene with the largest positive log2‑fold change (i.e., most up‑regulated in oligodendrocytes versus astrocytes). Answer format: a single gene symbol in uppercase (shape: gene_symbol).\n",
      "Answer: GPR141\n",
      "\n",
      "Task 9\n",
      "Description: Within the provided human brain snRNA‑seq dataset, consider only cells annotated as \"glutamatergic neuron\". Compare the gene expression between cells from donors with \"Alzheimer disease\" and those labeled as \"normal\". Perform a differential expression analysis (e.g., Scanpy's rank_genes_groups with the t‑test method) and identify the gene that shows the strongest positive log2 fold‑change (largest logFC) among those that are statistically significant (adjusted p‑value < 0.05). Provide the answer as the HGNC gene symbol of that gene (shape: gene_symbol).\n",
      "Answer: IGHG1\n",
      "\n",
      "Task 10\n",
      "Description: Identify the cell type–disease pair with the highest log2 enrichment (observed count divided by expected count under independence) in the provided dataset. Answer format: a single string \"CELLTYPE::DISEASE\" (shape: pair).\n",
      "Answer: pericyte::Pick disease\n",
      "\n",
      "Task 11\n",
      "Description: Identify the donor (donor_id) that exhibits the highest proportion of oligodendrocyte nuclei relative to its total nuclei count across all donors in the dataset. Answer format: a single donor identifier string (shape: label).\n",
      "Answer: R24_P2392\n",
      "\n",
      "Task 12\n",
      "Description: Using the provided single-nucleus RNA‑seq dataset, compute the percentage of mitochondrial gene expression for each cell (using genes whose feature_name starts with \"MT-\"). Then, within the cells annotated as \"oligodendrocyte\" (cell_type == \"oligodendrocyte\"), calculate the average mitochondrial percentage for each donor (donor_id). Identify the donor that has the highest average mitochondrial percentage among oligodendrocyte cells. Answer format: a single donor identifier string (shape: label).\n",
      "Answer: R24_P2730\n",
      "\n",
      "Task 13\n",
      "Description: Using the AnnData file at /mlbio_scratch/panigrah/agents/data/cell/paper1/data.h5ad, compute the silhouette score of donor batches (donor_id) for each cell type using the PCA embedding stored in X_pca. Only consider cell types that contain at least 10 cells and include cells from at least two distinct donors. Identify the cell type whose absolute silhouette score (|silhouette|) is the largest (i.e., shows the strongest donor-driven segregation). Provide the answer as a single lowercase cell type label (shape: label).\n",
      "Answer: pericyte\n",
      "\n",
      "Dataset: /mlbio_scratch/panigrah/agents/data/cell/paper2/data.h5ad\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Description: In the provided placenta infection scRNA‑seq dataset, consider only cells annotated as Hofbauer cells (celltype_annotation == 'HBC'). Define \"infected\" cells as those with infection status 'Tg', 'Pf', or 'Lm', and \"uninfected\" cells as those with status 'UI'. Using the normalized expression matrix, compute the mean expression of each gene within the infected group and within the uninfected group. Calculate the log2 fold‑change as log2(mean_infected + 1e‑6) - log2(mean_uninfected + 1e‑6). Among genes whose mean expression in the infected group exceeds 0.1, report the gene with the highest positive log2 fold‑change. Answer format: a single HGNC gene symbol (shape: gene_symbol).\n",
      "Answer: CSF3\n",
      "\n",
      "Task 1\n",
      "Description: In the provided placenta infection scRNA‑seq dataset, identify the cell type with the greatest number of cells. Within this cell type, compare cells infected with Listeria monocytogenes (infection label \"Lm\") to uninfected cells (infection label \"UI\"). Using the log‑normalized expression matrix, compute the mean expression of each gene in the Lm and UI groups, calculate the log2 fold change (mean_Lm – mean_UI), and report the HGNC gene symbol of the gene with the highest positive log2 fold change. Answer format: a single gene symbol (shape: gene_symbol).\n",
      "Answer: XIST\n",
      "\n",
      "Task 2\n",
      "Description: Within the 48 h time point, identify the cell type (given by its ontology ID in the column 'cell_type_ontology_term_id') that has the highest proportion of Plasmodium falciparum‑infected (Pf) cells. Answer format: a single label (the ontology ID, e.g., CL:0000235).\n",
      "Answer: CL:2000060\n",
      "\n",
      "Task 3\n",
      "Description: Identify the donor (value in the 'donor_id' column) that exhibits the highest proportion of cells marked as Tg_infected = True. Answer format: a single label (shape: label) exactly as it appears in the donor_id field.\n",
      "Answer: scDonor_Tg1_mother\n",
      "\n",
      "Task 4\n",
      "Description: Determine which cell type has the highest proportion of infected cells (i.e., cells where the 'infection' field is not 'UI') in the provided dataset. Return the cell type label exactly as it appears in the 'cell_type' column. Answer format: a single label (shape: label).\n",
      "Answer: Hofbauer cell\n",
      "\n",
      "Task 5\n",
      "Description: Using the provided single‑cell placenta infection dataset, compute the proportion of cells annotated as \"macrophage\" within each infection condition (UI, Tg, Pf, Lm). Identify the infection condition that has the highest macrophage proportion. Answer format: a single infection label exactly as it appears in the 'infection' column (shape: label).\n",
      "Answer: Lm\n",
      "\n",
      "Task 6\n",
      "Description: From the provided placenta infection single-cell dataset, determine which infection condition ('infection' column) has the highest relative abundance of Hofbauer cells ('cell_type' == 'Hofbauer cell'). Report the infection label exactly as it appears in the 'infection' column (case-sensitive). Answer format: a single label (shape: label).\n",
      "Answer: Tg\n",
      "\n",
      "Task 7\n",
      "Description: For each infection condition (UI, Lm, Pf, Tg) compute the proportion of cells that are annotated with the most abundant cell type identifier CL:3000001. Report the infection label (exact case) that has the highest proportion. Answer format: a single infection label (shape: label).\n",
      "Answer: Tg\n",
      "\n",
      "Task 8\n",
      "Description: Identify the infection condition (one of UI, Lm, Tg, Pf) that has the highest proportion of cells belonging to the most abundant cell type across the entire dataset (the cell type with the greatest total cell count). Answer format: a single lowercase label (shape: label).\n",
      "Answer: tg\n",
      "\n",
      "Task 9\n",
      "Description: Using the provided placenta scRNA‑seq dataset, consider only the cells collected at 48 hours post‑infection (hpi = '48h'). For each infection condition (values in the 'infection' column: UI, Tg, Pf, Lm), compute the proportion of cells annotated as 'extravillous trophoblast' (cell_type = 'extravillous trophoblast'). Then calculate the fold‑change of each infection's proportion relative to the uninfected control (UI). Report the infection condition that exhibits the highest fold‑change (i.e., the strongest enrichment of extravillous trophoblast at 48 h compared to UI). Answer format: a single infection label (shape: label).\n",
      "Answer: Tg\n",
      "\n",
      "Task 10\n",
      "Description: In the dataset, consider only Hofbauer cells sampled at 24 hours post‑infection. For each infection condition (Lm, Tg, Pf) perform a differential expression analysis versus the uninfected (UI) cells using Scanpy's Wilcoxon rank‑sum test (all genes, no preprocessing). Count the number of genes that satisfy adjusted p‑value < 0.05 and log2‑fold‑change > 0.25. Which infection condition has the largest such count? Answer format: a single infection label (one of UI, Lm, Tg, Pf).\n",
      "Answer: Pf\n",
      "\n",
      "Task 11\n",
      "Description: In the dataset, consider only the cells annotated as \"Hofbauer cell\". For each infection condition (UI, Lm, Pf, Tg), compute the average raw (unnormalized) count of the gene IL6 across those Hofbauer cells. Identify the infection condition with the highest mean IL6 expression. Answer format: a single two‑letter label exactly as it appears in the \"infection\" column (shape: label).\n",
      "Answer: Pf\n",
      "\n",
      "Task 12\n",
      "Description: Within the cell type 'Hofbauer cell', determine which infection condition ('UI', 'Tg', 'Pf', or 'Lm') exhibits the highest average expression of the gene CXCL10. Report the answer as the exact infection label (case-sensitive) with shape: label.\n",
      "Answer: Pf\n",
      "\n",
      "Task 13\n",
      "Description: From the provided placenta infection scRNA‑seq dataset, compute for each cell type the proportion of cells in each infection condition (UI, Lm, Pf, Tg). For each infected condition (Lm, Pf, Tg) calculate the enrichment as (proportion in infected) / (proportion in UI). Identify the cell‑type and infection condition pair with the maximal enrichment value (exclude UI). Return the answer as a pair formatted as CELLTYPE::INFECTION (exact cell‑type label as in the 'cell_type' column, infection label as in the 'infection' column).\n",
      "Answer: Hofbauer cell::Tg\n",
      "\n",
      "Task 14\n",
      "Description: Identify the cell type that exhibits the greatest enrichment (highest fold change of proportion) in Plasmodium falciparum (Pf) infected samples compared to uninfected (UI) samples at the 24 hour time point. Answer format: a single lowercase label without spaces (shape: label).\n",
      "Answer: fibroblast\n",
      "\n",
      "Dataset: /mlbio_scratch/panigrah/agents/data/cell/paper3/data.h5ad\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Description: Within the scRNA‑seq dataset of mouse barrel cortex, consider only cells annotated with the cluster_name \"Baseline\". Perform a differential expression analysis comparing the two experimental conditions (Deprived vs Control) using a Wilcoxon rank‑sum test on log‑normalized counts. Identify the gene that is significantly up‑regulated in the Deprived condition (adjusted p‑value < 0.05) and shows the largest positive log2 fold‑change. Provide the answer as a single HGNC gene symbol (shape: gene_symbol).\n",
      "Answer: Uba52\n",
      "\n",
      "Task 1\n",
      "Description: Using the provided scRNA‑seq dataset, compute the proportion of cells belonging to each transcriptionally defined cluster (column 'cluster_name') within each experimental condition ('Control' and 'Deprived'). Calculate the log2 fold‑change of the Deprived proportion over the Control proportion for every cluster (add a tiny epsilon = 1e‑6 to avoid division by zero). Report the exact cluster name that exhibits the highest positive log2 fold‑change (i.e., the greatest enrichment in the Deprived condition). Answer format: a single label (case‑preserved cluster name).\n",
      "Answer: Interferon-response\n",
      "\n",
      "Task 2\n",
      "Description: Determine which cell cluster (as labeled in the 'cluster_name' column) exhibits the greatest absolute change in its proportion of cells between the 'Control' and 'Deprived' experimental conditions. Report the cluster name exactly as it appears (case‑sensitive, including hyphens and spaces). Answer format: a single label (shape: label).\n",
      "Answer: Interferon-response\n",
      "\n",
      "Task 3\n",
      "Description: Using the provided scRNA‑seq AnnData object (data.h5ad), perform a log‑normalization (log1p) of the raw counts, then compute the average expression of each gene across cells in the two experimental conditions (Control vs Deprived). Calculate the log2 fold‑change (log2FC) for each gene as log2((mean_Deprived + 1e-6) / (mean_Control + 1e-6)). Identify the gene with the largest positive log2FC (i.e., most up‑regulated in the Deprived condition). Answer format: a single mouse gene symbol (shape: gene_symbol).\n",
      "Answer: 6720483E21Rik\n",
      "\n",
      "Task 4\n",
      "Description: Using the provided scRNA‑seq dataset of mouse barrel cortex microglia, normalize the raw counts to 10,000 per cell and apply log1p transformation. Then perform differential expression between the 'Deprived' and 'Control' conditions (Wilcoxon test, grouping by the 'condition' column). Report the official gene symbol of the gene with the highest positive log2 fold‑change (i.e., the most up‑regulated gene in the Deprived condition). Answer format: a single gene symbol (shape: gene_symbol).\n",
      "Answer: 6720483E21Rik\n",
      "\n",
      "Task 5\n",
      "Description: Using the provided scRNA‑seq dataset, compute the enrichment of each transcriptional cluster (column 'final_clusters') in the 'Deprived' condition relative to the 'Control' condition. For each cluster, calculate the log2 enrichment ratio: log2[(fraction of Deprived cells in the cluster) / (fraction of Control cells in the cluster)]. Identify the cluster with the highest positive log2 enrichment (i.e., most over‑represented in Deprived). Answer format: a single cluster label as it appears in 'final_clusters' (shape: label).\n",
      "Answer: 8\n",
      "\n",
      "Task 6\n",
      "Description: Consider the provided mouse barrel-cortex scRNA‑seq dataset. Identify the cell type that contains the largest number of cells among those with at least 200 cells. Using the PCA embedding stored in `adata.obsm['X_pca']` (or compute PCA if missing), calculate the integrated Local Inverse Simpson’s Index (iLISI) for each donor (`donor_id`) within this cell type (using 30 nearest neighbours in PCA space). Report the donor identifier that has the lowest average iLISI value. Answer format: a single donor identifier exactly as it appears in `adata.obs['donor_id']` (shape: label).\n",
      "Answer: pooled_Deprived_P7\n",
      "\n",
      "Task 7\n",
      "Description: Using the provided single-cell RNA‑seq dataset of mouse barrel cortex microglia, perform differential expression analysis between the 'Deprived' and 'Control' conditions within each subcluster defined by the 'cluster_name' annotation. For each subcluster, count the number of genes that are significantly up‑regulated in the Deprived condition (log2 fold‑change > 0.5 and Benjamini‑Hochberg adjusted p‑value < 0.05). Identify the subcluster with the highest such count. Answer format: a single label matching exactly one of the 'cluster_name' values (shape: label).\n",
      "Answer: G1/M Phase early\n",
      "\n",
      "Task 8\n",
      "Description: Identify the mouse gene that shows the strongest up‑regulation (largest positive log2 fold‑change) in microglial cells from the 'Deprived' condition compared to the 'Control' condition, after log‑normalising the raw counts (CP10K) and performing differential expression using the Wilcoxon rank‑sum test (no subsampling). Answer format: a single gene symbol in title case (shape: gene_symbol).\n",
      "Answer: Slfn14\n",
      "\n",
      "Task 9\n",
      "Description: Compare the two experimental conditions (Control vs Deprived) within each cell type present in the dataset. For each cell type, consider only those with at least 30 cells in each condition. Perform differential expression (Wilcoxon rank‑sum) between Deprived (treated) and Control (reference) and identify genes with adjusted p‑value < 0.05 and positive log2 fold‑change. Compute the mean log2 fold‑change of these significantly up‑regulated genes for each qualifying cell type. Return the cell‑type label (exactly as it appears in adata.obs['cell_type']) that shows the largest absolute mean log2 fold‑change. Answer format: a single label (shape: label).\n",
      "Answer: microglial cell\n",
      "\n",
      "Dataset: /mlbio_scratch/panigrah/agents/data/cell/paper5/data.h5ad\n",
      "----------------------------------------------------------------------------------------------------\n",
      "Task 0\n",
      "Description: Identify the author_cell_type (value in the 'author_cell_type' column of adata.obs) that exhibits the largest number of significantly up‑regulated genes (adjusted p‑value < 0.05 and log2 fold‑change > 0) when comparing cytokine‑treated cells versus no‑cytokine cells using Wilcoxon rank‑sum test. Answer format: a single cell‑type label (shape: label).\n",
      "Answer: TOM_VCT\n",
      "\n",
      "Task 1\n",
      "Description: Within the single-cell RNA‑seq dataset of human trophoblast organoids, consider only cells annotated as \"extravillous trophoblast\". Perform a differential expression analysis comparing the \"cytokines\" treatment group against the \"no_cytokines\" control group (using standard library‑size normalization, log‑transformation, and the Wilcoxon rank‑sum test on all genes). Identify the gene that shows the largest positive log2‑fold‑change (i.e., strongest up‑regulation) in the cytokine‑treated cells relative to control. Answer format: a single HGNC gene symbol (shape: gene_symbol).\n",
      "Answer: CSH1\n",
      "\n",
      "Task 2\n",
      "Description: Within the syncytiotrophoblast cell (SCT) population (author_cell_type == \"SCT\"), compute the average expression of each gene for cytokine‑treated cells (treatment == \"cytokines\") and for control cells (treatment == \"no_cytokines\"). Add a pseudocount of 1 to each average, calculate the log2 fold‑change (log2[(treated+1)/(control+1)]) for every gene, and report the gene symbol (HGNC uppercase) that has the highest positive log2 fold‑change. Answer format: a single gene symbol (shape: gene_symbol).\n",
      "Answer: TCHH\n",
      "\n",
      "Task 3\n",
      "Description: Using the provided single-cell RNA‑seq dataset, compare the two experimental conditions in the 'treatment' column (\"cytokines\" vs \"no_cytokines\"). For each cell type listed in the 'author_cell_type' column, compute the proportion of cells belonging to that type within each condition, then calculate the log2 fold‑change of the proportion (cytokines / no_cytokines). Identify the cell type with the largest positive log2 fold‑change (i.e., the greatest relative increase after cytokine treatment). Answer format: a single label exactly as it appears in the 'author_cell_type' column (shape: label).\n",
      "Answer: EVT_late_3\n",
      "\n",
      "Task 4\n",
      "Description: Using the provided single-cell RNA‑seq AnnData object, first normalize raw counts to 10 000 counts per cell (CP10K) and then apply log1p transformation. For each author‑annotated cell type (column 'author_cell_type'), compute the mean log‑expression of every gene separately in the two treatment groups ('cytokines' and 'no_cytokines'). Calculate the log2‑fold‑change for each gene as the difference of these means (treated minus control). Count, for each cell type, how many genes have log2‑fold‑change greater than 0.25. Report the cell type that has the largest such count. Answer format: a single label exactly as it appears in the 'author_cell_type' column (shape: label).\n",
      "Answer: EVT_late_3\n",
      "\n",
      "Task 5\n",
      "Description: For each cell type annotated in the 'cell_type' column, compute the average expression of every gene across cells under the two treatment conditions ('cytokines' and 'no_cytokines'). For each cell type, calculate the log2 fold‑change (log2FC) of each gene between the cytokine‑treated and untreated groups, adding a negligible pseudocount to avoid division by zero. Count how many genes have an absolute log2FC ≥ 1 (i.e., at least a 2‑fold change). Identify the cell type with the largest such gene count. Answer format: a single cell type label exactly as it appears in the 'cell_type' column (shape: label).\n",
      "Answer: syncytiotrophoblast cell\n",
      "\n",
      "Task 6\n",
      "Description: Identify the trophoblast cell type that exhibits the strongest treatment‑induced expression change, defined as the largest absolute log2 fold change (log2FC) observed for any gene when comparing cytokine‑treated cells versus no‑cytokine cells (averaged within each cell type). Answer format: a single lowercase label with underscores (shape: label).\n",
      "Answer: syncytiotrophoblast_cell\n",
      "\n",
      "Task 7\n",
      "Description: Using the provided single-cell RNA‑seq dataset of human trophoblast organoids, perform differential expression analysis between cytokine‑treated and untreated cells within each broad cell type (syncytiotrophoblast cell, extravillous trophoblast, placental villous trophoblast). Count the number of genes that are significantly up‑regulated in the cytokine condition (adjusted p‑value < 0.05 and log2‑fold‑change > 0.25). Identify the cell type with the greatest number of such up‑regulated genes. Answer format: a single label matching exactly one of the values in the 'cell_type' column (shape: label).\n",
      "Answer: extravillous trophoblast\n",
      "\n",
      "Task 8\n",
      "Description: Identify the gene that is most significantly up‑regulated by cytokine treatment within the trophoblast sub‑type 'EVT_proliferating', after accounting for donor variability. Use a Wilcoxon rank‑genes‑groups test per donor, combine p‑values across donors with Fisher's method, average log‑fold‑changes, keep only genes with positive average logFC, apply Benjamini‑Hochberg correction, and report the gene symbol (HGNC uppercase) with the smallest adjusted combined p‑value. Answer format: a single HGNC gene symbol (shape: gene_symbol).\n",
      "Answer: CXCL8\n",
      "\n",
      "Task 9\n",
      "Description: Within the VCT cell type (author_cell_type == \"VCT\"), compute the log2 fold change (log2FC) for each gene between cytokine‑treated cells (treatment == \"cytokines\") and control cells (treatment == \"no_cytokines\") using CPM normalization followed by log1p transformation and mean expression per group. Return the HGNC gene symbol (uppercase) of the gene with the highest positive log2FC. Answer format: a single gene_symbol.\n",
      "Answer: SRCIN1\n",
      "\n",
      "Task 10\n",
      "Description: Using the provided single‑cell RNA‑seq dataset of human trophoblast organoids, perform a standard scRNA‑seq analysis pipeline (QC filtering, normalization, log‑transformation, selection of the top 3000 highly variable genes, and differential expression testing between the 'cytokines' and 'no_cytokines' treatment groups within each author‑provided cell type). Count the number of genes that are significantly up‑regulated in the cytokine‑treated condition (log2 fold‑change > 0.25 and Benjamini‑Hochberg adjusted p‑value < 0.05). Report the cell‑type label (exactly as it appears in the 'author_cell_type' column) that has the largest number of such up‑regulated genes. Answer format: a single label (shape: label).\n",
      "Answer: SCT\n",
      "\n"
     ]
    }
   ],
   "source": [
    "for dataset_path, tasks in tasks_by_dataset.items():\n",
    "    print(f\"Dataset: {dataset_path}\")\n",
    "    print(100 * '-')\n",
    "    for tid, task in enumerate(tasks):\n",
    "        print(f\"Task {tid}\")\n",
    "        print(f\"Description: {task['task_description']}\")\n",
    "        print(f\"Answer: {task['ground_truth_answer']}\")\n",
    "        print()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "b1f113dd-d9ea-444b-9b66-dccbc1de246c",
   "metadata": {},
   "outputs": [],
   "source": [
    "duplicates_by_dataset= {\n",
    "    # \"/mlbio_scratch/panigrah/agents/data/nature/paper2/processed_MuSC_data.h5ad\": [[5, 6]],\n",
    "    # \"/mlbio_scratch/panigrah/agents/data/nature/paper2/processed_myonuclei_data.h5ad\": [[10, 9]],\n",
    "    \"/mlbio_scratch/panigrah/agents/data/nature/paper6/data.h5ad\": [[4, 0, 2]],\n",
    "    \"/mlbio_scratch/panigrah/agents/data/cell/paper2/data.h5ad\": [[6, 8]],\n",
    "    # \"/mlbio_scratch/panigrah/agents/data/cell/paper3/data.h5ad\": [[3, 8]],\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "970d1887-71ac-472b-a118-1d2b47b7075b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Tasks after dataset-wise filtration: 283\n"
     ]
    }
   ],
   "source": [
    "clean_tasks = get_clean_tasks(tasks_by_dataset, duplicates_by_dataset)\n",
    "print(f\"Tasks after dataset-wise filtration: {len(clean_tasks)}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "5482cb9d-92f0-4e6f-8698-ae8e3c6f88ce",
   "metadata": {},
   "outputs": [],
   "source": [
    "clean_tasks_path = tasks_dir / \"clean.json\"\n",
    "save_data(clean_tasks, clean_tasks_path)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
